<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-04-10" />

<title>data_download</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Macrophage signatures in TCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_macs">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">data_download</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-04-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_macs/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190409code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190409)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190409code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190409)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstreed4dc52756163d8a33c0c3a06bf89e78deac9df9ctargetblankd4dc527a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_macs/tree/d4dc52756163d8a33c0c3a06bf89e78deac9df9c" target="_blank">d4dc527</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstreed4dc52756163d8a33c0c3a06bf89e78deac9df9ctargetblankd4dc527a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/GSE46903_Affymetrix_Reanalyzed_data.txt.gz
    Ignored:    data/GSE46903_Affymetrix_metadata.txt.gz
    Ignored:    data/GSE46903_RAW.tar
    Ignored:    data/GSE46903_non-normalized.txt.gz

Untracked files:
    Untracked:  data/GSE46903_nonnormalized copy.tsv
    Untracked:  output/processed_lumi_46903.Rds
    Untracked:  output/processed_lumi_46903.tsv
    Untracked:  output/processed_lumi_exprs.txt

Unstaged changes:
    Modified:   analysis/invitro_mac_dge.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<div id="purpose-of-this-analysis" class="section level3">
<h3>Purpose of this analysis</h3>
<p>Find ranked differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes. These gene sets will be used for our downstream analyses.</p>
<p>The source for this method is:</p>
<blockquote>
<p>Gubin, Matthew M., Ekaterina Esaulova, Jeffrey P. Ward, Olga N. Malkova, Daniele Runci, Pamela Wong, Takuro Noguchi, et al. 2018. “High-Dimensional Analysis Delineates Myeloid and Lymphoid Compartment Remodeling during Successful Immune-Checkpoint Cancer Therapy.” Cell 175 (4): 1014–30.e19.</p>
</blockquote>
<p>This is a summary of the method from Gubin et al.:</p>
<blockquote>
<p>Expression data was retrieved from GEO with GSE46903 ID. For analysis, we selected only monocyte and macrophage samples. According to the dataset description, peripheral blood monocytes were stimulated with GM-CSF or M-CSF, and subsequently stimulated with different compounds. We excluded all M-CSF-derived macrophages and non-72 hour time points. M0-GM-CSF 72 hour samples were used as baseline. We used limma package to perform differential expression in order to understand gene changes upon treatments compared to baseline. For each comparison, gene lists were obtained and ranked by t-statistics. To compare human macrophages under different stimuli with mouse monocytes/macrophages from our scRNAseq data, we compared genes, upregulated or downregulated inside each scRNAseq cluster with genes, upregulated or downregulated in each human monocyte/macrophage treatment condition. The goal was to get a list of signed scores that will represent the level of concordance between treatments and each cluster.</p>
</blockquote>
<blockquote>
<p>For each cluster, we created two gene sets: 100 upregulated and downregulated genes. GSEA (with fgsea package) was performed using this gene set against ranked gene lists from treatment comparisons. To obtain a score, we subtracted the normalized enrichment score (NES) of downregulated genes from NES of upregulated genes. To further investigate the correlation (or lack thereof) between clusters and treatments, we visualized genes associated with each cluster in human macrophages. We reduced the dimensions of macrophage datasets with Rtsne package and plotted each sample in two dimensions using the ggplot2 package. Samples were then highlighted using Z-scored expression profiles of cluster-associated genes.</p>
</blockquote>
</div>
<div id="data-source" class="section level3">
<h3>Data Source</h3>
<p>The gene chip data used to derive differential expression for various macrophage subtypes was first published here:</p>
<blockquote>
<p>Xue J, Schmidt SV, Sander J, Draffehn A et al. Transcriptome-based network analysis reveals a spectrum model of human macrophage activation. Immunity 2014 Feb 20;40(2):274-88. PMID: 24530056</p>
</blockquote>
<p>The original experiment had two conditions.</p>
<ol style="list-style-type: decimal">
<li><p>Cell culture with monocytes exposed to growth factor and a number of different cytokines to induce different activation states. Microarray expression data from cell culture were generated with Illumina HumanHT-12 v3.0 beadchips.</p></li>
<li><p>Gene sets derived in cell culture were used to categorize expression profiles derived from alveolar macrophages from patients with COPD and asthma. Affymetrix chips were used to measure expression in human samples.</p></li>
</ol>
<p>We are interested in the first condition (macrophages in culture). Associated metadata is posted at the GEO:</p>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903">The experiment overview</a></p></li>
<li><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1140443">The record for the control group</a></p>
<ul>
<li><p>M0-GM-CSF treated monocytes at 72hours</p></li>
<li><p>Additional details about how the data were generated</p></li>
</ul></li>
<li><p>Expression matrices in which “The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite”</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?view=data&amp;acc=GSM1140443&amp;id=38019&amp;db=GeoDb_blob97">Example expression matrix</a> from the control group</li>
</ul></li>
</ul>
<p><em>We will obtain the raw data from the Illumina expression beadchips and normalize ourselves.</em></p>
</div>
<div id="set-up-workspace" class="section level3">
<h3>Set up workspace</h3>
<p>Clear workspace prior to analysis and load necessary packages</p>
<pre class="r"><code>rm(list = ls())</code></pre>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;GEOquery&quot;)
library(&quot;lumi&quot;)
library(&quot;lumiHumanIDMapping&quot;)</code></pre>
</div>
</div>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>GEO ID is: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903">GSE46903</a></p>
<pre class="r"><code># download normalized data with full metadata from GEO as a list of ExpressionSets
gse46903 &lt;- getGEO(&quot;GSE46903&quot;)</code></pre>
<pre><code>Found 1 file(s)</code></pre>
<pre><code>GSE46903_series_matrix.txt.gz</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  ID_REF = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre><code>File stored at: </code></pre>
<pre><code>/var/folders/dc/481f588d6ksdf0wn34p80stw0000gn/T//RtmpQ8ISai/GPL6947.soft</code></pre>
<p>Accessing raw data from the GEO: <a href="http://genomicsclass.github.io/book/pages/GEOquery.html" class="uri">http://genomicsclass.github.io/book/pages/GEOquery.html</a></p>
<p>Note about raw vs normalized files from GEO: <a href="https://www.biostars.org/p/259637/" class="uri">https://www.biostars.org/p/259637/</a></p>
<pre class="r"><code># download raw data
filePaths = getGEOSuppFiles(&quot;GSE46903&quot;, 
                            makeDirectory = TRUE)</code></pre>
<pre class="bash"><code># in terminal
pwd
ls
cd GSE46903
ls
mv GSE* ~/tcga_macs/data
cd ..
rmdir GSE46903
ls</code></pre>
<pre><code>/Users/mleukam/tcga_macs
GSE46903
README.md
_workflowr.yml
analysis
code
data
docs
output
tcga_macs.Rproj
GSE46903_Affymetrix_Reanalyzed_data.txt.gz
GSE46903_Affymetrix_metadata.txt.gz
GSE46903_RAW.tar
GSE46903_non-normalized.txt.gz
README.md
_workflowr.yml
analysis
code
data
docs
output
tcga_macs.Rproj</code></pre>
<pre class="bash"><code>cd ~/tcga_macs/data
gunzip GSE46903_non-normalized.txt.gz</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] lumiHumanIDMapping_1.10.1 AnnotationDbi_1.44.0     
 [3] IRanges_2.16.0            S4Vectors_0.20.1         
 [5] lumi_2.34.0               GEOquery_2.50.5          
 [7] Biobase_2.42.0            BiocGenerics_0.28.0      
 [9] forcats_0.4.0             stringr_1.4.0            
[11] dplyr_0.8.0.1             purrr_0.3.2              
[13] readr_1.3.1               tidyr_0.8.3              
[15] tibble_2.1.1              ggplot2_3.1.1            
[17] tidyverse_1.2.1          

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                backports_1.1.3            
  [3] workflowr_1.2.0             plyr_1.8.4                 
  [5] lazyeval_0.2.2              splines_3.5.3              
  [7] BiocParallel_1.16.6         GenomeInfoDb_1.18.2        
  [9] digest_0.6.18               foreach_1.4.4              
 [11] htmltools_0.3.6             magrittr_1.5               
 [13] memoise_1.1.0               limma_3.38.3               
 [15] Biostrings_2.50.2           annotate_1.60.1            
 [17] modelr_0.1.4                matrixStats_0.54.0         
 [19] askpass_1.1                 siggenes_1.56.0            
 [21] prettyunits_1.0.2           colorspace_1.4-1           
 [23] blob_1.1.1                  rvest_0.3.2                
 [25] haven_2.1.0                 xfun_0.6                   
 [27] crayon_1.3.4                RCurl_1.95-4.12            
 [29] jsonlite_1.6                genefilter_1.64.0          
 [31] survival_2.44-1.1           iterators_1.0.10           
 [33] glue_1.3.1                  registry_0.5-1             
 [35] gtable_0.3.0                zlibbioc_1.28.0            
 [37] XVector_0.22.0              DelayedArray_0.8.0         
 [39] Rhdf5lib_1.4.3              HDF5Array_1.10.1           
 [41] scales_1.0.0                DBI_1.0.0                  
 [43] rngtools_1.3.1              bibtex_0.4.2               
 [45] Rcpp_1.0.1                  xtable_1.8-3               
 [47] progress_1.2.0              bumphunter_1.24.5          
 [49] bit_1.1-14                  mclust_5.4.3               
 [51] preprocessCore_1.44.0       httr_1.4.0                 
 [53] RColorBrewer_1.1-2          pkgconfig_2.0.2            
 [55] reshape_0.8.8               XML_3.98-1.19              
 [57] locfit_1.5-9.1              tidyselect_0.2.5           
 [59] rlang_0.3.4                 munsell_0.5.0              
 [61] cellranger_1.1.0            tools_3.5.3                
 [63] cli_1.1.0                   generics_0.0.2             
 [65] RSQLite_2.1.1               broom_0.5.2                
 [67] evaluate_0.13               yaml_2.2.0                 
 [69] knitr_1.22                  bit64_0.9-7                
 [71] fs_1.2.7                    beanplot_1.2               
 [73] methylumi_2.28.0            nlme_3.1-138               
 [75] doRNG_1.7.1                 nor1mix_1.2-3              
 [77] xml2_1.2.0                  biomaRt_2.38.0             
 [79] compiler_3.5.3              rstudioapi_0.10            
 [81] curl_3.3                    affyio_1.52.0              
 [83] stringi_1.4.3               GenomicFeatures_1.34.7     
 [85] minfi_1.28.4                lattice_0.20-38            
 [87] Matrix_1.2-17               multtest_2.38.0            
 [89] pillar_1.3.1                BiocManager_1.30.4         
 [91] data.table_1.12.2           bitops_1.0-6               
 [93] rtracklayer_1.42.2          GenomicRanges_1.34.0       
 [95] R6_2.4.0                    affy_1.60.0                
 [97] KernSmooth_2.23-15          nleqslv_3.3.2              
 [99] codetools_0.2-16            MASS_7.3-51.4              
[101] assertthat_0.2.1            rhdf5_2.26.2               
[103] SummarizedExperiment_1.12.0 openssl_1.3                
[105] pkgmaker_0.27               rprojroot_1.3-2            
[107] withr_2.1.2                 GenomicAlignments_1.18.1   
[109] Rsamtools_1.34.1            GenomeInfoDbData_1.2.0     
[111] mgcv_1.8-28                 hms_0.4.2                  
[113] quadprog_1.5-5              grid_3.5.3                 
[115] base64_2.0                  rmarkdown_1.12             
[117] DelayedMatrixStats_1.4.0    illuminaio_0.24.0          
[119] git2r_0.25.2                lubridate_1.7.4            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
