<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Home</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Macrophage signatures in TCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_macs">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Home</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-11-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 2 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_macs/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstreec9fa79d9c0d3542af1f42453b4bdbe2cf6961185targetblankc9fa79da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_macs/tree/c9fa79d9c0d3542af1f42453b4bdbe2cf6961185" target="_blank">c9fa79d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstreec9fa79d9c0d3542af1f42453b4bdbe2cf6961185targetblankc9fa79da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    all.pdf
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/160genes_ensembl.gencode28.txt
    Ignored:    data/GSE46903_Affymetrix_Reanalyzed_data.txt.gz
    Ignored:    data/GSE46903_Affymetrix_metadata.txt.gz
    Ignored:    data/GSE46903_RAW.tar
    Ignored:    data/GSE46903_non-normalized.txt
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/gse_46903.rds
    Ignored:    data/htseq_counts/
    Ignored:    data/kyle.gdc_tcga_bam_metadata.txt
    Ignored:    output/AUC_comparison.pdf
    Ignored:    output/Combined_down_plots.pdf
    Ignored:    output/Combined_up_plots.pdf
    Ignored:    output/Fig1.png
    Ignored:    output/PFS_CoxPH.pdf
    Ignored:    output/Rplot.pdf
    Ignored:    output/arrange.pdf
    Ignored:    output/ave_scores_gsva_by_tumor.rds
    Ignored:    output/beeswarm_plot.pdf
    Ignored:    output/beeswarm_pub.pdf
    Ignored:    output/boxplot_total_pub.pdf
    Ignored:    output/combined_boxplots.pdf
    Ignored:    output/combined_up_plot.pdf
    Ignored:    output/coxph_pfs.pdf
    Ignored:    output/cpxph_os.pdf
    Ignored:    output/cv.lasso.merck.rds
    Ignored:    output/cv.lasso.rds
    Ignored:    output/dge_frame.csv
    Ignored:    output/downreg_c1.csv
    Ignored:    output/downreg_c2.csv
    Ignored:    output/downreg_c3.csv
    Ignored:    output/downreg_c4.csv
    Ignored:    output/downreg_c5.csv
    Ignored:    output/downreg_c6.csv
    Ignored:    output/downreg_c7.csv
    Ignored:    output/expr_matrix.csv
    Ignored:    output/expr_matrix.rds
    Ignored:    output/expression_set.rds
    Ignored:    output/figures/
    Ignored:    output/filtered.total.phenodata.csv
    Ignored:    output/filtered.total.phenodata.rds
    Ignored:    output/flat_total_counts.rds
    Ignored:    output/flat_total_pheno.rds
    Ignored:    output/gset_ids_complete.rds
    Ignored:    output/gset_ids_plusnew.rds
    Ignored:    output/heatmaps_fig3.pdf
    Ignored:    output/hm_chol.pdf
    Ignored:    output/imtx_expressionset.rds
    Ignored:    output/imtx_gsva_17geneset_results.rds
    Ignored:    output/imtx_matrix.csv
    Ignored:    output/km_cluster_assignments.csv
    Ignored:    output/lasso.min.model.merck.rds
    Ignored:    output/lasso.min.model.rds
    Ignored:    output/lasso.test.data.merck.rds
    Ignored:    output/lasso.test.data.rds
    Ignored:    output/lasso.train.data.merck.rds
    Ignored:    output/lasso.train.data.rds
    Ignored:    output/mac_eset.rds
    Ignored:    output/mac_eset_w_clusters.rds
    Ignored:    output/nested_list_dge_top50_updown.rds
    Ignored:    output/normalized_probe_intensities.Rds
    Ignored:    output/os_km_curves.pdf
    Ignored:    output/os_km_plots.pdf
    Ignored:    output/processed_lumi_46903.Rds
    Ignored:    output/processed_lumi_46903.tsv
    Ignored:    output/processed_lumi_exprs.txt
    Ignored:    output/pub_boxplot.pdf
    Ignored:    output/score_matrix.csv
    Ignored:    output/selected.total.counts.csv
    Ignored:    output/summary heatmap.pdf
    Ignored:    output/table_c1.csv
    Ignored:    output/table_c2.csv
    Ignored:    output/table_c3.csv
    Ignored:    output/table_c4.csv
    Ignored:    output/table_c5.csv
    Ignored:    output/table_c6.csv
    Ignored:    output/table_c7.csv
    Ignored:    output/tcga_counts_dataframe_list.rds
    Ignored:    output/tcga_counts_list.rds
    Ignored:    output/tcga_gdc_manifest.rds
    Ignored:    output/tcga_gsva_17geneset_results.rds
    Ignored:    output/tcga_metadata_list.rds
    Ignored:    output/tcga_total_counts.csv
    Ignored:    output/univariate_p_values.pdf
    Ignored:    output/upreg_c1.csv
    Ignored:    output/upreg_c2.csv
    Ignored:    output/upreg_c3.csv
    Ignored:    output/upreg_c4.csv
    Ignored:    output/upreg_c5.csv
    Ignored:    output/upreg_c6.csv
    Ignored:    output/upreg_c7.csv

Untracked files:
    Untracked:  analysis/expanded_gsva.Rmd
    Untracked:  analysis/expanded_gsva_analysis.Rmd
    Untracked:  analysis/gsva_results_exploration.Rmd
    Untracked:  analysis/survival.Rmd
    Untracked:  data/IMTX_datasets/
    Untracked:  data/xcel_signatures.xlsx
    Untracked:  output/PFS_CoxPH-concordance.tex
    Untracked:  output/PFS_CoxPH.Rnw
    Untracked:  output/PFS_CoxPH.log
    Untracked:  output/PFS_CoxPH.synctex.gz
    Untracked:  output/PFS_CoxPH.tex
    Untracked:  output/Rplot01.png
    Untracked:  output/coxph_pfs-concordance.tex
    Untracked:  output/coxph_pfs.Rnw
    Untracked:  output/coxph_pfs.log
    Untracked:  output/coxph_pfs.synctex.gz
    Untracked:  output/coxph_pfs.tex
    Untracked:  output/cpxph_os-concordance.tex
    Untracked:  output/cpxph_os.Rnw
    Untracked:  output/cpxph_os.log
    Untracked:  output/cpxph_os.synctex.gz
    Untracked:  output/cpxph_os.tex
    Untracked:  output/univariate_p_values-concordance.tex
    Untracked:  output/univariate_p_values.Rnw
    Untracked:  output/univariate_p_values.log
    Untracked:  output/univariate_p_values.synctex.gz
    Untracked:  output/univariate_p_values.tex

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/ss_gsea.Rmd
    Modified:   analysis/tcga_format.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/c9fa79d9c0d3542af1f42453b4bdbe2cf6961185/analysis/index.Rmd" target="_blank">c9fa79d</a>
</td>
<td>
mleukam
</td>
<td>
2019-11-09
</td>
<td>
wflow_publish(“analysis/index.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/0188666004a3e503b05be00244ef7524d98b3bc2/docs/index.html" target="_blank">0188666</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/202b4f473ac7efcabeb4b97e3f821e7906906bc2/analysis/index.Rmd" target="_blank">202b4f4</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-25
</td>
<td>
added link to GSVA results exploration
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/dd3f2277aac1e53cd86b8612ab2ed9221a76630a/docs/index.html" target="_blank">dd3f227</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/66464dba6507d2ad1dc446ff7c7fb046c2f99749/analysis/index.Rmd" target="_blank">66464db</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
<td>
updates to homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/4f63f2cae6c35ca57cdb79799ab7e4ceb219a4cf/docs/index.html" target="_blank">4f63f2c</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/499940d74c3d7a6e5f292aa066598410d8ac4bd8/analysis/index.Rmd" target="_blank">499940d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
minor text update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/08836cd7350cc17a30772f404d6bc1e1fcaa73dc/docs/index.html" target="_blank">08836cd</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/9b1985d0a311ea7bfc5f242faf949848ee301292/analysis/index.Rmd" target="_blank">9b1985d</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
added summary images
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/8f7d6840af2071b9f16f976507edc4d2fe988ecf/docs/index.html" target="_blank">8f7d684</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/5648679449d35ae85b71d4985e025ce5058f1a4c/analysis/index.Rmd" target="_blank">5648679</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
updates after new TCGA format notebook
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/db31ea42b57bc29a2c7072debf34039a4506b4cf/docs/index.html" target="_blank">db31ea4</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/e32a26708fe84e8e3ab9201c37b069b943e11a1e/analysis/index.Rmd" target="_blank">e32a267</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-09
</td>
<td>
update homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/788b028234e0a85eabd9431383ec9d3dd0c9ef17/docs/index.html" target="_blank">788b028</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-05
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/25733dca8b5bf5ec76e3175a848b29f1641a0171/analysis/index.Rmd" target="_blank">25733dc</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-05
</td>
<td>
major updates including clustering and correlation
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/eeb13016ced56f75963afdfdc363e5b386fc496a/docs/index.html" target="_blank">eeb1301</a>
</td>
<td>
mleukam
</td>
<td>
2019-05-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/3ff66a482aa909617869ea27d840b4ba533e57e5/analysis/index.Rmd" target="_blank">3ff66a4</a>
</td>
<td>
mleukam
</td>
<td>
2019-05-08
</td>
<td>
rename and update preprocessing notebook, update index links
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/7d5810027dc196ca521842a610e3a1fee88f4fb6/docs/index.html" target="_blank">7d58100</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/9e174951cf216ce095530da3a6255e9a5328da83/analysis/index.Rmd" target="_blank">9e17495</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
update home page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/9891f0da1fdd3828a8fdc567b5505558ad96379d/docs/index.html" target="_blank">9891f0d</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/2957568961555117a6dc6d969c53e06d89d6bf0e/analysis/index.Rmd" target="_blank">2957568</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
update home page
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/94fdc7ec5f90d8bdbd570fe594daa99c6f27250d/docs/index.html" target="_blank">94fdc7e</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/d8458b9403f9ab5c62e498efb87bcbd28c3caef0/analysis/index.Rmd" target="_blank">d8458b9</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-11
</td>
<td>
major updates to mac dge including subsets, updates to index
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/363b5df5770f40d8b98527c9ced243c7aa7a06d9/docs/index.html" target="_blank">363b5df</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/9910c1dc1011a4b29711916c5e257236b0e15659/analysis/index.Rmd" target="_blank">9910c1d</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Change the theme
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/70f9e015216659359a3e6ba484a683e561bed3c5/docs/index.html" target="_blank">70f9e01</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/17f3020720fb6faa9b18e0fae66c7741d157c6d3/analysis/index.Rmd" target="_blank">17f3020</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
beginning the text on the homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/bae7e193ef27a69f574dd3a3083b5ea031133c09/docs/index.html" target="_blank">bae7e19</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/3428b498b8d13d86d44fd5dc8b335ccd87c39b38/analysis/index.Rmd" target="_blank">3428b49</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>Welcome to the lab notebook for our research project describing the distribution of macrophage gene signatures across various tumor types in The Cancer Genome Atlas (TCGA) data.</p>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>A T cell inflamed signature has been defined and described across tumor types in TCGA that corresponds to response to T cell directed immunotherapy. Is there a similar signature for macrophages that might predict the tumors most likely to respond to macrophage-directed immunotherapy? One challenge has been the diversity and plasticity of activated macrophage states compared to cytotoxic T cells. This project will use experimentally derived activation gene signatures for a range of macrophage subtypes and use high dimensional clustering to computationally define macrophage subsets. The distribution of these subsets will then be examined in multiple tumor types in TCGA data.</p>
</div>
<div id="defining-macrophage-gene-sets" class="section level2">
<h2>Defining macrophage gene sets</h2>
<p>Derived differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes for downstream analyses. The original data is obtained from GEO, ID=<a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903">GSE46903</a>.</p>
<ul>
<li><a href="data_download.html">Download data from GEO</a></li>
<li><a href="preprocessing.html">Data cleaning</a></li>
<li><a href="clustering.html">Clustering and correlation analyses</a></li>
<li><a href="invitro_mac_dge.html">Differential gene expression to derive gene lists</a></li>
</ul>
<div id="figures" class="section level4">
<h4>Figures:</h4>
<p>Dissimilarity matrix showing correlation scores among samples:</p>
<p><img src="assets/corrmatrix.png" /></p>
<p>K-means clustering of treatment groups based on Pearson correlation coefficients (k =7):</p>
<p><img src="assets/k7cluster.png" /></p>
</div>
</div>
<div id="investigating-patterns-in-tgca-data" class="section level2">
<h2>Investigating patterns in TGCA data</h2>
<p>Apply gene lists from above to TCGA pancancer bulk RNAseq counts data</p>
<ul>
<li><a href="tcga_download.html">Download counts data from Genomic Data Commons</a></li>
<li><a href="tcga_format.html">Format counts data</a></li>
<li><a href="ss_gsea.html">Gene set variation analysis</a></li>
<li><a href="gsva_results_exploration.html">Exploration and analysis of GSVA results</a></li>
</ul>
<div id="figures-1" class="section level4">
<h4>Figures:</h4>
<p>Figure 1: derivation of macrophage signatures:</p>
<p>Figure 2: Beeswarm plots of upregulated macrophage cluster genes applied to TCGA data:</p>
<p><img src="assets/Combined_up_plots.png" /></p>
<p>Figure 2B: Boxplots for selected tumor types showing GSVA score for each cluster and T-cell inflammation signature:</p>
<p><img src="assets/combined_boxplots.png" /></p>
<p>Figure 3: Heatmaps of selected tumor types</p>
<p>Figure 4: IMTX survival/outcomes</p>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
