<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-04-09" />

<title>Differential gene expression: _in vitro_ macrophage model</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Macrophage signatures in TCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_macs">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential gene expression: <em>in vitro</em> macrophage model</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-04-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-16
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_macs/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190409code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190409)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190409code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190409)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstree907c8c6432543158aa4e20ef73c605ff3003c31btargetblank907c8c6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_macs/tree/907c8c6432543158aa4e20ef73c605ff3003c31b" target="_blank">907c8c6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstree907c8c6432543158aa4e20ef73c605ff3003c31btargetblank907c8c6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/GSE46903_Affymetrix_Reanalyzed_data.txt.gz
    Ignored:    data/GSE46903_Affymetrix_metadata.txt.gz
    Ignored:    data/GSE46903_RAW.tar
    Ignored:    data/GSE46903_non-normalized.txt
    Ignored:    data/GSE46903_nonnormalized copy.tsv
    Ignored:    data/gse_46903.rds
    Ignored:    output/expression_set.rds
    Ignored:    output/normalized_probe_intensities.Rds
    Ignored:    output/processed_lumi_46903.Rds
    Ignored:    output/processed_lumi_46903.tsv
    Ignored:    output/processed_lumi_exprs.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/907c8c6432543158aa4e20ef73c605ff3003c31b/analysis/invitro_mac_dge.Rmd" target="_blank">907c8c6</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-16
</td>
<td>
wet code for a table of DGE T values, do I even need a table of T
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/b964e86bf21d6619ba5aec5855d2f0973e134e4d/docs/invitro_mac_dge.html" target="_blank">b964e86</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/2d0c6dacc8e10b1be88c2aacf9b039f5889f6681/analysis/invitro_mac_dge.Rmd" target="_blank">2d0c6da</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
split dge and pca into separate notebooks
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/acfe8578d11e84e4450f2feb472a1338485725bc/docs/invitro_mac_dge.html" target="_blank">acfe857</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/b1389aef8b1d262e5c00456e5ea2f7f03349021a/analysis/invitro_mac_dge.Rmd" target="_blank">b1389ae</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-14
</td>
<td>
PCA and limma DGE
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/94fdc7ec5f90d8bdbd570fe594daa99c6f27250d/docs/invitro_mac_dge.html" target="_blank">94fdc7e</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/6b41c090d03b38c8e20e8d6ee459ce9d5f6c424c/docs/invitro_mac_dge.html" target="_blank">6b41c09</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/413333cf7bfae072abbcf67d2f2d93682fd6ecf5/analysis/invitro_mac_dge.Rmd" target="_blank">413333c</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
updated download and data cleaning from GEO
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/65a4fa44eb069b1d02b19cafd57e7ae382cca4c0/docs/invitro_mac_dge.html" target="_blank">65a4fa4</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
updated download of raw data and normalization from GEO
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/6cd5270cacf3ac8d11c7faea7ca4cb0de42b89c3/docs/invitro_mac_dge.html" target="_blank">6cd5270</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/1ecaa5cf559a8344497d369dc8687431dcb72dc0/analysis/invitro_mac_dge.Rmd" target="_blank">1ecaa5c</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
wflow_publish(“analysis/invitro_mac_dge.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/c43686198fa5b5f0718f028386ecd48036e5e4ce/docs/invitro_mac_dge.html" target="_blank">c436861</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/fcefd997d3b694d52c85f702a49784d945619e96/analysis/invitro_mac_dge.Rmd" target="_blank">fcefd99</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
successful raw data import and normalization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/1c4ad0246f938ca2348e7ab0df1a0b4e8a5631f0/docs/invitro_mac_dge.html" target="_blank">1c4ad02</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/2555a2a7e7eea7fccfeabb6e392926723e7cff25/analysis/invitro_mac_dge.Rmd" target="_blank">2555a2a</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
cleaned up quotation parts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/8ff4e151b900839aee8321314a9c5167224bb3f3/docs/invitro_mac_dge.html" target="_blank">8ff4e15</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8c06425b5aba4ece8ed25814b330d833d3e487dd/analysis/invitro_mac_dge.Rmd" target="_blank">8c06425</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
added protocols
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/363b5df5770f40d8b98527c9ced243c7aa7a06d9/docs/invitro_mac_dge.html" target="_blank">363b5df</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/17f3020720fb6faa9b18e0fae66c7741d157c6d3/analysis/invitro_mac_dge.Rmd" target="_blank">17f3020</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
beginning the text on the homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/6dc88381bf3cf9dd2d2ccb88e8e9467bbd9bd590/docs/invitro_mac_dge.html" target="_blank">6dc8838</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/13ee5e07deb1c432977fc7f2d21d334232961afd/analysis/invitro_mac_dge.Rmd" target="_blank">13ee5e0</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
working data download, added citations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/eacbbe0ca3e23418644b1c547f3846e5510fe778/docs/invitro_mac_dge.html" target="_blank">eacbbe0</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/b27b467677db6859f76f017c1be5e532adfebfe5/analysis/invitro_mac_dge.Rmd" target="_blank">b27b467</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
citations added and packages loaded
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear workspace prior to analysis and load necessary packages</p>
<pre class="r"><code>rm(list = ls())</code></pre>
</div>
<div id="load-packages" class="section level2">
<h2>Load Packages</h2>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;GEOquery&quot;)
library(&quot;lumi&quot;)
library(&quot;lumiHumanIDMapping&quot;)
library(&quot;lumiHumanAll.db&quot;)
library(&quot;limma&quot;)
library(&quot;annotate&quot;)
library(&quot;factoextra&quot;)
library(&quot;FactoMineR&quot;)
library(&quot;ggsci&quot;)
library(&quot;ggpubr&quot;)</code></pre>
</div>
<div id="read-data" class="section level2">
<h2>Read data</h2>
<pre class="r"><code># Expression set created in PCA_MCA notebook 
# Contains normalized expression matrix made from raw illumina beadchip data
# Also contains metadata from GEO
# Probe ID is unique sequence ID from lumiHumanIDMapping
mac_eset &lt;- readRDS(&quot;~/tcga_macs/output/expression_set.rds&quot;)

# Expression set created by lumi
# Contains raw reads from illumina beadchip
# variance stablized by log2 transformation
# normalized with RSN (robust spline normalization)
lumi_46903 &lt;- readRDS(&quot;~/tcga_macs/output/normalized_probe_intensities.Rds&quot;)</code></pre>
</div>
<div id="filter" class="section level2">
<h2>Filter</h2>
<pre class="r"><code># retreive normalized data
dataMatrix &lt;- exprs(mac_eset)

# remove the probes that are not detected in raw data
presentCount &lt;- detectionCall(lumi_46903)
selDataMatrix &lt;- dataMatrix[presentCount &gt; 0,]
probeList &lt;- rownames(selDataMatrix)</code></pre>
</div>
<div id="define-groups" class="section level2">
<h2>Define Groups</h2>
<pre class="r"><code># Get phenotype data from expressionset
pheno_data &lt;- pData(mac_eset)
str(pheno_data)</code></pre>
<pre><code>&#39;data.frame&#39;:   111 obs. of  40 variables:
 $ geo_id                     : chr  &quot;GSM1140443&quot; &quot;GSM1140444&quot; &quot;GSM1140445&quot; &quot;GSM1140446&quot; ...
 $ cell_type                  : chr  &quot;M0_GMCSF_72h &quot; &quot;M_GMCSF_IL4_72h &quot; &quot;M_GMCSF_TPP_72h &quot; &quot;M_GMCSF_IFNg_72h &quot; ...
 $ raw_id                     : chr  &quot;4487653414_E&quot; &quot;4487653415_E&quot; &quot;4487653415_G&quot; &quot;4487653415_H&quot; ...
 $ geo_accession              : chr  &quot;GSM1140443&quot; &quot;GSM1140444&quot; &quot;GSM1140445&quot; &quot;GSM1140446&quot; ...
 $ status                     : Factor w/ 1 level &quot;Public on Feb 13 2014&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ submission_date            : Factor w/ 1 level &quot;May 14 2013&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ last_update_date           : Factor w/ 3 levels &quot;Feb 13 2014&quot;,..: 1 1 1 1 2 1 1 2 1 1 ...
 $ type                       : Factor w/ 1 level &quot;RNA&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ channel_count              : Factor w/ 1 level &quot;1&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ source_name_ch1            : Factor w/ 1 level &quot;peripheral blood&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ organism_ch1               : Factor w/ 1 level &quot;Homo sapiens&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ characteristics_ch1        : Factor w/ 13 levels &quot;initial differentiation: -&quot;,..: 8 8 8 8 8 8 8 8 8 8 ...
 $ characteristics_ch1.1      : Factor w/ 41 levels &quot;activation stimuli: CD3/28&quot;,..: 4 16 37 12 30 15 14 29 12 16 ...
 $ characteristics_ch1.2      : Factor w/ 12 levels &quot;time: 0h&quot;,&quot;time: 12h&quot;,..: 11 11 11 11 11 11 11 11 11 11 ...
 $ characteristics_ch1.3      : Factor w/ 73 levels &quot;donor: BC 65&quot;,..: 48 48 48 48 39 31 40 31 40 40 ...
 $ molecule_ch1               : Factor w/ 1 level &quot;total RNA&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ extract_protocol_ch1       : Factor w/ 1 level &quot;RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit i&quot;| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...
 $ label_ch1                  : Factor w/ 1 level &quot;biotin&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ label_protocol_ch1         : Factor w/ 1 level &quot;Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ taxid_ch1                  : Factor w/ 1 level &quot;9606&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ hyb_protocol               : Factor w/ 1 level &quot;Standard Illumina hybridization protocol&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ scan_protocol              : Factor w/ 1 level &quot;Standard Illumina scanning protocol&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ description                : Factor w/ 6 levels &quot;B cell&quot;,&quot;Dendritic cell&quot;,..: 3 3 3 3 3 3 3 3 3 3 ...
 $ data_processing            : Factor w/ 1 level &quot;The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ platform_id                : Factor w/ 1 level &quot;GPL6947&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_name               : Factor w/ 1 level &quot;Joachim,,Schultze&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_email              : Factor w/ 1 level &quot;j.schultze@uni-bonn.de&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_department         : Factor w/ 1 level &quot;Genomics and Immunoregulation&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_institute          : Factor w/ 1 level &quot;LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_address            : Factor w/ 1 level &quot;Carl-Troll-Strasse 31&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_city               : Factor w/ 1 level &quot;Bonn&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_state              : Factor w/ 1 level &quot;NRW&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_zip/postal_code    : Factor w/ 1 level &quot;53115&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_country            : Factor w/ 1 level &quot;Germany&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ supplementary_file         : Factor w/ 1 level &quot;NONE&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ data_row_count             : Factor w/ 1 level &quot;47463&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ activation stimuli:ch1     : chr  &quot;con&quot; &quot;IL4&quot; &quot;TNFa+PGE2+P3C&quot; &quot;IFNg&quot; ...
 $ donor:ch1                  : chr  &quot;BC9&quot; &quot;BC9&quot; &quot;BC9&quot; &quot;BC9&quot; ...
 $ initial differentiation:ch1: chr  &quot;GM-CSF&quot; &quot;GM-CSF&quot; &quot;GM-CSF&quot; &quot;GM-CSF&quot; ...
 $ time:ch1                   : chr  &quot;72h&quot; &quot;72h&quot; &quot;72h&quot; &quot;72h&quot; ...</code></pre>
<pre class="r"><code># OK if rownames are dropped, column is duplicated as &quot;raw_id&quot;
pheno_data &lt;- pheno_data %&gt;%
  as_tibble() %&gt;%
# clean up experimental group data
  mutate(treatment = str_sub(cell_type, 9, 25),
         treatment = str_replace(treatment, &quot;_72h&quot;, &quot;&quot;),
         treatment = ifelse(`activation stimuli:ch1` == &quot;con&quot;, &quot;control&quot;, treatment),
         treatment = trimws(treatment)) %&gt;%
  dplyr::select(raw_id, geo_id, treatment, everything()) %&gt;%
  print()</code></pre>
<pre><code># A tibble: 111 x 41
   raw_id geo_id treatment cell_type geo_accession status submission_date
   &lt;chr&gt;  &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;         &lt;fct&gt;  &lt;fct&gt;          
 1 44876… GSM11… control   &quot;M0_GMCS… GSM1140443    Publi… May 14 2013    
 2 44876… GSM11… IL4       &quot;M_GMCSF… GSM1140444    Publi… May 14 2013    
 3 44876… GSM11… TPP       &quot;M_GMCSF… GSM1140445    Publi… May 14 2013    
 4 44876… GSM11… IFNg      &quot;M_GMCSF… GSM1140446    Publi… May 14 2013    
 5 46761… GSM11… sLPS_IC   &quot;M_GMCSF… GSM1140484    Publi… May 14 2013    
 6 46761… GSM11… IL13      &quot;M_GMCSF… GSM1140485    Publi… May 14 2013    
 7 46761… GSM11… IL10      &quot;M_GMCSF… GSM1140486    Publi… May 14 2013    
 8 46761… GSM11… sLPS      &quot;M_GMCSF… GSM1140489    Publi… May 14 2013    
 9 46761… GSM11… IFNg      &quot;M_GMCSF… GSM1140490    Publi… May 14 2013    
10 46761… GSM11… IL4       &quot;M_GMCSF… GSM1140492    Publi… May 14 2013    
# … with 101 more rows, and 34 more variables: last_update_date &lt;fct&gt;,
#   type &lt;fct&gt;, channel_count &lt;fct&gt;, source_name_ch1 &lt;fct&gt;,
#   organism_ch1 &lt;fct&gt;, characteristics_ch1 &lt;fct&gt;,
#   characteristics_ch1.1 &lt;fct&gt;, characteristics_ch1.2 &lt;fct&gt;,
#   characteristics_ch1.3 &lt;fct&gt;, molecule_ch1 &lt;fct&gt;,
#   extract_protocol_ch1 &lt;fct&gt;, label_ch1 &lt;fct&gt;, label_protocol_ch1 &lt;fct&gt;,
#   taxid_ch1 &lt;fct&gt;, hyb_protocol &lt;fct&gt;, scan_protocol &lt;fct&gt;,
#   description &lt;fct&gt;, data_processing &lt;fct&gt;, platform_id &lt;fct&gt;,
#   contact_name &lt;fct&gt;, contact_email &lt;fct&gt;, contact_department &lt;fct&gt;,
#   contact_institute &lt;fct&gt;, contact_address &lt;fct&gt;, contact_city &lt;fct&gt;,
#   contact_state &lt;fct&gt;, `contact_zip/postal_code` &lt;fct&gt;,
#   contact_country &lt;fct&gt;, supplementary_file &lt;fct&gt;, data_row_count &lt;fct&gt;,
#   `activation stimuli:ch1` &lt;chr&gt;, `donor:ch1` &lt;chr&gt;, `initial
#   differentiation:ch1` &lt;chr&gt;, `time:ch1` &lt;chr&gt;</code></pre>
<pre class="r"><code># assign treatments into clusters from Xue et al PMID: 24530056
pheno_data &lt;- pheno_data %&gt;%
  mutate(treatment = as.factor(treatment),
         clusternum = fct_recode(treatment,
                                 &quot;0&quot; = &quot;control&quot;,
                                 &quot;1&quot; = &quot;IFNb&quot;,
                                 &quot;1&quot; = &quot;IL10&quot;,
                                 &quot;1&quot; = &quot;GC&quot;,
                                 &quot;2&quot; = &quot;IL4&quot;,
                                 &quot;2&quot; = &quot;L4&quot;,
                                 &quot;2&quot; = &quot;IL13&quot;,
                                 &quot;2&quot; = &quot;IL4_upLPS&quot;,
                                 &quot;2&quot; = &quot;L4_upLPS&quot;,
                                 &quot;3&quot; = &quot;upLPS_IC&quot;,
                                 &quot;3&quot; = &quot;upLPS&quot;,
                                 &quot;3&quot; = &quot;pLPS&quot;,
                                 &quot;4&quot; = &quot;P3C_PGE2&quot;,
                                 &quot;4&quot; = &quot;P3C&quot;,
                                 &quot;4&quot; = &quot;PGE2&quot;,
                                 &quot;6&quot; = &quot;IFNg&quot;,
                                 &quot;6&quot; = &quot;TNFa&quot;,
                                 &quot;6&quot; = &quot;IFNg_TNFa&quot;,
                                 &quot;7&quot; = &quot;sLPS&quot;,
                                 &quot;7&quot; = &quot;sLPS_IFNg&quot;,
                                 &quot;7&quot; = &quot;sLPS_IC&quot;,
                                 &quot;8&quot; = &quot;TNFa_PGE2&quot;,
                                 &quot;8&quot; = &quot;TNFa_P3C&quot;,
                                 &quot;9&quot; = &quot;TPP&quot;,
                                 &quot;9&quot; = &quot;TPP_IFNb&quot;,
                                 &quot;9&quot; = &quot;TPP_IFNb_IFNg&quot;),
         treatment = fct_recode(treatment, 
                                &quot;IL4_upLPS&quot; = &quot;L4_upLPS&quot;,
                                &quot;upLPS&quot; = &quot;pLPS&quot;,
                                &quot;IL4&quot; = &quot;L4&quot;)) %&gt;%
  dplyr::select(raw_id, geo_id, treatment, clusternum, everything()) %&gt;%
  print()</code></pre>
<pre><code># A tibble: 111 x 42
   raw_id geo_id treatment clusternum cell_type geo_accession status
   &lt;chr&gt;  &lt;chr&gt;  &lt;fct&gt;     &lt;fct&gt;      &lt;chr&gt;     &lt;chr&gt;         &lt;fct&gt; 
 1 44876… GSM11… control   0          &quot;M0_GMCS… GSM1140443    Publi…
 2 44876… GSM11… IL4       2          &quot;M_GMCSF… GSM1140444    Publi…
 3 44876… GSM11… TPP       9          &quot;M_GMCSF… GSM1140445    Publi…
 4 44876… GSM11… IFNg      6          &quot;M_GMCSF… GSM1140446    Publi…
 5 46761… GSM11… sLPS_IC   7          &quot;M_GMCSF… GSM1140484    Publi…
 6 46761… GSM11… IL13      2          &quot;M_GMCSF… GSM1140485    Publi…
 7 46761… GSM11… IL10      1          &quot;M_GMCSF… GSM1140486    Publi…
 8 46761… GSM11… sLPS      7          &quot;M_GMCSF… GSM1140489    Publi…
 9 46761… GSM11… IFNg      6          &quot;M_GMCSF… GSM1140490    Publi…
10 46761… GSM11… IL4       2          &quot;M_GMCSF… GSM1140492    Publi…
# … with 101 more rows, and 35 more variables: submission_date &lt;fct&gt;,
#   last_update_date &lt;fct&gt;, type &lt;fct&gt;, channel_count &lt;fct&gt;,
#   source_name_ch1 &lt;fct&gt;, organism_ch1 &lt;fct&gt;, characteristics_ch1 &lt;fct&gt;,
#   characteristics_ch1.1 &lt;fct&gt;, characteristics_ch1.2 &lt;fct&gt;,
#   characteristics_ch1.3 &lt;fct&gt;, molecule_ch1 &lt;fct&gt;,
#   extract_protocol_ch1 &lt;fct&gt;, label_ch1 &lt;fct&gt;, label_protocol_ch1 &lt;fct&gt;,
#   taxid_ch1 &lt;fct&gt;, hyb_protocol &lt;fct&gt;, scan_protocol &lt;fct&gt;,
#   description &lt;fct&gt;, data_processing &lt;fct&gt;, platform_id &lt;fct&gt;,
#   contact_name &lt;fct&gt;, contact_email &lt;fct&gt;, contact_department &lt;fct&gt;,
#   contact_institute &lt;fct&gt;, contact_address &lt;fct&gt;, contact_city &lt;fct&gt;,
#   contact_state &lt;fct&gt;, `contact_zip/postal_code` &lt;fct&gt;,
#   contact_country &lt;fct&gt;, supplementary_file &lt;fct&gt;, data_row_count &lt;fct&gt;,
#   `activation stimuli:ch1` &lt;chr&gt;, `donor:ch1` &lt;chr&gt;, `initial
#   differentiation:ch1` &lt;chr&gt;, `time:ch1` &lt;chr&gt;</code></pre>
<pre class="r"><code>summary(pheno_data$treatment)</code></pre>
<pre><code>      control            GC          IFNb          IFNg     IFNg_TNFa 
            7             3             3             7             3 
         IL10          IL13           IL4     IL4_upLPS           P3C 
            3             3            13             6             3 
     P3C_PGE2          PGE2         upLPS          sLPS       sLPS_IC 
            3             3             9             8             3 
    sLPS_IFNg          TNFa      TNFa_P3C     TNFa_PGE2           TPP 
            3             6             3             3            10 
     TPP_IFNb TPP_IFNb_IFNg      upLPS_IC 
            3             3             3 </code></pre>
<pre class="r"><code>summary(pheno_data$clusternum)</code></pre>
<pre><code> 0  1  6  2  4  3  7  8  9 
 7  9 16 22  9 12 14  6 16 </code></pre>
</div>
<div id="limma" class="section level2">
<h2>Limma</h2>
<p>Design matrix and contrast matrix for individual treatments</p>
<pre class="r"><code># set up design matrix
rnames &lt;- pheno_data$raw_id
groups &lt;- as.factor(pheno_data$treatment)
design &lt;- model.matrix(~ 0 + groups)
colnames(design) &lt;- gsub(&quot;groups&quot;, &quot;&quot;, colnames(design))
rownames(design) &lt;- rnames
design[1:5, 1:5]</code></pre>
<pre><code>             control GC IFNb IFNg IFNg_TNFa
4487653414_E       1  0    0    0         0
4487653415_E       0  0    0    0         0
4487653415_G       0  0    0    0         0
4487653415_H       0  0    0    1         0
4676163025_B       0  0    0    0         0</code></pre>
<pre class="r"><code># define contrasts
contr.matrix &lt;- makeContrasts(GC - control,
                              IFNb - control,
                              IFNg - control,
                              IFNg_TNFa - control,
                              IL10 - control,
                              IL13 - control,
                              IL4 - control,
                              IL4_upLPS - control,
                              P3C - control,
                              P3C_PGE2 - control,
                              PGE2 - control,
                              upLPS - control,
                              sLPS - control,
                              sLPS_IC - control,
                              sLPS_IFNg - control,
                              TNFa - control,
                              TNFa_P3C - control,
                              TNFa_PGE2 - control,
                              TPP - control,
                              TPP_IFNb - control,
                              TPP_IFNb_IFNg - control,
                              upLPS_IC - control,
                              levels = design)
contr.matrix[1:5, 1:5]</code></pre>
<pre><code>           Contrasts
Levels      GC - control IFNb - control IFNg - control IFNg_TNFa - control
  control             -1             -1             -1                  -1
  GC                   1              0              0                   0
  IFNb                 0              1              0                   0
  IFNg                 0              0              1                   0
  IFNg_TNFa            0              0              0                   1
           Contrasts
Levels      IL10 - control
  control               -1
  GC                     0
  IFNb                   0
  IFNg                   0
  IFNg_TNFa              0</code></pre>
<p>Fit linear models with limma</p>
<pre class="r"><code># fit linear models
fit &lt;- lmFit(selDataMatrix, design)
fit2 &lt;- contrasts.fit(fit, contr.matrix)
fit2 &lt;- eBayes(fit2)

# add gene names
geneSymbol &lt;- getSYMBOL(probeList, &#39;lumiHumanAll.db&#39;)
geneName &lt;- sapply(lookUp(probeList, &#39;lumiHumanAll.db&#39;, &#39;GENENAME&#39;),
                   function(x) x[1])
fit2$genes &lt;- data.frame(ID = probeList, 
                         geneSymbol = geneSymbol, 
                         geneName = geneName, 
                         stringsAsFactors = FALSE)

# store matrix of significant results
results &lt;- decideTests(fit2)

# collect results
table_GC        &lt;- topTable(fit2, coef = 1, 
                            adjust.method = &quot;BH&quot;, 
                            number = 500)
table_IFNb      &lt;- topTable(fit2, 
                            coef = 2, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_IFNg      &lt;- topTable(fit2, 
                            coef = 3, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_IFNg_TNFa &lt;- topTable(fit2, 
                            coef = 4, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_IL10      &lt;- topTable(fit2, 
                            coef = 5, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_IL13      &lt;- topTable(fit2, 
                            coef = 6, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_IL4       &lt;- topTable(fit2, 
                            coef = 7, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_IL4_upLPS &lt;- topTable(fit2, 
                            coef = 8, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_P3C       &lt;- topTable(fit2, 
                            coef = 9, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_P3C_PGE2  &lt;- topTable(fit2, 
                            coef = 10, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_PGE2      &lt;- topTable(fit2, 
                            coef = 11, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_upLPS     &lt;- topTable(fit2, 
                            coef = 12, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_sLPS      &lt;- topTable(fit2, 
                            coef = 13, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_sLPS_IC   &lt;- topTable(fit2, 
                            coef = 14, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_sLPS_IFNg &lt;- topTable(fit2, 
                            coef = 15, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_TNFa      &lt;- topTable(fit2, 
                            coef = 16, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_TNFa_P3C  &lt;- topTable(fit2, 
                            coef = 17, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_TNFa_PGE2 &lt;- topTable(fit2, 
                            coef = 18, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_TPP       &lt;- topTable(fit2, 
                            coef = 19, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_TPP_IFNb  &lt;- topTable(fit2, 
                            coef = 20, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)
table_TPP_IFNb_IFNg &lt;- topTable(fit2, 
                                coef = 21, 
                                adjust.method = &quot;BH&quot;,
                                number = 500)
table_upLPS_IC  &lt;- topTable(fit2, 
                            coef = 22, 
                            adjust.method = &quot;BH&quot;,
                            number = 500)</code></pre>
<p>Collecting results</p>
<pre class="r"><code># take only significantly differentially expressed genes
# using FDR-corrected P-values
# pull out t value and gene symbol
subtable_GC &lt;- table_GC %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, GC = t)
subtable_IFNb &lt;- table_IFNb %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IFNb = t)
subtable_IFNg &lt;- table_IFNg %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IFNg = t)
subtable_IFNg_TNFa &lt;- table_IFNg_TNFa %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IFNg_TNFa = t)
subtable_IL10 &lt;- table_IL10 %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IL10 = t)
subtable_IL13 &lt;- table_IL13 %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IL13 = t)
subtable_IL4 &lt;- table_IL4 %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IL4 = t)
subtable_IL4_upLPS &lt;- table_IL4_upLPS %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, IL4_upLPS = t)
subtable_P3C &lt;- table_P3C %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, P3C = t)
subtable_P3C_PGE2 &lt;- table_P3C_PGE2 %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, P3C_PGE2 = t)
subtable_PGE2 &lt;- table_PGE2 %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, PGE2 = t)
subtable_upLPS &lt;- table_upLPS %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, upLPS = t)
subtable_sLPS &lt;- table_sLPS %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, sLPS = t)
subtable_sLPS_IC &lt;- table_sLPS_IC %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, sLPS_IC = t)
subtable_sLPS_IFNg &lt;- table_sLPS_IFNg %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, sLPS_IFNg = t)
subtable_TNFa &lt;- table_TNFa %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, TNFa = t)
subtable_TNFa_P3C &lt;- table_TNFa_P3C %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, TNFa_P3C = t)
subtable_TNFa_PGE2 &lt;- table_TNFa_PGE2 %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, TNFa_PGE2 = t)
subtable_TPP &lt;- table_TPP %&gt;% 
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, TPP = t)
subtable_TPP_IFNb &lt;- table_TPP_IFNb %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, TPP_IFNb = t)
subtable_TPP_IFNb_IFNg &lt;- table_TPP_IFNb_IFNg %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, TPP_IFNb_IFNg = t)
subtable_upLPS_IC &lt;- table_upLPS_IC %&gt;%
  filter(adj.P.Val &lt; 0.05) %&gt;%
  dplyr::select(ID, geneSymbol, upLPS = t)

# get  of all the probe symbols in all the lists
GC        &lt;- pull(subtable_GC, ID) 
IFNb      &lt;- pull(subtable_IFNb, ID)
IFNg      &lt;- pull(subtable_IFNg, ID)
IFNg_TNFa &lt;- pull(subtable_IFNg_TNFa, ID)
IL10      &lt;- pull(subtable_IL10, ID)
IL13      &lt;- pull(subtable_IL13, ID)
IL4       &lt;- pull(subtable_IL4, ID)
IL4_upLPS &lt;- pull(subtable_IL4_upLPS, ID)
P3C       &lt;- pull(subtable_P3C, ID)
P3C_PGE2  &lt;- pull(subtable_P3C_PGE2, ID)
PGE2      &lt;- pull(subtable_PGE2, ID)
upLPS     &lt;- pull(subtable_upLPS, ID)
sLPS      &lt;- pull(subtable_sLPS, ID)
sLPS_IC   &lt;- pull(subtable_sLPS_IC, ID)
sLPS_IFNg &lt;- pull(subtable_sLPS_IFNg, ID)
TNFa      &lt;- pull(subtable_TNFa, ID)
TNFa_P3C  &lt;- pull(subtable_TNFa_P3C, ID)
TNFa_PGE2 &lt;- pull(subtable_TNFa_PGE2, ID)
TPP       &lt;- pull(subtable_TPP, ID)
TPP_IFNb  &lt;- pull(subtable_TPP_IFNb, ID)
TPP_IFNbg &lt;- pull(subtable_TPP_IFNb_IFNg, ID)
upLPS_IC  &lt;- pull(subtable_upLPS_IC, ID)

unique_ids &lt;- unique(c(GC, IFNb, IFNg, IFNg_TNFa, IL10, IL13, IL4, IL4_upLPS, P3C, P3C_PGE2, PGE2, upLPS, sLPS, sLPS_IC, sLPS_IFNg, TNFa, TNFa_P3C, TNFa_PGE2, TPP, TPP_IFNb, TPP_IFNbg, upLPS_IC))

# get gene symbols for each of the unique probe IDs
unique_ids_geneSymbol &lt;- getSYMBOL(unique_ids, &#39;lumiHumanAll.db&#39;) %&gt;%
  enframe() %&gt;%
  dplyr::rename(ID = name, geneSymbol = value)

# make table of differential gene expression t values
dge_frame &lt;- enframe(unique_ids) %&gt;%
  dplyr::rename(ID = value) %&gt;%
  left_join(unique_ids_geneSymbol, by = &quot;ID&quot;) %&gt;%
  left_join(subtable_GC, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IFNb, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IFNg, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IFNg_TNFa, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IL10, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IL13, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IL4, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_IL4_upLPS, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_P3C, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_P3C_PGE2, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_PGE2, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_upLPS, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_sLPS, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_sLPS_IC, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_sLPS_IFNg, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_TNFa, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_TNFa_P3C, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_TNFa_PGE2, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_TPP, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_TPP_IFNb, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_TPP_IFNb_IFNg, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;)) %&gt;%
  left_join(subtable_upLPS_IC, by = c(&quot;ID&quot;, &quot;geneSymbol&quot;))</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggpubr_0.2                magrittr_1.5             
 [3] ggsci_2.9                 FactoMineR_1.41          
 [5] factoextra_1.0.5          annotate_1.60.1          
 [7] XML_3.98-1.19             limma_3.38.3             
 [9] lumiHumanAll.db_1.22.0    org.Hs.eg.db_3.7.0       
[11] lumiHumanIDMapping_1.10.1 AnnotationDbi_1.44.0     
[13] IRanges_2.16.0            S4Vectors_0.20.1         
[15] lumi_2.34.0               GEOquery_2.50.5          
[17] Biobase_2.42.0            BiocGenerics_0.28.0      
[19] forcats_0.4.0             stringr_1.4.0            
[21] dplyr_0.8.0.1             purrr_0.3.2              
[23] readr_1.3.1               tidyr_0.8.3              
[25] tibble_2.1.1              ggplot2_3.1.1            
[27] tidyverse_1.2.1          

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                backports_1.1.4            
  [3] workflowr_1.2.0             plyr_1.8.4                 
  [5] lazyeval_0.2.2              splines_3.5.3              
  [7] BiocParallel_1.16.6         GenomeInfoDb_1.18.2        
  [9] digest_0.6.18               foreach_1.4.4              
 [11] htmltools_0.3.6             fansi_0.4.0                
 [13] memoise_1.1.0               cluster_2.0.8              
 [15] Biostrings_2.50.2           modelr_0.1.4               
 [17] matrixStats_0.54.0          askpass_1.1                
 [19] siggenes_1.56.0             prettyunits_1.0.2          
 [21] colorspace_1.4-1            ggrepel_0.8.0              
 [23] blob_1.1.1                  rvest_0.3.3                
 [25] haven_2.1.0                 xfun_0.6                   
 [27] crayon_1.3.4                RCurl_1.95-4.12            
 [29] jsonlite_1.6                genefilter_1.64.0          
 [31] survival_2.44-1.1           iterators_1.0.10           
 [33] glue_1.3.1                  registry_0.5-1             
 [35] gtable_0.3.0                zlibbioc_1.28.0            
 [37] XVector_0.22.0              DelayedArray_0.8.0         
 [39] Rhdf5lib_1.4.3              HDF5Array_1.10.1           
 [41] scales_1.0.0                DBI_1.0.0                  
 [43] rngtools_1.3.1              bibtex_0.4.2               
 [45] Rcpp_1.0.1                  xtable_1.8-3               
 [47] progress_1.2.0              bumphunter_1.24.5          
 [49] flashClust_1.01-2           bit_1.1-14                 
 [51] mclust_5.4.3                preprocessCore_1.44.0      
 [53] httr_1.4.0                  RColorBrewer_1.1-2         
 [55] pkgconfig_2.0.2             reshape_0.8.8              
 [57] utf8_1.1.4                  locfit_1.5-9.1             
 [59] tidyselect_0.2.5            rlang_0.3.4                
 [61] munsell_0.5.0               cellranger_1.1.0           
 [63] tools_3.5.3                 cli_1.1.0                  
 [65] generics_0.0.2              RSQLite_2.1.1              
 [67] broom_0.5.2                 evaluate_0.13              
 [69] yaml_2.2.0                  knitr_1.22                 
 [71] bit64_0.9-7                 fs_1.2.7                   
 [73] beanplot_1.2                methylumi_2.28.0           
 [75] nlme_3.1-139                doRNG_1.7.1                
 [77] whisker_0.3-2               nor1mix_1.2-3              
 [79] leaps_3.0                   xml2_1.2.0                 
 [81] biomaRt_2.38.0              compiler_3.5.3             
 [83] rstudioapi_0.10             affyio_1.52.0              
 [85] stringi_1.4.3               GenomicFeatures_1.34.8     
 [87] minfi_1.28.4                lattice_0.20-38            
 [89] Matrix_1.2-17               multtest_2.38.0            
 [91] pillar_1.3.1                BiocManager_1.30.4         
 [93] data.table_1.12.2           bitops_1.0-6               
 [95] rtracklayer_1.42.2          GenomicRanges_1.34.0       
 [97] R6_2.4.0                    affy_1.60.0                
 [99] KernSmooth_2.23-15          nleqslv_3.3.2              
[101] codetools_0.2-16            MASS_7.3-51.4              
[103] assertthat_0.2.1            rhdf5_2.26.2               
[105] SummarizedExperiment_1.12.0 openssl_1.3                
[107] pkgmaker_0.27               rprojroot_1.3-2            
[109] withr_2.1.2                 GenomicAlignments_1.18.1   
[111] Rsamtools_1.34.1            GenomeInfoDbData_1.2.0     
[113] mgcv_1.8-28                 hms_0.4.2                  
[115] quadprog_1.5-5              grid_3.5.3                 
[117] base64_2.0                  rmarkdown_1.12             
[119] DelayedMatrixStats_1.4.0    illuminaio_0.24.0          
[121] git2r_0.25.2                scatterplot3d_0.3-41       
[123] lubridate_1.7.4            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
