<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-04-09" />

<title>Differential gene expression: _in vitro_ macrophage model</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Macrophage signatures in TCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_macs">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Differential gene expression: <em>in vitro</em> macrophage model</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-04-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#report"> Report <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-04-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_macs/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.2.0). The <em>Report</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="report" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190409code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190409)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190409code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190409)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstree413333cf7bfae072abbcf67d2f2d93682fd6ecf5targetblank413333ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_macs/tree/413333cf7bfae072abbcf67d2f2d93682fd6ecf5" target="_blank">413333c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstree413333cf7bfae072abbcf67d2f2d93682fd6ecf5targetblank413333ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/GSE46903_Affymetrix_Reanalyzed_data.txt.gz
    Ignored:    data/GSE46903_Affymetrix_metadata.txt.gz
    Ignored:    data/GSE46903_RAW.tar
    Ignored:    data/GSE46903_non-normalized.txt.gz
    Ignored:    output/processed_lumi_46903.txt

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/413333cf7bfae072abbcf67d2f2d93682fd6ecf5/analysis/invitro_mac_dge.Rmd" target="_blank">413333c</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
updated download and data cleaning from GEO
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/65a4fa44eb069b1d02b19cafd57e7ae382cca4c0/docs/invitro_mac_dge.html" target="_blank">65a4fa4</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
updated download of raw data and normalization from GEO
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/6cd5270cacf3ac8d11c7faea7ca4cb0de42b89c3/docs/invitro_mac_dge.html" target="_blank">6cd5270</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/1ecaa5cf559a8344497d369dc8687431dcb72dc0/analysis/invitro_mac_dge.Rmd" target="_blank">1ecaa5c</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
wflow_publish(“analysis/invitro_mac_dge.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/c43686198fa5b5f0718f028386ecd48036e5e4ce/docs/invitro_mac_dge.html" target="_blank">c436861</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/fcefd997d3b694d52c85f702a49784d945619e96/analysis/invitro_mac_dge.Rmd" target="_blank">fcefd99</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-10
</td>
<td>
successful raw data import and normalization
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/1c4ad0246f938ca2348e7ab0df1a0b4e8a5631f0/docs/invitro_mac_dge.html" target="_blank">1c4ad02</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/2555a2a7e7eea7fccfeabb6e392926723e7cff25/analysis/invitro_mac_dge.Rmd" target="_blank">2555a2a</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
cleaned up quotation parts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/8ff4e151b900839aee8321314a9c5167224bb3f3/docs/invitro_mac_dge.html" target="_blank">8ff4e15</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8c06425b5aba4ece8ed25814b330d833d3e487dd/analysis/invitro_mac_dge.Rmd" target="_blank">8c06425</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
added protocols
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/363b5df5770f40d8b98527c9ced243c7aa7a06d9/docs/invitro_mac_dge.html" target="_blank">363b5df</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/17f3020720fb6faa9b18e0fae66c7741d157c6d3/analysis/invitro_mac_dge.Rmd" target="_blank">17f3020</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
beginning the text on the homepage
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/6dc88381bf3cf9dd2d2ccb88e8e9467bbd9bd590/docs/invitro_mac_dge.html" target="_blank">6dc8838</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/13ee5e07deb1c432977fc7f2d21d334232961afd/analysis/invitro_mac_dge.Rmd" target="_blank">13ee5e0</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
working data download, added citations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/eacbbe0ca3e23418644b1c547f3846e5510fe778/docs/invitro_mac_dge.html" target="_blank">eacbbe0</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/b27b467677db6859f76f017c1be5e532adfebfe5/analysis/invitro_mac_dge.Rmd" target="_blank">b27b467</a>
</td>
<td>
mleukam
</td>
<td>
2019-04-09
</td>
<td>
citations added and packages loaded
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<div id="purpose-of-this-analysis" class="section level3">
<h3>Purpose of this analysis</h3>
<p>Find ranked differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes. These gene sets will be used for our downstream analyses.</p>
<p>The source for this method is:</p>
<blockquote>
<p>Gubin, Matthew M., Ekaterina Esaulova, Jeffrey P. Ward, Olga N. Malkova, Daniele Runci, Pamela Wong, Takuro Noguchi, et al. 2018. “High-Dimensional Analysis Delineates Myeloid and Lymphoid Compartment Remodeling during Successful Immune-Checkpoint Cancer Therapy.” Cell 175 (4): 1014–30.e19.</p>
</blockquote>
<p>This is a summary of the method from Gubin et al.:</p>
<blockquote>
<p>Expression data was retrieved from GEO with GSE46903 ID. For analysis, we selected only monocyte and macrophage samples. According to the dataset description, peripheral blood monocytes were stimulated with GM-CSF or M-CSF, and subsequently stimulated with different compounds. We excluded all M-CSF-derived macrophages and non-72 hour time points. M0-GM-CSF 72 hour samples were used as baseline. We used limma package to perform differential expression in order to understand gene changes upon treatments compared to baseline. For each comparison, gene lists were obtained and ranked by t-statistics. To compare human macrophages under different stimuli with mouse monocytes/macrophages from our scRNAseq data, we compared genes, upregulated or downregulated inside each scRNAseq cluster with genes, upregulated or downregulated in each human monocyte/macrophage treatment condition. The goal was to get a list of signed scores that will represent the level of concordance between treatments and each cluster.</p>
</blockquote>
<blockquote>
<p>For each cluster, we created two gene sets: 100 upregulated and downregulated genes. GSEA (with fgsea package) was performed using this gene set against ranked gene lists from treatment comparisons. To obtain a score, we subtracted the normalized enrichment score (NES) of downregulated genes from NES of upregulated genes. To further investigate the correlation (or lack thereof) between clusters and treatments, we visualized genes associated with each cluster in human macrophages. We reduced the dimensions of macrophage datasets with Rtsne package and plotted each sample in two dimensions using the ggplot2 package. Samples were then highlighted using Z-scored expression profiles of cluster-associated genes.</p>
</blockquote>
</div>
<div id="data-source" class="section level3">
<h3>Data Source</h3>
<p>The gene chip data used to derive differential expression for various macrophage subtypes was first published here:</p>
<blockquote>
<p>Xue J, Schmidt SV, Sander J, Draffehn A et al. Transcriptome-based network analysis reveals a spectrum model of human macrophage activation. Immunity 2014 Feb 20;40(2):274-88. PMID: 24530056</p>
</blockquote>
<p>The original experiment had two conditions.</p>
<ol style="list-style-type: decimal">
<li><p>Cell culture with monocytes exposed to growth factor and a number of different cytokines to induce different activation states. Microarray expression data from cell culture were generated with Illumina HumanHT-12 v3.0 beadchips.</p></li>
<li><p>Gene sets derived in cell culture were used to categorize expression profiles derived from alveolar macrophages from patients with COPD and asthma. Affymetrix chips were used to measure expression in human samples.</p></li>
</ol>
<p>We are interested in the first condition (macrophages in culture). Associated metadata is posted at the GEO:</p>
<ul>
<li><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903">The experiment overview</a></p></li>
<li><p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1140443">The record for the control group</a></p>
<ul>
<li><p>M0-GM-CSF treated monocytes at 72hours</p></li>
<li><p>Additional details about how the data were generated</p></li>
</ul></li>
<li><p>Expression matrices in which “The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite”</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?view=data&amp;acc=GSM1140443&amp;id=38019&amp;db=GeoDb_blob97">Example expression matrix</a> from the control group</li>
</ul></li>
</ul>
<p><em>We will obtain the raw data from the Illumina expression beadchips and normalize ourselves.</em></p>
</div>
<div id="set-up-workspace" class="section level3">
<h3>Set up workspace</h3>
<p>Clear workspace prior to analysis and load necessary packages</p>
<pre class="r"><code>rm(list=ls())</code></pre>
<pre class="r"><code>library(&quot;tidyverse&quot;)
library(&quot;GEOquery&quot;)
library(&quot;lumi&quot;)</code></pre>
</div>
</div>
<div id="data-import" class="section level2">
<h2>Data import</h2>
<p>GEO ID is: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903">GSE46903</a></p>
<pre class="r"><code># download normalized data with full metadata from GEO as a list of ExpressionSets
gse46903 &lt;- getGEO(&quot;GSE46903&quot;)</code></pre>
<pre><code>Found 1 file(s)</code></pre>
<pre><code>GSE46903_series_matrix.txt.gz</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  ID_REF = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<pre><code>File stored at: </code></pre>
<pre><code>/var/folders/dc/481f588d6ksdf0wn34p80stw0000gn/T//Rtmp05Zbb6/GPL6947.soft</code></pre>
<p>Accessing raw data from the GEO: <a href="http://genomicsclass.github.io/book/pages/GEOquery.html" class="uri">http://genomicsclass.github.io/book/pages/GEOquery.html</a></p>
<p>Note about raw vs normalized files from GEO: <a href="https://www.biostars.org/p/259637/" class="uri">https://www.biostars.org/p/259637/</a></p>
<pre class="r"><code># download raw data
filePaths = getGEOSuppFiles(&quot;GSE46903&quot;, 
                            makeDirectory = TRUE)</code></pre>
<pre class="bash"><code># in terminal
pwd
ls
cd GSE46903
ls
mv GSE* ~/tcga_macs/data
cd ..
rmdir GSE46903
ls</code></pre>
<pre><code>/Users/mleukam/tcga_macs
GSE46903
README.md
_workflowr.yml
analysis
code
data
docs
output
tcga_macs.Rproj
GSE46903_Affymetrix_Reanalyzed_data.txt.gz
GSE46903_Affymetrix_metadata.txt.gz
GSE46903_RAW.tar
GSE46903_non-normalized.txt.gz
README.md
_workflowr.yml
analysis
code
data
docs
output
tcga_macs.Rproj</code></pre>
<pre class="r"><code>lumi_input &lt;- lumiR(&quot;~/tcga_macs/data/GSE46903_non-normalized.txt.gz&quot;) </code></pre>
<pre><code>Please provide Illumina ID Mapping library!</code></pre>
<pre class="r"><code>lumi_46903 &lt;- lumiExpresso(lumi_input, 
                         normalize.param = list(method = &#39;rsn&#39;))</code></pre>
<pre><code>Due to the input is not a LumiBatch object, no background adjustment will be performed.
Background Correction: none 
Due to the lack of &#39;se.exprs&#39; information, &#39;log2&#39; transformation will be used.
Variance Stabilizing Transform method: log2 
Normalization method: rsn 


Background correction ...
done.

Variance stabilizing ...
Perform log2 transformation ...
done.

Normalizing ...
Perform rsn normalization ...
2019-04-10 12:20:37 , processing array  1 
2019-04-10 12:20:38 , processing array  2 
2019-04-10 12:20:38 , processing array  3 
2019-04-10 12:20:38 , processing array  4 
2019-04-10 12:20:38 , processing array  5 
2019-04-10 12:20:38 , processing array  6 
2019-04-10 12:20:38 , processing array  7 
2019-04-10 12:20:38 , processing array  8 
2019-04-10 12:20:38 , processing array  9 
2019-04-10 12:20:38 , processing array  10 
2019-04-10 12:20:38 , processing array  11 
2019-04-10 12:20:38 , processing array  12 
2019-04-10 12:20:38 , processing array  13 
2019-04-10 12:20:38 , processing array  14 
2019-04-10 12:20:39 , processing array  15 
2019-04-10 12:20:39 , processing array  16 
2019-04-10 12:20:39 , processing array  17 
2019-04-10 12:20:39 , processing array  18 
2019-04-10 12:20:39 , processing array  19 
2019-04-10 12:20:39 , processing array  20 
2019-04-10 12:20:39 , processing array  21 
2019-04-10 12:20:39 , processing array  22 
2019-04-10 12:20:39 , processing array  23 
2019-04-10 12:20:39 , processing array  24 
2019-04-10 12:20:39 , processing array  25 
2019-04-10 12:20:39 , processing array  26 
2019-04-10 12:20:40 , processing array  27 
2019-04-10 12:20:40 , processing array  28 
2019-04-10 12:20:40 , processing array  29 
2019-04-10 12:20:40 , processing array  30 
2019-04-10 12:20:40 , processing array  31 
2019-04-10 12:20:40 , processing array  32 
2019-04-10 12:20:40 , processing array  33 
2019-04-10 12:20:40 , processing array  34 
2019-04-10 12:20:40 , processing array  35 
2019-04-10 12:20:40 , processing array  36 
2019-04-10 12:20:40 , processing array  37 
2019-04-10 12:20:40 , processing array  38 
2019-04-10 12:20:40 , processing array  39 
2019-04-10 12:20:41 , processing array  40 
2019-04-10 12:20:41 , processing array  41 
2019-04-10 12:20:41 , processing array  42 
2019-04-10 12:20:41 , processing array  43 
2019-04-10 12:20:41 , processing array  44 
2019-04-10 12:20:41 , processing array  45 
2019-04-10 12:20:41 , processing array  46 
2019-04-10 12:20:41 , processing array  47 
2019-04-10 12:20:41 , processing array  48 
2019-04-10 12:20:41 , processing array  49 
2019-04-10 12:20:41 , processing array  50 
2019-04-10 12:20:41 , processing array  51 
2019-04-10 12:20:41 , processing array  52 
2019-04-10 12:20:41 , processing array  53 
2019-04-10 12:20:42 , processing array  54 
2019-04-10 12:20:42 , processing array  55 
2019-04-10 12:20:42 , processing array  56 
2019-04-10 12:20:42 , processing array  57 
2019-04-10 12:20:42 , processing array  58 
2019-04-10 12:20:42 , processing array  59 
2019-04-10 12:20:42 , processing array  60 
2019-04-10 12:20:42 , processing array  61 
2019-04-10 12:20:42 , processing array  62 
2019-04-10 12:20:42 , processing array  63 
2019-04-10 12:20:42 , processing array  64 
2019-04-10 12:20:42 , processing array  65 
2019-04-10 12:20:42 , processing array  66 
2019-04-10 12:20:43 , processing array  67 
2019-04-10 12:20:43 , processing array  68 
2019-04-10 12:20:43 , processing array  69 
2019-04-10 12:20:43 , processing array  70 
2019-04-10 12:20:43 , processing array  71 
2019-04-10 12:20:43 , processing array  72 
2019-04-10 12:20:43 , processing array  73 
2019-04-10 12:20:43 , processing array  74 
2019-04-10 12:20:43 , processing array  75 
2019-04-10 12:20:43 , processing array  76 
2019-04-10 12:20:43 , processing array  77 
2019-04-10 12:20:43 , processing array  78 
2019-04-10 12:20:43 , processing array  79 
2019-04-10 12:20:44 , processing array  80 
2019-04-10 12:20:44 , processing array  81 
2019-04-10 12:20:44 , processing array  82 
2019-04-10 12:20:44 , processing array  83 
2019-04-10 12:20:44 , processing array  84 
2019-04-10 12:20:44 , processing array  85 
2019-04-10 12:20:44 , processing array  86 
2019-04-10 12:20:44 , processing array  87 
2019-04-10 12:20:44 , processing array  88 
2019-04-10 12:20:44 , processing array  89 
2019-04-10 12:20:44 , processing array  90 
2019-04-10 12:20:44 , processing array  91 
2019-04-10 12:20:44 , processing array  92 
2019-04-10 12:20:44 , processing array  93 
2019-04-10 12:20:45 , processing array  94 
2019-04-10 12:20:45 , processing array  95 
2019-04-10 12:20:45 , processing array  96 
2019-04-10 12:20:45 , processing array  97 
2019-04-10 12:20:45 , processing array  98 
2019-04-10 12:20:45 , processing array  99 
2019-04-10 12:20:45 , processing array  100 
2019-04-10 12:20:45 , processing array  101 
2019-04-10 12:20:45 , processing array  102 
2019-04-10 12:20:45 , processing array  103 
2019-04-10 12:20:45 , processing array  104 
2019-04-10 12:20:45 , processing array  105 
2019-04-10 12:20:45 , processing array  106 
2019-04-10 12:20:46 , processing array  107 
2019-04-10 12:20:46 , processing array  108 
2019-04-10 12:20:46 , processing array  109 
2019-04-10 12:20:46 , processing array  110 
2019-04-10 12:20:46 , processing array  111 
2019-04-10 12:20:46 , processing array  112 
2019-04-10 12:20:46 , processing array  113 
2019-04-10 12:20:46 , processing array  114 
2019-04-10 12:20:46 , processing array  115 
2019-04-10 12:20:46 , processing array  116 
2019-04-10 12:20:46 , processing array  117 
2019-04-10 12:20:46 , processing array  118 
2019-04-10 12:20:46 , processing array  119 
2019-04-10 12:20:46 , processing array  120 
2019-04-10 12:20:47 , processing array  121 
2019-04-10 12:20:47 , processing array  122 
2019-04-10 12:20:47 , processing array  123 
2019-04-10 12:20:47 , processing array  124 
2019-04-10 12:20:47 , processing array  125 
2019-04-10 12:20:47 , processing array  126 
2019-04-10 12:20:47 , processing array  127 
2019-04-10 12:20:47 , processing array  128 
2019-04-10 12:20:47 , processing array  129 
2019-04-10 12:20:47 , processing array  130 
2019-04-10 12:20:47 , processing array  131 
2019-04-10 12:20:47 , processing array  132 
2019-04-10 12:20:47 , processing array  133 
2019-04-10 12:20:48 , processing array  134 
2019-04-10 12:20:48 , processing array  135 
2019-04-10 12:20:48 , processing array  136 
2019-04-10 12:20:48 , processing array  137 
2019-04-10 12:20:48 , processing array  138 
2019-04-10 12:20:48 , processing array  139 
2019-04-10 12:20:48 , processing array  140 
2019-04-10 12:20:48 , processing array  141 
2019-04-10 12:20:48 , processing array  142 
2019-04-10 12:20:48 , processing array  143 
2019-04-10 12:20:48 , processing array  144 
2019-04-10 12:20:48 , processing array  145 
2019-04-10 12:20:48 , processing array  146 
2019-04-10 12:20:48 , processing array  147 
2019-04-10 12:20:49 , processing array  148 
2019-04-10 12:20:49 , processing array  149 
2019-04-10 12:20:49 , processing array  150 
2019-04-10 12:20:49 , processing array  151 
2019-04-10 12:20:49 , processing array  152 
2019-04-10 12:20:49 , processing array  153 
2019-04-10 12:20:49 , processing array  154 
2019-04-10 12:20:49 , processing array  155 
2019-04-10 12:20:49 , processing array  156 
2019-04-10 12:20:49 , processing array  157 
2019-04-10 12:20:49 , processing array  158 
2019-04-10 12:20:49 , processing array  159 
2019-04-10 12:20:49 , processing array  160 
2019-04-10 12:20:50 , processing array  161 
2019-04-10 12:20:50 , processing array  162 
2019-04-10 12:20:50 , processing array  163 
2019-04-10 12:20:50 , processing array  164 
2019-04-10 12:20:50 , processing array  165 
2019-04-10 12:20:50 , processing array  166 
2019-04-10 12:20:50 , processing array  167 
2019-04-10 12:20:50 , processing array  168 
2019-04-10 12:20:50 , processing array  169 
2019-04-10 12:20:50 , processing array  170 
2019-04-10 12:20:50 , processing array  171 
2019-04-10 12:20:50 , processing array  172 
2019-04-10 12:20:50 , processing array  173 
2019-04-10 12:20:51 , processing array  174 
2019-04-10 12:20:51 , processing array  175 
2019-04-10 12:20:51 , processing array  176 
2019-04-10 12:20:51 , processing array  177 
2019-04-10 12:20:51 , processing array  178 
2019-04-10 12:20:51 , processing array  179 
2019-04-10 12:20:51 , processing array  180 
2019-04-10 12:20:51 , processing array  181 
2019-04-10 12:20:51 , processing array  182 
2019-04-10 12:20:51 , processing array  183 
2019-04-10 12:20:51 , processing array  184 
2019-04-10 12:20:51 , processing array  185 
2019-04-10 12:20:51 , processing array  186 
2019-04-10 12:20:52 , processing array  187 
2019-04-10 12:20:52 , processing array  188 
2019-04-10 12:20:52 , processing array  189 
2019-04-10 12:20:52 , processing array  190 
2019-04-10 12:20:52 , processing array  191 
2019-04-10 12:20:52 , processing array  192 
2019-04-10 12:20:52 , processing array  193 
2019-04-10 12:20:52 , processing array  194 
2019-04-10 12:20:52 , processing array  195 
2019-04-10 12:20:52 , processing array  196 
2019-04-10 12:20:52 , processing array  197 
2019-04-10 12:20:52 , processing array  198 
2019-04-10 12:20:52 , processing array  199 
2019-04-10 12:20:53 , processing array  200 
2019-04-10 12:20:53 , processing array  201 
2019-04-10 12:20:53 , processing array  202 
2019-04-10 12:20:53 , processing array  203 
2019-04-10 12:20:53 , processing array  204 
2019-04-10 12:20:53 , processing array  205 
2019-04-10 12:20:53 , processing array  206 
2019-04-10 12:20:53 , processing array  207 
2019-04-10 12:20:53 , processing array  208 
2019-04-10 12:20:53 , processing array  209 
2019-04-10 12:20:53 , processing array  210 
2019-04-10 12:20:53 , processing array  211 
2019-04-10 12:20:53 , processing array  212 
2019-04-10 12:20:53 , processing array  213 
2019-04-10 12:20:54 , processing array  214 
2019-04-10 12:20:54 , processing array  215 
2019-04-10 12:20:54 , processing array  216 
2019-04-10 12:20:54 , processing array  217 
2019-04-10 12:20:54 , processing array  218 
2019-04-10 12:20:54 , processing array  219 
2019-04-10 12:20:54 , processing array  220 
2019-04-10 12:20:54 , processing array  221 
2019-04-10 12:20:54 , processing array  222 
2019-04-10 12:20:54 , processing array  223 
2019-04-10 12:20:54 , processing array  224 
2019-04-10 12:20:54 , processing array  225 
2019-04-10 12:20:54 , processing array  226 
2019-04-10 12:20:54 , processing array  227 
2019-04-10 12:20:55 , processing array  228 
2019-04-10 12:20:55 , processing array  229 
2019-04-10 12:20:55 , processing array  230 
2019-04-10 12:20:55 , processing array  231 
2019-04-10 12:20:55 , processing array  232 
2019-04-10 12:20:55 , processing array  233 
2019-04-10 12:20:55 , processing array  234 
2019-04-10 12:20:55 , processing array  235 
2019-04-10 12:20:55 , processing array  236 
2019-04-10 12:20:55 , processing array  237 
2019-04-10 12:20:55 , processing array  238 
2019-04-10 12:20:55 , processing array  239 
2019-04-10 12:20:55 , processing array  240 
2019-04-10 12:20:56 , processing array  241 
2019-04-10 12:20:56 , processing array  242 
2019-04-10 12:20:56 , processing array  243 
2019-04-10 12:20:56 , processing array  244 
2019-04-10 12:20:56 , processing array  245 
2019-04-10 12:20:56 , processing array  246 
2019-04-10 12:20:56 , processing array  247 
2019-04-10 12:20:56 , processing array  248 
2019-04-10 12:20:56 , processing array  249 
2019-04-10 12:20:56 , processing array  250 
2019-04-10 12:20:56 , processing array  251 
2019-04-10 12:20:56 , processing array  252 
2019-04-10 12:20:56 , processing array  253 
2019-04-10 12:20:57 , processing array  254 
2019-04-10 12:20:57 , processing array  255 
2019-04-10 12:20:57 , processing array  256 
2019-04-10 12:20:57 , processing array  257 
2019-04-10 12:20:57 , processing array  258 
2019-04-10 12:20:57 , processing array  259 
2019-04-10 12:20:57 , processing array  260 
2019-04-10 12:20:57 , processing array  261 
2019-04-10 12:20:57 , processing array  262 
2019-04-10 12:20:57 , processing array  263 
2019-04-10 12:20:57 , processing array  264 
2019-04-10 12:20:57 , processing array  265 
2019-04-10 12:20:57 , processing array  266 
2019-04-10 12:20:58 , processing array  267 
2019-04-10 12:20:58 , processing array  268 
2019-04-10 12:20:58 , processing array  269 
2019-04-10 12:20:58 , processing array  270 
2019-04-10 12:20:58 , processing array  271 
2019-04-10 12:20:58 , processing array  272 
2019-04-10 12:20:58 , processing array  273 
2019-04-10 12:20:58 , processing array  274 
2019-04-10 12:20:58 , processing array  275 
2019-04-10 12:20:58 , processing array  276 
2019-04-10 12:20:58 , processing array  277 
2019-04-10 12:20:58 , processing array  278 
2019-04-10 12:20:58 , processing array  279 
2019-04-10 12:20:59 , processing array  280 
2019-04-10 12:20:59 , processing array  281 
2019-04-10 12:20:59 , processing array  282 
2019-04-10 12:20:59 , processing array  283 
2019-04-10 12:20:59 , processing array  284 
2019-04-10 12:20:59 , processing array  285 
2019-04-10 12:20:59 , processing array  286 
2019-04-10 12:20:59 , processing array  287 
2019-04-10 12:20:59 , processing array  288 
2019-04-10 12:20:59 , processing array  289 
2019-04-10 12:20:59 , processing array  290 
2019-04-10 12:20:59 , processing array  291 
2019-04-10 12:20:59 , processing array  292 
2019-04-10 12:21:00 , processing array  293 
2019-04-10 12:21:00 , processing array  294 
2019-04-10 12:21:00 , processing array  295 
2019-04-10 12:21:00 , processing array  296 
2019-04-10 12:21:00 , processing array  297 
2019-04-10 12:21:00 , processing array  298 
2019-04-10 12:21:00 , processing array  299 
2019-04-10 12:21:00 , processing array  300 
2019-04-10 12:21:00 , processing array  301 
2019-04-10 12:21:00 , processing array  302 
2019-04-10 12:21:00 , processing array  303 
2019-04-10 12:21:00 , processing array  304 
2019-04-10 12:21:01 , processing array  305 
2019-04-10 12:21:01 , processing array  306 
2019-04-10 12:21:01 , processing array  307 
2019-04-10 12:21:01 , processing array  308 
2019-04-10 12:21:01 , processing array  309 
2019-04-10 12:21:01 , processing array  310 
2019-04-10 12:21:01 , processing array  311 
2019-04-10 12:21:01 , processing array  312 
2019-04-10 12:21:01 , processing array  313 
2019-04-10 12:21:01 , processing array  314 
2019-04-10 12:21:01 , processing array  315 
2019-04-10 12:21:01 , processing array  316 
2019-04-10 12:21:01 , processing array  317 
2019-04-10 12:21:02 , processing array  318 
2019-04-10 12:21:02 , processing array  319 
2019-04-10 12:21:02 , processing array  320 
2019-04-10 12:21:02 , processing array  321 
2019-04-10 12:21:02 , processing array  322 
2019-04-10 12:21:02 , processing array  323 
2019-04-10 12:21:02 , processing array  324 
2019-04-10 12:21:02 , processing array  325 
2019-04-10 12:21:02 , processing array  326 
2019-04-10 12:21:02 , processing array  327 
2019-04-10 12:21:02 , processing array  328 
2019-04-10 12:21:02 , processing array  329 
2019-04-10 12:21:02 , processing array  330 
2019-04-10 12:21:03 , processing array  331 
2019-04-10 12:21:03 , processing array  332 
2019-04-10 12:21:03 , processing array  333 
2019-04-10 12:21:03 , processing array  334 
2019-04-10 12:21:03 , processing array  335 
2019-04-10 12:21:03 , processing array  336 
2019-04-10 12:21:03 , processing array  337 
2019-04-10 12:21:03 , processing array  338 
2019-04-10 12:21:03 , processing array  339 
2019-04-10 12:21:03 , processing array  340 
2019-04-10 12:21:03 , processing array  341 
2019-04-10 12:21:03 , processing array  342 
2019-04-10 12:21:03 , processing array  343 
2019-04-10 12:21:04 , processing array  344 
2019-04-10 12:21:04 , processing array  345 
2019-04-10 12:21:04 , processing array  346 
2019-04-10 12:21:04 , processing array  347 
2019-04-10 12:21:04 , processing array  348 
2019-04-10 12:21:04 , processing array  349 
2019-04-10 12:21:04 , processing array  350 
2019-04-10 12:21:04 , processing array  351 
2019-04-10 12:21:04 , processing array  352 
2019-04-10 12:21:04 , processing array  353 
2019-04-10 12:21:04 , processing array  354 
2019-04-10 12:21:04 , processing array  355 
2019-04-10 12:21:04 , processing array  356 
2019-04-10 12:21:05 , processing array  357 
2019-04-10 12:21:05 , processing array  358 
2019-04-10 12:21:05 , processing array  359 
2019-04-10 12:21:05 , processing array  360 
2019-04-10 12:21:05 , processing array  361 
2019-04-10 12:21:05 , processing array  362 
2019-04-10 12:21:05 , processing array  363 
2019-04-10 12:21:05 , processing array  364 
2019-04-10 12:21:05 , processing array  365 
2019-04-10 12:21:05 , processing array  366 
2019-04-10 12:21:05 , processing array  367 
2019-04-10 12:21:05 , processing array  368 
2019-04-10 12:21:05 , processing array  369 
2019-04-10 12:21:06 , processing array  370 
2019-04-10 12:21:06 , processing array  371 
2019-04-10 12:21:06 , processing array  372 
2019-04-10 12:21:06 , processing array  373 
2019-04-10 12:21:06 , processing array  374 
2019-04-10 12:21:06 , processing array  375 
2019-04-10 12:21:06 , processing array  376 
2019-04-10 12:21:06 , processing array  377 
2019-04-10 12:21:06 , processing array  378 
2019-04-10 12:21:06 , processing array  379 
2019-04-10 12:21:06 , processing array  380 
2019-04-10 12:21:06 , processing array  381 
2019-04-10 12:21:06 , processing array  382 
2019-04-10 12:21:07 , processing array  383 
2019-04-10 12:21:07 , processing array  384 
done.

Quality control after preprocessing ...
No Quality Control assessment of the object because it is not a &quot;LumiBatch&quot; object.
done.</code></pre>
<pre class="r"><code>write.exprs(lumi_46903, file = &quot;~/tcga_macs/output/processed_lumi_46903.txt&quot;)</code></pre>
<pre class="r"><code># review metadata
show(lumi_46903)</code></pre>
<pre><code>ExpressionSet (storageMode: lockedEnvironment)
assayData: 47463 features, 384 samples 
  element names: detection, exprs 
protocolData: none
phenoData
  sampleNames: 4233291007_A 4233291007_B ... 5376074056_L (384
    total)
  varLabels: sampleID
  varMetadata: labelDescription
featureData
  featureNames: 6450255 2570615 ... 4120753 (47463 total)
  fvarLabels: ProbeID
  fvarMetadata: labelDescription
experimentData: use &#39;experimentData(object)&#39;
Annotation:  </code></pre>
<pre class="r"><code>raw_metadata &lt;- pData(phenoData(lumi_46903))
head(raw_metadata)</code></pre>
<pre><code>                 sampleID
4233291007_A 4233291007_A
4233291007_B 4233291007_B
4233291007_C 4233291007_C
4233291007_D 4233291007_D
4233291007_E 4233291007_E
4233291007_F 4233291007_F</code></pre>
<p>The raw data is labeled with alternate sample names (a 10-digit number followed by underscore and a letter) compared to the GEO records and the sample name in the processed data GSE ExpressionSet object.</p>
<ul>
<li><p>Check phenotype records in GSE object to see if I can merge with raw data using 10-digit letter name as a key</p></li>
<li><p>Documentation for manipulating ExpressionSet objects in general <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf">here</a></p></li>
<li><p>Documentation for maniupulating ExpressionSets created by GEOquery <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/GEOquery/inst/doc/GEOquery.html#getting-gse-series-matrix-files-as-an-expressionset">here</a></p></li>
</ul>
<pre class="r"><code>show(pData(phenoData(gse46903[[1]]))[1:20,c(1,6,8)])</code></pre>
<pre><code>                                    title type  source_name_ch1
GSM1140407          B cell [5342595013_G]  RNA peripheral blood
GSM1140408          B cell [5342595013_I]  RNA peripheral blood
GSM1140409          B cell [5342595027_I]  RNA peripheral blood
GSM1140410          DC_reg [5342595013_D]  RNA peripheral blood
GSM1140411          DC_imm [5342595013_E]  RNA peripheral blood
GSM1140412          DC_mat [5342595013_H]  RNA peripheral blood
GSM1140413          DC_mat [5342595013_L]  RNA peripheral blood
GSM1140414          DC_imm [5342595027_A]  RNA peripheral blood
GSM1140415          DC_mat [5342595027_D]  RNA peripheral blood
GSM1140416          DC_reg [5342595027_E]  RNA peripheral blood
GSM1140417          DC_imm [5342595027_H]  RNA peripheral blood
GSM1140418          DC_reg [5342595027_J]  RNA peripheral blood
GSM1140419          DC_imm [5376074046_A]  RNA peripheral blood
GSM1140420 DC_upLPS_10_24h [5376074046_B]  RNA peripheral blood
GSM1140421 DC_upLPS_10_24h [5376074046_C]  RNA peripheral blood
GSM1140422 DC_upLPS_10_24h [5376074046_D]  RNA peripheral blood
GSM1140423 DC_upLPS_10_24h [5376074046_E]  RNA peripheral blood
GSM1140424          DC_imm [5376074046_F]  RNA peripheral blood
GSM1140425          DC_imm [5376074046_G]  RNA peripheral blood
GSM1140426 DC_upLPS_10_24h [5376074046_H]  RNA peripheral blood</code></pre>
<p>The 10-digit sample names correspond to experimental conditions and are appended to the phenodata sample names! I will have to get each component ready and then assemble a new expressionset with complete metadata and raw data but this is do-able.</p>
<ul>
<li>Get data and annotations out of AnnotatedDataFrame object, documentation <a href="https://www.rdocumentation.org/packages/Biobase/versions/2.32.0/topics/AnnotatedDataFrame">here</a></li>
</ul>
<pre class="r"><code># pull phenoData from expression set as an AnnotatedDataFrame
meta46903 &lt;- phenoData(gse46903[[1]])

# get data and annotations as separate objects
meta_data46903 &lt;- pData(meta46903)
anno_data46903 &lt;- varMetadata(meta46903)

# doublecheck structure
str(meta_data46903)</code></pre>
<pre><code>&#39;data.frame&#39;:   384 obs. of  38 variables:
 $ title                      : Factor w/ 384 levels &quot;B cell [5342595013_G]&quot;,..: 1 2 3 22 4 19 20 5 21 23 ...
 $ geo_accession              : chr  &quot;GSM1140407&quot; &quot;GSM1140408&quot; &quot;GSM1140409&quot; &quot;GSM1140410&quot; ...
 $ status                     : Factor w/ 1 level &quot;Public on Feb 13 2014&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ submission_date            : Factor w/ 1 level &quot;May 14 2013&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ last_update_date           : Factor w/ 3 levels &quot;Feb 13 2014&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...
 $ type                       : Factor w/ 1 level &quot;RNA&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ channel_count              : Factor w/ 1 level &quot;1&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ source_name_ch1            : Factor w/ 1 level &quot;peripheral blood&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ organism_ch1               : Factor w/ 1 level &quot;Homo sapiens&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ characteristics_ch1        : Factor w/ 13 levels &quot;initial differentiation: -&quot;,..: 1 1 1 9 9 9 9 9 9 9 ...
 $ characteristics_ch1.1      : Factor w/ 41 levels &quot;activation stimuli: CD3/28&quot;,..: 39 39 39 34 21 33 33 21 33 34 ...
 $ characteristics_ch1.2      : Factor w/ 12 levels &quot;time: 0h&quot;,&quot;time: 12h&quot;,..: 1 1 1 11 1 11 11 1 11 11 ...
 $ characteristics_ch1.3      : Factor w/ 73 levels &quot;donor: BC 65&quot;,..: 56 68 67 71 70 69 70 71 71 70 ...
 $ molecule_ch1               : Factor w/ 1 level &quot;total RNA&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ extract_protocol_ch1       : Factor w/ 1 level &quot;RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit i&quot;| __truncated__: 1 1 1 1 1 1 1 1 1 1 ...
 $ label_ch1                  : Factor w/ 1 level &quot;biotin&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ label_protocol_ch1         : Factor w/ 1 level &quot;Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ taxid_ch1                  : Factor w/ 1 level &quot;9606&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ hyb_protocol               : Factor w/ 1 level &quot;Standard Illumina hybridization protocol&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ scan_protocol              : Factor w/ 1 level &quot;Standard Illumina scanning protocol&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ description                : Factor w/ 6 levels &quot;B cell&quot;,&quot;Dendritic cell&quot;,..: 1 1 1 2 2 2 2 2 2 2 ...
 $ data_processing            : Factor w/ 1 level &quot;The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ platform_id                : Factor w/ 1 level &quot;GPL6947&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_name               : Factor w/ 1 level &quot;Joachim,,Schultze&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_email              : Factor w/ 1 level &quot;j.schultze@uni-bonn.de&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_department         : Factor w/ 1 level &quot;Genomics and Immunoregulation&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_institute          : Factor w/ 1 level &quot;LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_address            : Factor w/ 1 level &quot;Carl-Troll-Strasse 31&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_city               : Factor w/ 1 level &quot;Bonn&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_state              : Factor w/ 1 level &quot;NRW&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_zip/postal_code    : Factor w/ 1 level &quot;53115&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ contact_country            : Factor w/ 1 level &quot;Germany&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ supplementary_file         : Factor w/ 1 level &quot;NONE&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ data_row_count             : Factor w/ 1 level &quot;47463&quot;: 1 1 1 1 1 1 1 1 1 1 ...
 $ activation stimuli:ch1     : chr  &quot;unstimulated&quot; &quot;unstimulated&quot; &quot;unstimulated&quot; &quot;TNFa 800 u/ml+ PGE2 1µg/ml+ P3C 1µg/ml&quot; ...
 $ donor:ch1                  : chr  &quot;D1&quot; &quot;D4&quot; &quot;D3&quot; &quot;D7&quot; ...
 $ initial differentiation:ch1: chr  &quot;-&quot; &quot;-&quot; &quot;-&quot; &quot;GM-CSF + IL4&quot; ...
 $ time:ch1                   : chr  &quot;0h&quot; &quot;0h&quot; &quot;0h&quot; &quot;72h&quot; ...</code></pre>
<pre class="r"><code>head(meta_data46903)</code></pre>
<pre><code>                           title geo_accession                status
GSM1140407 B cell [5342595013_G]    GSM1140407 Public on Feb 13 2014
GSM1140408 B cell [5342595013_I]    GSM1140408 Public on Feb 13 2014
GSM1140409 B cell [5342595027_I]    GSM1140409 Public on Feb 13 2014
GSM1140410 DC_reg [5342595013_D]    GSM1140410 Public on Feb 13 2014
GSM1140411 DC_imm [5342595013_E]    GSM1140411 Public on Feb 13 2014
GSM1140412 DC_mat [5342595013_H]    GSM1140412 Public on Feb 13 2014
           submission_date last_update_date type channel_count
GSM1140407     May 14 2013      Feb 13 2014  RNA             1
GSM1140408     May 14 2013      Feb 13 2014  RNA             1
GSM1140409     May 14 2013      Feb 13 2014  RNA             1
GSM1140410     May 14 2013      Feb 13 2014  RNA             1
GSM1140411     May 14 2013      Feb 13 2014  RNA             1
GSM1140412     May 14 2013      Feb 13 2014  RNA             1
            source_name_ch1 organism_ch1
GSM1140407 peripheral blood Homo sapiens
GSM1140408 peripheral blood Homo sapiens
GSM1140409 peripheral blood Homo sapiens
GSM1140410 peripheral blood Homo sapiens
GSM1140411 peripheral blood Homo sapiens
GSM1140412 peripheral blood Homo sapiens
                             characteristics_ch1
GSM1140407            initial differentiation: -
GSM1140408            initial differentiation: -
GSM1140409            initial differentiation: -
GSM1140410 initial differentiation: GM-CSF + IL4
GSM1140411 initial differentiation: GM-CSF + IL4
GSM1140412 initial differentiation: GM-CSF + IL4
                                                characteristics_ch1.1
GSM1140407                           activation stimuli: unstimulated
GSM1140408                           activation stimuli: unstimulated
GSM1140409                           activation stimuli: unstimulated
GSM1140410 activation stimuli: TNFa 800 u/ml+ PGE2 1µg/ml+ P3C 1µg/ml
GSM1140411                                   activation stimuli: none
GSM1140412                          activation stimuli: TNFa 800 u/ml
           characteristics_ch1.2 characteristics_ch1.3 molecule_ch1
GSM1140407              time: 0h             donor: D1    total RNA
GSM1140408              time: 0h             donor: D4    total RNA
GSM1140409              time: 0h             donor: D3    total RNA
GSM1140410             time: 72h             donor: D7    total RNA
GSM1140411              time: 0h             donor: D6    total RNA
GSM1140412             time: 72h             donor: D5    total RNA
                                                                                                                                                                                                                              extract_protocol_ch1
GSM1140407 RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.
GSM1140408 RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.
GSM1140409 RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.
GSM1140410 RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.
GSM1140411 RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.
GSM1140412 RNA was extracted with Trizol reagent, followed by clean-up and DNase I treatment with QIAGEN RNeasy mini kit in accordance with the prescribed protocol provided with the kit. Quality control was performed with Agilent Bioanalyser.
           label_ch1
GSM1140407    biotin
GSM1140408    biotin
GSM1140409    biotin
GSM1140410    biotin
GSM1140411    biotin
GSM1140412    biotin
                                                                           label_protocol_ch1
GSM1140407 Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays
GSM1140408 Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays
GSM1140409 Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays
GSM1140410 Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays
GSM1140411 Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays
GSM1140412 Biotinylated cRNA were prepared with the Ambion MessageAmp kit for Illumina arrays
           taxid_ch1                             hyb_protocol
GSM1140407      9606 Standard Illumina hybridization protocol
GSM1140408      9606 Standard Illumina hybridization protocol
GSM1140409      9606 Standard Illumina hybridization protocol
GSM1140410      9606 Standard Illumina hybridization protocol
GSM1140411      9606 Standard Illumina hybridization protocol
GSM1140412      9606 Standard Illumina hybridization protocol
                                 scan_protocol    description
GSM1140407 Standard Illumina scanning protocol         B cell
GSM1140408 Standard Illumina scanning protocol         B cell
GSM1140409 Standard Illumina scanning protocol         B cell
GSM1140410 Standard Illumina scanning protocol Dendritic cell
GSM1140411 Standard Illumina scanning protocol Dendritic cell
GSM1140412 Standard Illumina scanning protocol Dendritic cell
                                                                                                 data_processing
GSM1140407 The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite
GSM1140408 The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite
GSM1140409 The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite
GSM1140410 The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite
GSM1140411 The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite
GSM1140412 The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite
           platform_id      contact_name          contact_email
GSM1140407     GPL6947 Joachim,,Schultze j.schultze@uni-bonn.de
GSM1140408     GPL6947 Joachim,,Schultze j.schultze@uni-bonn.de
GSM1140409     GPL6947 Joachim,,Schultze j.schultze@uni-bonn.de
GSM1140410     GPL6947 Joachim,,Schultze j.schultze@uni-bonn.de
GSM1140411     GPL6947 Joachim,,Schultze j.schultze@uni-bonn.de
GSM1140412     GPL6947 Joachim,,Schultze j.schultze@uni-bonn.de
                      contact_department
GSM1140407 Genomics and Immunoregulation
GSM1140408 Genomics and Immunoregulation
GSM1140409 Genomics and Immunoregulation
GSM1140410 Genomics and Immunoregulation
GSM1140411 Genomics and Immunoregulation
GSM1140412 Genomics and Immunoregulation
                                                                contact_institute
GSM1140407 LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)
GSM1140408 LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)
GSM1140409 LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)
GSM1140410 LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)
GSM1140411 LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)
GSM1140412 LIMES (Life and Medical Sciences Center Genomics and Immunoregulation)
                 contact_address contact_city contact_state
GSM1140407 Carl-Troll-Strasse 31         Bonn           NRW
GSM1140408 Carl-Troll-Strasse 31         Bonn           NRW
GSM1140409 Carl-Troll-Strasse 31         Bonn           NRW
GSM1140410 Carl-Troll-Strasse 31         Bonn           NRW
GSM1140411 Carl-Troll-Strasse 31         Bonn           NRW
GSM1140412 Carl-Troll-Strasse 31         Bonn           NRW
           contact_zip/postal_code contact_country supplementary_file
GSM1140407                   53115         Germany               NONE
GSM1140408                   53115         Germany               NONE
GSM1140409                   53115         Germany               NONE
GSM1140410                   53115         Germany               NONE
GSM1140411                   53115         Germany               NONE
GSM1140412                   53115         Germany               NONE
           data_row_count                 activation stimuli:ch1 donor:ch1
GSM1140407          47463                           unstimulated        D1
GSM1140408          47463                           unstimulated        D4
GSM1140409          47463                           unstimulated        D3
GSM1140410          47463 TNFa 800 u/ml+ PGE2 1µg/ml+ P3C 1µg/ml        D7
GSM1140411          47463                                   none        D6
GSM1140412          47463                          TNFa 800 u/ml        D5
           initial differentiation:ch1 time:ch1
GSM1140407                           -       0h
GSM1140408                           -       0h
GSM1140409                           -       0h
GSM1140410                GM-CSF + IL4      72h
GSM1140411                GM-CSF + IL4       0h
GSM1140412                GM-CSF + IL4      72h</code></pre>
<pre class="r"><code>str(anno_data46903)</code></pre>
<pre><code>&#39;data.frame&#39;:   38 obs. of  1 variable:
 $ labelDescription: chr  NA NA NA NA ...</code></pre>
<pre class="r"><code>head(anno_data46903)</code></pre>
<pre><code>                 labelDescription
title                        &lt;NA&gt;
geo_accession                &lt;NA&gt;
status                       &lt;NA&gt;
submission_date              &lt;NA&gt;
last_update_date             &lt;NA&gt;
type                         &lt;NA&gt;</code></pre>
<pre class="r"><code># anno_data is pretty useless - just empty variables with matching rownames from pData</code></pre>
<br> <br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] lumi_2.34.0         GEOquery_2.50.5     Biobase_2.42.0     
 [4] BiocGenerics_0.28.0 forcats_0.4.0       stringr_1.4.0      
 [7] dplyr_0.8.0.1       purrr_0.3.2         readr_1.3.1        
[10] tidyr_0.8.3         tibble_2.1.1        ggplot2_3.1.1      
[13] tidyverse_1.2.1    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                backports_1.1.3            
  [3] workflowr_1.2.0             plyr_1.8.4                 
  [5] lazyeval_0.2.2              splines_3.5.3              
  [7] BiocParallel_1.16.6         GenomeInfoDb_1.18.2        
  [9] digest_0.6.18               foreach_1.4.4              
 [11] htmltools_0.3.6             magrittr_1.5               
 [13] memoise_1.1.0               limma_3.38.3               
 [15] Biostrings_2.50.2           annotate_1.60.1            
 [17] modelr_0.1.4                matrixStats_0.54.0         
 [19] askpass_1.1                 siggenes_1.56.0            
 [21] prettyunits_1.0.2           colorspace_1.4-1           
 [23] blob_1.1.1                  rvest_0.3.2                
 [25] haven_2.1.0                 xfun_0.6                   
 [27] crayon_1.3.4                RCurl_1.95-4.12            
 [29] jsonlite_1.6                genefilter_1.64.0          
 [31] survival_2.44-1.1           iterators_1.0.10           
 [33] glue_1.3.1                  registry_0.5-1             
 [35] gtable_0.3.0                zlibbioc_1.28.0            
 [37] XVector_0.22.0              DelayedArray_0.8.0         
 [39] Rhdf5lib_1.4.3              HDF5Array_1.10.1           
 [41] scales_1.0.0                DBI_1.0.0                  
 [43] rngtools_1.3.1              bibtex_0.4.2               
 [45] Rcpp_1.0.1                  xtable_1.8-3               
 [47] progress_1.2.0              bumphunter_1.24.5          
 [49] bit_1.1-14                  mclust_5.4.3               
 [51] preprocessCore_1.44.0       stats4_3.5.3               
 [53] httr_1.4.0                  RColorBrewer_1.1-2         
 [55] pkgconfig_2.0.2             reshape_0.8.8              
 [57] XML_3.98-1.19               locfit_1.5-9.1             
 [59] tidyselect_0.2.5            rlang_0.3.4                
 [61] AnnotationDbi_1.44.0        munsell_0.5.0              
 [63] cellranger_1.1.0            tools_3.5.3                
 [65] cli_1.1.0                   generics_0.0.2             
 [67] RSQLite_2.1.1               broom_0.5.2                
 [69] evaluate_0.13               yaml_2.2.0                 
 [71] knitr_1.22                  bit64_0.9-7                
 [73] fs_1.2.7                    beanplot_1.2               
 [75] methylumi_2.28.0            nlme_3.1-138               
 [77] doRNG_1.7.1                 whisker_0.3-2              
 [79] nor1mix_1.2-3               xml2_1.2.0                 
 [81] biomaRt_2.38.0              compiler_3.5.3             
 [83] rstudioapi_0.10             curl_3.3                   
 [85] affyio_1.52.0               stringi_1.4.3              
 [87] GenomicFeatures_1.34.7      minfi_1.28.4               
 [89] lattice_0.20-38             Matrix_1.2-17              
 [91] multtest_2.38.0             pillar_1.3.1               
 [93] BiocManager_1.30.4          data.table_1.12.2          
 [95] bitops_1.0-6                rtracklayer_1.42.2         
 [97] GenomicRanges_1.34.0        R6_2.4.0                   
 [99] affy_1.60.0                 KernSmooth_2.23-15         
[101] IRanges_2.16.0              nleqslv_3.3.2              
[103] codetools_0.2-16            MASS_7.3-51.4              
[105] assertthat_0.2.1            rhdf5_2.26.2               
[107] SummarizedExperiment_1.12.0 openssl_1.3                
[109] pkgmaker_0.27               rprojroot_1.3-2            
[111] withr_2.1.2                 GenomicAlignments_1.18.1   
[113] Rsamtools_1.34.1            S4Vectors_0.20.1           
[115] GenomeInfoDbData_1.2.0      mgcv_1.8-28                
[117] hms_0.4.2                   quadprog_1.5-5             
[119] grid_3.5.3                  base64_2.0                 
[121] rmarkdown_1.12              DelayedMatrixStats_1.4.0   
[123] illuminaio_0.24.0           git2r_0.25.2               
[125] lubridate_1.7.4            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
