<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-09" />

<title>ssGSEA</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Macrophage signatures in TCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_macs">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">ssGSEA</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-11-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_macs/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.5.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190409code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190409)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190409code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190409)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstreee196ab1cea5a05b8e4ff292aa979b6c5dd701ac3targetblanke196ab1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_macs/tree/e196ab1cea5a05b8e4ff292aa979b6c5dd701ac3" target="_blank">e196ab1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstreee196ab1cea5a05b8e4ff292aa979b6c5dd701ac3targetblanke196ab1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    all.pdf
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/160genes_ensembl.gencode28.txt
    Ignored:    data/GSE46903_Affymetrix_Reanalyzed_data.txt.gz
    Ignored:    data/GSE46903_Affymetrix_metadata.txt.gz
    Ignored:    data/GSE46903_RAW.tar
    Ignored:    data/GSE46903_non-normalized.txt
    Ignored:    data/gencode.v22.primary_assembly.annotation.gtf.geneinfo
    Ignored:    data/gse_46903.rds
    Ignored:    data/htseq_counts/
    Ignored:    data/kyle.gdc_tcga_bam_metadata.txt
    Ignored:    output/AUC_comparison.pdf
    Ignored:    output/Combined_down_plots.pdf
    Ignored:    output/Combined_up_plots.pdf
    Ignored:    output/Fig1.png
    Ignored:    output/PFS_CoxPH.pdf
    Ignored:    output/Rplot.pdf
    Ignored:    output/arrange.pdf
    Ignored:    output/ave_scores_gsva_by_tumor.rds
    Ignored:    output/beeswarm_plot.pdf
    Ignored:    output/beeswarm_pub.pdf
    Ignored:    output/boxplot_total_pub.pdf
    Ignored:    output/combined_boxplots.pdf
    Ignored:    output/combined_up_plot.pdf
    Ignored:    output/coxph_pfs.pdf
    Ignored:    output/cpxph_os.pdf
    Ignored:    output/cv.lasso.merck.rds
    Ignored:    output/cv.lasso.rds
    Ignored:    output/dge_frame.csv
    Ignored:    output/downreg_c1.csv
    Ignored:    output/downreg_c2.csv
    Ignored:    output/downreg_c3.csv
    Ignored:    output/downreg_c4.csv
    Ignored:    output/downreg_c5.csv
    Ignored:    output/downreg_c6.csv
    Ignored:    output/downreg_c7.csv
    Ignored:    output/expr_matrix.rds
    Ignored:    output/expression_set.rds
    Ignored:    output/figures/
    Ignored:    output/filtered.total.phenodata.csv
    Ignored:    output/filtered.total.phenodata.rds
    Ignored:    output/flat_total_counts.rds
    Ignored:    output/flat_total_pheno.rds
    Ignored:    output/gset_ids_complete.rds
    Ignored:    output/gset_ids_plusnew.rds
    Ignored:    output/heatmaps_fig3.pdf
    Ignored:    output/hm_chol.pdf
    Ignored:    output/imtx_expressionset.rds
    Ignored:    output/imtx_gsva_17geneset_results.rds
    Ignored:    output/imtx_matrix.csv
    Ignored:    output/km_cluster_assignments.csv
    Ignored:    output/lasso.min.model.merck.rds
    Ignored:    output/lasso.min.model.rds
    Ignored:    output/lasso.test.data.merck.rds
    Ignored:    output/lasso.test.data.rds
    Ignored:    output/lasso.train.data.merck.rds
    Ignored:    output/lasso.train.data.rds
    Ignored:    output/mac_eset.rds
    Ignored:    output/mac_eset_w_clusters.rds
    Ignored:    output/nested_list_dge_top50_updown.rds
    Ignored:    output/normalized_probe_intensities.Rds
    Ignored:    output/os_km_curves.pdf
    Ignored:    output/os_km_plots.pdf
    Ignored:    output/processed_lumi_46903.Rds
    Ignored:    output/processed_lumi_46903.tsv
    Ignored:    output/processed_lumi_exprs.txt
    Ignored:    output/pub_boxplot.pdf
    Ignored:    output/score_matrix.csv
    Ignored:    output/selected.total.counts.csv
    Ignored:    output/summary heatmap.pdf
    Ignored:    output/table_c1.csv
    Ignored:    output/table_c2.csv
    Ignored:    output/table_c3.csv
    Ignored:    output/table_c4.csv
    Ignored:    output/table_c5.csv
    Ignored:    output/table_c6.csv
    Ignored:    output/table_c7.csv
    Ignored:    output/tcga_counts_dataframe_list.rds
    Ignored:    output/tcga_counts_list.rds
    Ignored:    output/tcga_gdc_manifest.rds
    Ignored:    output/tcga_gsva_17geneset_results.rds
    Ignored:    output/tcga_metadata_list.rds
    Ignored:    output/tcga_total_counts.csv
    Ignored:    output/univariate_p_values.pdf
    Ignored:    output/upreg_c1.csv
    Ignored:    output/upreg_c2.csv
    Ignored:    output/upreg_c3.csv
    Ignored:    output/upreg_c4.csv
    Ignored:    output/upreg_c5.csv
    Ignored:    output/upreg_c6.csv
    Ignored:    output/upreg_c7.csv

Untracked files:
    Untracked:  analysis/expanded_gsva.Rmd
    Untracked:  analysis/expanded_gsva_analysis.Rmd
    Untracked:  analysis/gsva_results_exploration.Rmd
    Untracked:  analysis/survival.Rmd
    Untracked:  data/IMTX_datasets/
    Untracked:  data/xcel_signatures.xlsx
    Untracked:  output/PFS_CoxPH-concordance.tex
    Untracked:  output/PFS_CoxPH.Rnw
    Untracked:  output/PFS_CoxPH.log
    Untracked:  output/PFS_CoxPH.synctex.gz
    Untracked:  output/PFS_CoxPH.tex
    Untracked:  output/Rplot01.png
    Untracked:  output/coxph_pfs-concordance.tex
    Untracked:  output/coxph_pfs.Rnw
    Untracked:  output/coxph_pfs.log
    Untracked:  output/coxph_pfs.synctex.gz
    Untracked:  output/coxph_pfs.tex
    Untracked:  output/cpxph_os-concordance.tex
    Untracked:  output/cpxph_os.Rnw
    Untracked:  output/cpxph_os.log
    Untracked:  output/cpxph_os.synctex.gz
    Untracked:  output/cpxph_os.tex
    Untracked:  output/univariate_p_values-concordance.tex
    Untracked:  output/univariate_p_values.Rnw
    Untracked:  output/univariate_p_values.log
    Untracked:  output/univariate_p_values.synctex.gz
    Untracked:  output/univariate_p_values.tex

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/tcga_format.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/e196ab1cea5a05b8e4ff292aa979b6c5dd701ac3/analysis/ss_gsea.Rmd" target="_blank">e196ab1</a>
</td>
<td>
mleukam
</td>
<td>
2019-11-09
</td>
<td>
patially done updating all the new mac groups
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/13fe6d072308ee5194511dfaff5a0242bb75788f/analysis/ss_gsea.Rmd" target="_blank">13fe6d0</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-19
</td>
<td>
updated box plots for presention
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/7215e3842c79010d7f93a6388421ef788f345f77/docs/ss_gsea.html" target="_blank">7215e38</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/db2b959936fbb6d77046a8f5716cb971784cc61c/analysis/ss_gsea.Rmd" target="_blank">db2b959</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
<td>
additional figures
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/3f5af70b78e3bce377c516a0412784251a98bba5/analysis/ss_gsea.Rmd" target="_blank">3f5af70</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
gsva R scripts
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/8d832f2ae5facbd32d81de3a667b61e9c599a25d/docs/ss_gsea.html" target="_blank">8d832f2</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/6288da104a3f55b98db0f6b8f1c26bd5264a72c5/analysis/ss_gsea.Rmd" target="_blank">6288da1</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-12
</td>
<td>
partially working GVSA analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/f7f6d34f895f240a8359c6260ad62912e243e865/docs/ss_gsea.html" target="_blank">f7f6d34</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/b3f3025e937ef56c3036575fae8b8934093f69c4/analysis/ss_gsea.Rmd" target="_blank">b3f3025</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
start of gsea notebook, reading in files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/64eaee846100fcfc638c0c2218400034aa71e7f9/analysis/ss_gsea.Rmd" target="_blank">64eaee8</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
multiple updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Clear environment</p>
<pre class="r"><code># clear environment
rm(list = ls())</code></pre>
<p>Load packages</p>
<pre class="r"><code>library(&quot;tidyverse&quot;)</code></pre>
<pre><code>── Attaching packages ───────────────────────── tidyverse 1.2.1 ──</code></pre>
<pre><code>✔ ggplot2 3.2.1     ✔ purrr   0.3.3
✔ tibble  2.1.3     ✔ dplyr   0.8.3
✔ tidyr   1.0.0     ✔ stringr 1.4.0
✔ readr   1.3.1     ✔ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ──────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;limma&quot;)
library(&quot;GSVA&quot;)
library(&quot;GenomicDataCommons&quot;)</code></pre>
<pre><code>Loading required package: magrittr</code></pre>
<pre><code>
Attaching package: &#39;magrittr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:purrr&#39;:

    set_names</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    extract</code></pre>
<pre><code>
Attaching package: &#39;GenomicDataCommons&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:dplyr&#39;:

    count, filter, select</code></pre>
<pre><code>The following object is masked from &#39;package:tidyr&#39;:

    expand</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    filter</code></pre>
</div>
<div id="gene-list-preprocessing" class="section level2">
<h2>Gene list preprocessing</h2>
<p>Read in custom gene lists</p>
<pre class="r"><code>up_files &lt;- list.files(&quot;output/&quot;, &quot;^upreg_c&quot;)
down_files &lt;- list.files(&quot;output/&quot;, &quot;downreg_c&quot;)
file_list &lt;- c(up_files, down_files)
path_list &lt;- paste0(&quot;output/&quot;, file_list)
dge_list &lt;- map(path_list, read_csv)</code></pre>
<pre><code>Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)
Parsed with column specification:
cols(
  gene = col_character(),
  logFC = col_double(),
  AveExpr = col_double(),
  t = col_double(),
  P.Value = col_double(),
  adj.P.Val = col_double(),
  B = col_double()
)</code></pre>
<pre class="r"><code>names(dge_list) &lt;- str_replace_all(file_list, &quot;.csv&quot;, &quot;&quot;)</code></pre>
<p>Read in lookup table for features, gencode v22 (used by GDC to label features)</p>
<pre class="r"><code>gencode_gtf &lt;- read_tsv(&quot;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  gene_id = col_character(),
  gene_type = col_character(),
  gene_status = col_character(),
  gene_name = col_character(),
  level = col_double(),
  havana_gene = col_character()
)</code></pre>
<pre><code>Warning: 10327 parsing failures.
  row col  expected    actual                                                        file
 5304  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
12445  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
12913  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
13082  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
13310  -- 6 columns 5 columns &#39;data/gencode.v22.primary_assembly.annotation.gtf.geneinfo&#39;
..... ... ......... ......... ...........................................................
See problems(...) for more details.</code></pre>
<pre class="r"><code># read in T-cell inflammation signature
tcell &lt;- read_tsv(&quot;data/160genes_ensembl.gencode28.txt&quot;, col_names = FALSE)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character(),
  X3 = col_character()
)</code></pre>
<pre class="r"><code># convert to gene_id
gene_ids &lt;- gencode_gtf %&gt;%
  dplyr::select(gene_id, gene_name)
tcell &lt;- tcell %&gt;% dplyr::rename(gene_name = X1)
tcell_gset &lt;- tcell %&gt;% left_join(gene_ids, by = &quot;gene_name&quot;) %&gt;%
  pull(gene_id)

tcell_gset &lt;- tcell_gset[!is.na(tcell_gset)]</code></pre>
<pre class="r"><code># define function that uses lookup table to convert gene symbols to gene id
# then returns a list of gene ids that define the gene set
ensemblgset &lt;- function(df, gtf){
  gene_ids &lt;- gtf %&gt;%
  dplyr::select(gene_id, gene_name) %&gt;%
  dplyr::rename(gene = gene_name)
gset &lt;- df %&gt;% left_join(gene_ids) %&gt;%
  pull(gene_id)
gset
}

# apply function to list of gene sets
gset_ids &lt;- map(dge_list, function(x){ensemblgset(x, gtf = gencode_gtf)})</code></pre>
<pre><code>Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;
Joining, by = &quot;gene&quot;</code></pre>
<pre class="r"><code># remove nas (small number of gene symbols do not map to gencode gene ids)
gset_ids &lt;- map(gset_ids, function(x){
  x[!is.na(x)]
})

# add T-cell signature to gset list
gset_ids_complete &lt;- c(gset_ids, tcell_gset = list(tcell_gset))</code></pre>
</div>
<div id="count-data-preprocessing" class="section level2">
<h2>Count data preprocessing</h2>
<p>Preprocessing following methods outlined here: <a href="https://f1000research.com/articles/5-1408/v3" class="uri">https://f1000research.com/articles/5-1408/v3</a></p>
<p>Read in data</p>
<pre class="r"><code>total_counts &lt;- read_csv(&quot;output/selected.total.counts.csv&quot;)</code></pre>
<pre><code>Parsed with column specification:
cols(
  .default = col_double(),
  gene = col_character()
)</code></pre>
<pre><code>See spec(...) for full column specifications.</code></pre>
<div id="filter-for-protein-coding-genes" class="section level4">
<h4>Filter for protein coding genes</h4>
<pre class="r"><code># filter for protein coding genes
total_counts_prcode &lt;- total_counts %&gt;%
  dplyr::rename(gene_id = gene) %&gt;%
  left_join(gencode_gtf, by = &quot;gene_id&quot;) %&gt;%
  dplyr::filter(gene_type == &quot;protein_coding&quot;) %&gt;%
  dplyr::select(-gene_name, -gene_type, -gene_status, -level, -havana_gene) %&gt;%
  dplyr::select(gene_id, everything())

nrow(total_counts_prcode)</code></pre>
<pre><code>[1] 19814</code></pre>
</div>
<div id="correct-for-library-size-convert-to-cpm" class="section level4">
<h4>Correct for library size: convert to CPM</h4>
<pre class="r"><code># normalize rows by log cpm using EdgeR
df_data &lt;- total_counts_prcode %&gt;% 
  dplyr::select(-gene_id) %&gt;% as.matrix()
df_names &lt;- total_counts_prcode %&gt;% dplyr::select(gene_id) 
out_data &lt;- cpm(df_data, log = FALSE) %&gt;% as_tibble()
total_counts_prcode_cpm &lt;- bind_cols(df_names, out_data)</code></pre>
</div>
<div id="density-plots" class="section level4">
<h4>Density plots</h4>
<pre class="r"><code># Density plots
# tidy data
tidy_cpm &lt;- total_counts_prcode_cpm %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene_id) 

tidy_cpm &lt;- tidy_cpm %&gt;%
  mutate(group = str_sub(sampleID, 1, 4))

# representative plots
brca_cpm &lt;- tidy_cpm %&gt;%
  dplyr::filter(group == &quot;BRCA&quot;)
dplot_brca &lt;- ggplot(brca_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;)
dplot_brca</code></pre>
<p><img src="figure/ss_gsea.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/7215e3842c79010d7f93a6388421ef788f345f77/docs/figure/ss_gsea.Rmd/unnamed-chunk-10-1.png" target="_blank">7215e38</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8d832f2ae5facbd32d81de3a667b61e9c599a25d/docs/figure/ss_gsea.Rmd/unnamed-chunk-10-1.png" target="_blank">8d832f2</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gbm_cpm &lt;- tidy_cpm %&gt;%
  dplyr::filter(grepl(&quot;GBM*&quot;, group))
dplot_gbm &lt;- ggplot(gbm_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-6, 15)
dplot_gbm</code></pre>
<pre><code>Warning: Removed 1570260 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/ss_gsea.Rmd/unnamed-chunk-10-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-2">
Past versions of unnamed-chunk-10-2.png
</button>
</p>
<div id="fig-unnamed-chunk-10-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/7215e3842c79010d7f93a6388421ef788f345f77/docs/figure/ss_gsea.Rmd/unnamed-chunk-10-2.png" target="_blank">7215e38</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8d832f2ae5facbd32d81de3a667b61e9c599a25d/docs/figure/ss_gsea.Rmd/unnamed-chunk-10-2.png" target="_blank">8d832f2</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filter-by-expression-levels" class="section level4">
<h4>Filter by expression levels</h4>
<pre class="r"><code># move gene names to rownames
totcounts_prcode_cpm_matrix &lt;- total_counts_prcode_cpm %&gt;%
  as.data.frame() %&gt;%
  column_to_rownames(var = &quot;gene_id&quot;)
totcounts_prcode_cpm_matrix[1:5, 1:5]</code></pre>
<pre><code>                       BRCA1      BRCA2      BRCA4     BRCA5       BRCA6
ENSG00000000003.13 37.019035 86.2823791 55.8841542 35.525131 25.71919321
ENSG00000000005.5   4.102186  0.5067543  0.1215401  2.208254  0.06486556
ENSG00000000419.11 23.576924 37.8277212 17.2100831 24.495930 40.00583206
ENSG00000000457.12 29.538578 38.3344755  9.4315145 34.270167 36.30849533
ENSG00000000460.15  9.708980 19.2566650  2.2606465 13.539130 21.60023036</code></pre>
<pre class="r"><code># filter out genes that aren&#39;t at least expressed greater than 1 in at least 45 cases (45 being about the size of the smallest group)
total_cpm_stats &lt;- data.frame(
  total = apply(totcounts_prcode_cpm_matrix, 1, function(x) sum(x &gt; 1, na.rm = TRUE)))
keep &lt;- which(total_cpm_stats$total &gt;= 45) 

dim(totcounts_prcode_cpm_matrix)</code></pre>
<pre><code>[1] 19814  9609</code></pre>
<pre class="r"><code>total_cpm_filtered = totcounts_prcode_cpm_matrix[keep,]
dim(total_cpm_filtered)</code></pre>
<pre><code>[1] 18017  9609</code></pre>
</div>
<div id="normalize-gene-expression-distributions" class="section level4">
<h4>Normalize gene expression distributions</h4>
<p>Normalisation by the method of trimmed mean of M-values (TMM) is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes.</p>
<pre class="r"><code># get normalization factors
norm_factors &lt;- calcNormFactors(total_cpm_filtered, method = &quot;TMM&quot;)

# apply factor to each column
total_cpm_norm &lt;- map2_dfc(total_cpm_filtered, norm_factors, `*`)
total_cpm_norm &lt;- as.data.frame(total_cpm_norm)
rownames(total_cpm_norm) &lt;- rownames(total_cpm_filtered)
total_cpm_norm[1:5, 1:5]</code></pre>
<pre><code>                       BRCA1      BRCA2      BRCA4     BRCA5       BRCA6
ENSG00000000003.13 44.345921 97.8872495 51.5969898 41.637123 30.30620657
ENSG00000000005.5   4.914099  0.5749122  0.1122162  2.588177  0.07643432
ENSG00000000419.11 28.243318 42.9155017 15.8898081 28.710380 47.14086482
ENSG00000000457.12 35.384916 43.4904138  8.7079739 40.166246 42.78410877
ENSG00000000460.15 11.630602 21.8466620  2.0872206 15.868496 25.45262746</code></pre>
</div>
<div id="log-transformation" class="section level4">
<h4>Log transformation</h4>
<pre class="r"><code>total_log_cpm_filtered_norm &lt;- log2(total_cpm_norm)</code></pre>
</div>
<div id="review-representative-density-plots" class="section level4">
<h4>Review representative density plots</h4>
<pre class="r"><code># Density plots
# tidy data
final_total_df &lt;- total_log_cpm_filtered_norm %&gt;%
  rownames_to_column(var = &quot;gene_id&quot;) %&gt;%
  dplyr::select(gene_id, everything())

tidy_norm_cpm &lt;- final_total_df %&gt;% 
  gather(key = &quot;sampleID&quot;, value = &quot;intensity&quot;, -gene_id) 

tidy_norm_cpm &lt;- tidy_norm_cpm %&gt;%
  mutate(group = str_sub(sampleID, 1, 4))

# representative plots
brca_norm_cpm &lt;- tidy_norm_cpm %&gt;%
  dplyr::filter(group == &quot;BRCA&quot;)
dplot_norm_brca &lt;- ggplot(brca_norm_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;)
dplot_norm_brca</code></pre>
<pre><code>Warning: Removed 1082325 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/ss_gsea.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/7215e3842c79010d7f93a6388421ef788f345f77/docs/figure/ss_gsea.Rmd/unnamed-chunk-14-1.png" target="_blank">7215e38</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8d832f2ae5facbd32d81de3a667b61e9c599a25d/docs/figure/ss_gsea.Rmd/unnamed-chunk-14-1.png" target="_blank">8d832f2</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>gbm_norm_cpm &lt;- tidy_norm_cpm %&gt;%
  dplyr::filter(grepl(&quot;GBM*&quot;, group))
dplot_norm_gbm &lt;- ggplot(gbm_norm_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = &quot;none&quot;) +
  xlim(-6, 15)
dplot_norm_gbm</code></pre>
<pre><code>Warning: Removed 147916 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/ss_gsea.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-2">
Past versions of unnamed-chunk-14-2.png
</button>
</p>
<div id="fig-unnamed-chunk-14-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/7215e3842c79010d7f93a6388421ef788f345f77/docs/figure/ss_gsea.Rmd/unnamed-chunk-14-2.png" target="_blank">7215e38</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-15
</td>
</tr>
<tr>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8d832f2ae5facbd32d81de3a667b61e9c599a25d/docs/figure/ss_gsea.Rmd/unnamed-chunk-14-2.png" target="_blank">8d832f2</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="gsva" class="section level2">
<h2>GSVA</h2>
<p>Will be done on cluster due to extreme system requirements Write out matrix and gene set lists for import into Gardner HPC cluster.</p>
<pre class="r"><code># gset list
saveRDS(gset_ids_complete, &quot;output/gset_ids_complete.rds&quot;)

expr_matrix &lt;- total_log_cpm_filtered_norm %&gt;% 
  as.data.frame() %&gt;%
  rownames_to_column(var = &quot;gene&quot;)

write_csv(expr_matrix, &quot;output/expr_matrix.csv&quot;)</code></pre>
<p>Get macrophage gene set GSVA scores</p>
<pre class="r"><code># Run on cluster
# tcga_es &lt;- gsva(expr_matrix, gset_ids_complete,
#                annotation = NULL,
#                method = &quot;gsva&quot;,
#                mx.diff = FALSE,
#                verbose = TRUE)

# saveRDS(tcga_es, &quot;/gpfs/data/kline-lab/tcga_macs/tcga_es.rds&quot;)
# moved to local hard drive from cluster and stored in output folder</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.3 (2019-03-11)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Mojave 10.14.4

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] GenomicDataCommons_1.6.0 magrittr_1.5            
 [3] GSVA_1.30.0              edgeR_3.24.3            
 [5] limma_3.38.3             forcats_0.4.0           
 [7] stringr_1.4.0            dplyr_0.8.3             
 [9] purrr_0.3.3              readr_1.3.1             
[11] tidyr_1.0.0              tibble_2.1.3            
[13] ggplot2_3.2.1            tidyverse_1.2.1         

loaded via a namespace (and not attached):
 [1] nlme_3.1-141                matrixStats_0.55.0         
 [3] bitops_1.0-6                fs_1.3.1                   
 [5] lubridate_1.7.4             bit64_0.9-7                
 [7] RColorBrewer_1.1-2          httr_1.4.1                 
 [9] GenomeInfoDb_1.18.2         rprojroot_1.3-2            
[11] tools_3.5.3                 backports_1.1.5            
[13] R6_2.4.0                    DBI_1.0.0                  
[15] lazyeval_0.2.2              BiocGenerics_0.28.0        
[17] colorspace_1.4-1            withr_2.1.2                
[19] tidyselect_0.2.5            bit_1.1-14                 
[21] compiler_3.5.3              git2r_0.26.1               
[23] graph_1.60.0                cli_1.1.0                  
[25] rvest_0.3.4                 Biobase_2.42.0             
[27] xml2_1.2.2                  DelayedArray_0.8.0         
[29] labeling_0.3                scales_1.0.0               
[31] rappdirs_0.3.1              digest_0.6.21              
[33] rmarkdown_1.16              XVector_0.22.0             
[35] pkgconfig_2.0.3             htmltools_0.4.0            
[37] fastmap_1.0.1               rlang_0.4.0                
[39] readxl_1.3.1                rstudioapi_0.10            
[41] RSQLite_2.1.2               shiny_1.4.0                
[43] generics_0.0.2              jsonlite_1.6               
[45] BiocParallel_1.16.6         RCurl_1.95-4.12            
[47] GenomeInfoDbData_1.2.0      Matrix_1.2-17              
[49] Rcpp_1.0.2                  munsell_0.5.0              
[51] S4Vectors_0.20.1            lifecycle_0.1.0            
[53] stringi_1.4.3               whisker_0.4                
[55] yaml_2.2.0                  SummarizedExperiment_1.12.0
[57] zlibbioc_1.28.0             grid_3.5.3                 
[59] blob_1.2.0                  parallel_3.5.3             
[61] promises_1.1.0              crayon_1.3.4               
[63] lattice_0.20-38             haven_2.1.1                
[65] annotate_1.60.1             hms_0.5.2                  
[67] locfit_1.5-9.1              zeallot_0.1.0              
[69] knitr_1.25                  pillar_1.4.2               
[71] GenomicRanges_1.34.0        geneplotter_1.60.0         
[73] stats4_3.5.3                XML_3.98-1.20              
[75] glue_1.3.1                  evaluate_0.14              
[77] modelr_0.1.5                vctrs_0.2.0                
[79] httpuv_1.5.2                cellranger_1.1.0           
[81] gtable_0.3.0                assertthat_0.2.1           
[83] xfun_0.10                   mime_0.7                   
[85] xtable_1.8-4                broom_0.5.2                
[87] later_1.0.0                 shinythemes_1.1.2          
[89] AnnotationDbi_1.44.0        memoise_1.1.0              
[91] IRanges_2.16.0              workflowr_1.5.0            
[93] ellipsis_0.3.0              GSEABase_1.44.0            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
