<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="mleukam" />

<meta name="date" content="2019-06-08" />

<title>Download and Format TCGA Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Macrophage signatures in TCGA</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mleukam/tcga_macs">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Download and Format TCGA Data</h1>
<h4 class="author">mleukam</h4>
<h4 class="date">2019-06-08</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2019-06-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>tcga_macs/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.4.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20190409code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20190409)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20190409code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20190409)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/tcga_macs/data/htseq_counts/
</td>
<td style="text-align:left;">
data/htseq_counts
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstree30e1cfea263652b2e5df99fbc1a6a2817ba560e0targetblank30e1cfea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mleukam/tcga_macs/tree/30e1cfea263652b2e5df99fbc1a6a2817ba560e0" target="_blank">30e1cfe</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommleukamtcgamacstree30e1cfea263652b2e5df99fbc1a6a2817ba560e0targetblank30e1cfea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/GSE46903_Affymetrix_Reanalyzed_data.txt.gz
    Ignored:    data/GSE46903_Affymetrix_metadata.txt.gz
    Ignored:    data/GSE46903_RAW.tar
    Ignored:    data/GSE46903_non-normalized.txt
    Ignored:    data/GSE46903_nonnormalized copy.tsv
    Ignored:    data/gse_46903.rds
    Ignored:    output/dge_frame.csv
    Ignored:    output/expression_set.rds
    Ignored:    output/mac_eset.rds
    Ignored:    output/mac_eset_w_clusters.rds
    Ignored:    output/normalized_probe_intensities.Rds
    Ignored:    output/processed_lumi_46903.Rds
    Ignored:    output/processed_lumi_46903.tsv
    Ignored:    output/processed_lumi_exprs.txt
    Ignored:    output/tcga_counts_dataframe_list.rds
    Ignored:    output/tcga_gdc_manifest.rds

Untracked files:
    Untracked:  analysis/ss_gsea.Rmd
    Untracked:  data/htseq_counts/
    Untracked:  output/PCA_with_km7cluster_labels.pdf
    Untracked:  output/downreg_c1.csv
    Untracked:  output/downreg_c2.csv
    Untracked:  output/downreg_c3.csv
    Untracked:  output/downreg_c4.csv
    Untracked:  output/downreg_c5.csv
    Untracked:  output/downreg_c6.csv
    Untracked:  output/downreg_c7.csv
    Untracked:  output/km_cluster_assignments.csv
    Untracked:  output/table_c1.csv
    Untracked:  output/table_c2.csv
    Untracked:  output/table_c3.csv
    Untracked:  output/table_c4.csv
    Untracked:  output/table_c5.csv
    Untracked:  output/table_c6.csv
    Untracked:  output/table_c7.csv
    Untracked:  output/upreg_c1.csv
    Untracked:  output/upreg_c2.csv
    Untracked:  output/upreg_c3.csv
    Untracked:  output/upreg_c4.csv
    Untracked:  output/upreg_c5.csv
    Untracked:  output/upreg_c6.csv
    Untracked:  output/upreg_c7.csv

Unstaged changes:
    Modified:   analysis/invitro_mac_dge.Rmd
    Modified:   analysis/preprocessing.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the R Markdown and HTML files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view them.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/30e1cfea263652b2e5df99fbc1a6a2817ba560e0/analysis/tcga_download.Rmd" target="_blank">30e1cfe</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
completed tcga download notebook - ending with list of dataframes of
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/25f3d7f6bff1eede0e29ec56e06bf892cb8647b2/docs/tcga_download.html" target="_blank">25f3d7f</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/146b5f7c75db1f93b5889ae5bf304120d334c674/analysis/tcga_download.Rmd" target="_blank">146b5f7</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
completed tcga download notebook - ending with list of dataframes of
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/d264e8ca5b9bbdf402a8810dc721c3fcf1bce6d3/docs/tcga_download.html" target="_blank">d264e8c</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/8358b4e6dc2836ea571a525b148d04db9fda9acf/analysis/tcga_download.Rmd" target="_blank">8358b4e</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-10
</td>
<td>
completed tcga download notebook - ending with list of dataframes of
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mleukam/tcga_macs/784e9c649e573a7c26df64f6d11454efce2d64ed/docs/tcga_download.html" target="_blank">784e9c6</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mleukam/tcga_macs/blob/89b0b276c99fbae30bfbb355c6a3c7a1cf864c1e/analysis/tcga_download.Rmd" target="_blank">89b0b27</a>
</td>
<td>
mleukam
</td>
<td>
2019-06-09
</td>
<td>
working download with all files
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>All of this is recorded for completeness - this notebook is not intended to be run more than once and the global options for evaluation are set to FALSE. Once this notebook was completed, results were stored in a list of dataframes saved with the saveRDS function.</p>
<pre class="r"><code>knitr::opts_chunk$set(eval=FALSE)</code></pre>
<pre class="r"><code>library(&quot;GenomicDataCommons&quot;)
library(&quot;tidyverse&quot;)</code></pre>
<div id="test-connection" class="section level4">
<h4>Test connection</h4>
<pre class="r"><code>GenomicDataCommons::status()

# how many cases available in TCGA
cases() %&gt;% 
  GenomicDataCommons::filter(~ project.program.name==&#39;TCGA&#39;) %&gt;%
  facet(&#39;samples.sample_type&#39;) %&gt;% 
  aggregations()</code></pre>
</div>
</div>
<div id="download-files" class="section level2">
<h2>Download files</h2>
<div id="get-manifest" class="section level4">
<h4>Get manifest</h4>
<pre class="r"><code># function to get manifest id&#39;s for HTSeq raw counts
getrawcounts &lt;- function(i){
  manifest &lt;- files() %&gt;%
    GenomicDataCommons::select(available_fields(&#39;files&#39;)) %&gt;%
    GenomicDataCommons::filter(~ cases.project.project_id == i &amp;
             data_type==&#39;Gene Expression Quantification&#39; &amp;
             analysis.workflow_type == &#39;HTSeq - Counts&#39;) %&gt;%
    GenomicDataCommons::select(&#39;file_id&#39;) %&gt;%
    manifest()
  manifest
}

# list of TGCA projects
tcgalist &lt;- list(&quot;BRCA&quot; = &quot;TCGA-BRCA&quot;,
                 &quot;GBM&quot; = &quot;TCGA-GBM&quot;,
                 &quot;OV&quot; = &quot;TCGA-OV&quot;,
                 &quot;LUAD&quot; = &quot;TCGA-LUAD&quot;,
                 &quot;UCEC&quot; = &quot;TCGA-UCEC&quot;,
                 &quot;KIRC&quot; = &quot;TCGA-KIRC&quot;,
                 &quot;HNSC&quot; = &quot;TCGA-HNSC&quot;,
                 &quot;LGG&quot; = &quot;TCGA-LGG&quot;,
                 &quot;THCA&quot; = &quot;TCGA-THCA&quot;,
                 &quot;LUSC&quot; = &quot;TCGA-LUSC&quot;,
                 &quot;PRAD&quot; = &quot;TCGA-PRAD&quot;,
                 &quot;SKCM&quot; = &quot;TCGA-SKCM&quot;,
                 &quot;COAD&quot; = &quot;TCGA-COAD&quot;,
                 &quot;STAD&quot; = &quot;TCGA-STAD&quot;,
                 &quot;BLCA&quot; = &quot;TCGA-BLCA&quot;,
                 &quot;LIHC&quot; = &quot;TCGA-LIHC&quot;,
                 &quot;CESC&quot; = &quot;TCGA-CESC&quot;,
                 &quot;KIRP&quot; = &quot;TCGA-KIRP&quot;,
                 &quot;SARC&quot; = &quot;TCGA-SARC&quot;,
                 &quot;LAML&quot; = &quot;TCGA-LAML&quot;,
                 &quot;ESCA&quot; = &quot;TCGA-ESCA&quot;,
                 &quot;PAAD&quot; = &quot;TCGA-PAAD&quot;,
                 &quot;PCPG&quot; = &quot;TCGA-PCPG&quot;,
                 &quot;READ&quot; = &quot;TCGA-READ&quot;,
                 &quot;TGCT&quot; = &quot;TCGA-TGCT&quot;,
                 &quot;THYM&quot; = &quot;TCGA-THYM&quot;,
                 &quot;KICH&quot; = &quot;TCGA-KICH&quot;,
                 &quot;ACC&quot; = &quot;TCGA-ACC&quot;,
                 &quot;MESO&quot; = &quot;TCGA-MESO&quot;,
                 &quot;UVM&quot; = &quot;TCGA-UVM&quot;,
                 &quot;DLBC&quot; = &quot;TCGA-DLBC&quot;,
                 &quot;UCS&quot; =  &quot;TCGA-UCS&quot;,
                 &quot;CHOL&quot; = &quot;TCGA-CHOL&quot;)

# apply function to list elements
tcga_countfile_manifest &lt;- map(tcgalist, getrawcounts)
str(tcga_countfile_manifest)

# save for later use
saveRDS(tcga_countfile_manifest, &quot;output/tcga_gdc_manifest.rds&quot;)

# combined into a single table
combined_manifest &lt;- bind_rows(tcga_countfile_manifest, .id = &quot;column_label&quot;)

#----------------------------
# select filenames
sel_fnames &lt;- function(df){
  df %&gt;% dplyr::select(filename)
}
# apply to list
all_fnames &lt;- map(tcga_countfile_manifest, sel_fnames)
# create vector
all_fnames_vector &lt;- unlist(all_fnames)
attributes(all_fnames_vector) &lt;- NULL
#----------------------------
# select ids
sel_ids &lt;- function(df){
  df %&gt;% dplyr::select(id)
}
# apply to list
all_ids &lt;- map(tcga_countfile_manifest, sel_ids)
# create vector
all_id_vector &lt;- unlist(all_ids)
attributes(all_id_vector) &lt;- NULL</code></pre>
</div>
<div id="download-data" class="section level4">
<h4>Download data</h4>
<pre class="r"><code># provide IDs for download
library(BiocParallel)
register(MulticoreParam())
destdir = tempdir()
fnames = lapply(all_id_vector, gdcdata)</code></pre>
<pre class="bash"><code>## done in command line, stored here for documentation only - not part of notebook
## get files out of subfolders and move to parent
# find . -mindepth 2 -type f -print -exec mv {} . \;

## create a folder for counts
# mkdir htseq_counts

## move all sequence files into a folder
# find . -name &#39;*.htseq.counts.gz&#39; -exec mv {} ~/Library/Caches/GenomicDataCommons/htseq_counts \;

## unzip
# find . -name &#39;*.htseq.counts.gz&#39; -exec gunzip {} \;

## move to local repo data folder
# find . -name &#39;*.htseq.counts&#39; -exec mv {} ~/tcga_macs/data/htseq_counts \;
</code></pre>
<p>Check for missing files</p>
<pre class="r"><code># get filenames that were downloaded
downloaded_fnames &lt;- list.files(&quot;~/tcga_macs/data/htseq_counts/&quot;, all.files = FALSE) 

all_fnames_v &lt;- all_fnames_vector %&gt;% str_replace(&quot;.gz&quot;, &quot;&quot;)

# find the difference
missing &lt;- setdiff(all_fnames_v, downloaded_fnames)

# get ids for missing filenames
combined_man_tbl &lt;- combined_manifest %&gt;%
  mutate(filename = str_replace(filename, &quot;.gz&quot;,&quot;&quot;))

missing_ids &lt;- combined_man_tbl %&gt;%
  filter(filename %in% missing) %&gt;%
  pull(id)</code></pre>
<p>About 1000 files were missing</p>
</div>
<div id="download-missing-files" class="section level4">
<h4>Download missing files</h4>
<pre class="r"><code># provide IDs for download
library(BiocParallel)
register(MulticoreParam())
destdir = tempdir()
fnames = lapply(missing_ids, gdcdata)</code></pre>
<p>Unzipped and moved in command line similar to above</p>
<p>Check for missing files again</p>
<pre class="r"><code># get filenames that were downloaded
downloaded_fnames &lt;- list.files(&quot;data/htseq_counts/&quot;, all.files = FALSE) 

all_fnames_v &lt;- all_fnames_vector %&gt;% str_replace(&quot;.gz&quot;, &quot;&quot;)

# find the difference
missing &lt;- setdiff(all_fnames_v, downloaded_fnames)

# get ids for missing filenames
combined_man_tbl &lt;- combined_manifest %&gt;%
  mutate(filename = str_replace(filename, &quot;.gz&quot;,&quot;&quot;))

missing_ids &lt;- combined_man_tbl %&gt;%
  filter(filename %in% missing) %&gt;%
  pull(id)</code></pre>
<p>No further missing files</p>
</div>
</div>
<div id="combine-reads-into-single-dataframes-by-cancer-type" class="section level2">
<h2>Combine reads into single dataframes by cancer type</h2>
<p>Load data</p>
<pre class="r"><code>tcga_manifest &lt;- readRDS(&quot;output/tcga_gdc_manifest.rds&quot;)</code></pre>
<p>Combine files into dataframes</p>
<pre class="r"><code>filepaths &lt;- map(tcga_manifest, function(df){
  df %&gt;% dplyr::select(filename) %&gt;% 
    mutate(filename = (str_replace(filename, &quot;.gz&quot;, &quot;&quot;))) %&gt;%
    mutate(filepath = (str_c(&quot;data/htseq_counts/&quot;, filename)))
           })</code></pre>
<pre class="r"><code># read in BRCA files
BRCA_paths &lt;- filepaths$BRCA %&gt;% pull(filepath)

BRCA_df &lt;- map(BRCA_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$BRCA %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(BRCA_df) &lt;- cols
BRCA_df</code></pre>
<pre class="r"><code># read in GBM files
GBM_paths &lt;- filepaths$GBM %&gt;% pull(filepath)

GBM_df &lt;- map(GBM_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$GBM %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(GBM_df) &lt;- cols
GBM_df</code></pre>
<pre class="r"><code># read in OV files
OV_paths &lt;- filepaths$OV %&gt;% pull(filepath)

OV_df &lt;- map(OV_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$OV %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(OV_df) &lt;- cols
OV_df</code></pre>
<pre class="r"><code># read in LUAD files
LUAD_paths &lt;- filepaths$LUAD %&gt;% pull(filepath)

LUAD_df &lt;- map(LUAD_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$LUAD %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(LUAD_df) &lt;- cols
LUAD_df</code></pre>
<pre class="r"><code># read in UCEC files
UCEC_paths &lt;- filepaths$UCEC %&gt;% pull(filepath)

UCEC_df &lt;- map(UCEC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$UCEC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(UCEC_df) &lt;- cols
UCEC_df</code></pre>
<pre class="r"><code># read in KIRC files
KIRC_paths &lt;- filepaths$KIRC %&gt;% pull(filepath)

KIRC_df &lt;- map(KIRC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$KIRC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(KIRC_df) &lt;- cols
KIRC_df</code></pre>
<pre class="r"><code># read in HNSC files
HNSC_paths &lt;- filepaths$HNSC %&gt;% pull(filepath)

HNSC_df &lt;- map(HNSC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$HNSC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(HNSC_df) &lt;- cols
HNSC_df</code></pre>
<pre class="r"><code># read in LGG files
LGG_paths &lt;- filepaths$LGG %&gt;% pull(filepath)

LGG_df &lt;- map(LGG_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$LGG %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(LGG_df) &lt;- cols
LGG_df</code></pre>
<pre class="r"><code># read in THCA files
THCA_paths &lt;- filepaths$THCA %&gt;% pull(filepath)

THCA_df &lt;- map(THCA_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$THCA %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(THCA_df) &lt;- cols
THCA_df</code></pre>
<pre class="r"><code># read in LUSC files
LUSC_paths &lt;- filepaths$LUSC %&gt;% pull(filepath)

LUSC_df &lt;- map(LUSC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$LUSC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(LUSC_df) &lt;- cols
LUSC_df</code></pre>
<pre class="r"><code># read in PRAD files
PRAD_paths &lt;- filepaths$PRAD %&gt;% pull(filepath)

PRAD_df &lt;- map(PRAD_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$PRAD %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(PRAD_df) &lt;- cols
PRAD_df</code></pre>
<pre class="r"><code># read in SKCM files
SKCM_paths &lt;- filepaths$SKCM %&gt;% pull(filepath)

SKCM_df &lt;- map(SKCM_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$SKCM %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(SKCM_df) &lt;- cols
SKCM_df</code></pre>
<pre class="r"><code># read in COAD files
COAD_paths &lt;- filepaths$COAD %&gt;% pull(filepath)

COAD_df &lt;- map(COAD_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$COAD %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(COAD_df) &lt;- cols
COAD_df</code></pre>
<pre class="r"><code># read in STAD files
STAD_paths &lt;- filepaths$STAD %&gt;% pull(filepath)

STAD_df &lt;- map(STAD_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$STAD %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(STAD_df) &lt;- cols
STAD_df</code></pre>
<pre class="r"><code># read in BLCA files
BLCA_paths &lt;- filepaths$BLCA %&gt;% pull(filepath)

BLCA_df &lt;- map(BLCA_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$BLCA %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(BLCA_df) &lt;- cols
BLCA_df</code></pre>
<pre class="r"><code># read in LIHC files
LIHC_paths &lt;- filepaths$LIHC %&gt;% pull(filepath)

LIHC_df &lt;- map(LIHC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$LIHC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(LIHC_df) &lt;- cols
LIHC_df</code></pre>
<pre class="r"><code># read in CESC files
CESC_paths &lt;- filepaths$CESC %&gt;% pull(filepath)

CESC_df &lt;- map(CESC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$CESC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(CESC_df) &lt;- cols
CESC_df</code></pre>
<pre class="r"><code># read in KIRP files
KIRP_paths &lt;- filepaths$KIRP %&gt;% pull(filepath)

KIRP_df &lt;- map(KIRP_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$KIRP %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(KIRP_df) &lt;- cols
KIRP_df</code></pre>
<pre class="r"><code># read in SARC files
SARC_paths &lt;- filepaths$SARC %&gt;% pull(filepath)

SARC_df &lt;- map(SARC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$SARC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(SARC_df) &lt;- cols
SARC_df</code></pre>
<pre class="r"><code># read in LAML files
LAML_paths &lt;- filepaths$LAML %&gt;% pull(filepath)

LAML_df &lt;- map(LAML_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$LAML %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(LAML_df) &lt;- cols
LAML_df</code></pre>
<pre class="r"><code># read in ESCA files
ESCA_paths &lt;- filepaths$ESCA %&gt;% pull(filepath)

ESCA_df &lt;- map(ESCA_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$ESCA %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(ESCA_df) &lt;- cols
ESCA_df</code></pre>
<pre class="r"><code># read in PAAD files
PAAD_paths &lt;- filepaths$PAAD %&gt;% pull(filepath)

PAAD_df &lt;- map(PAAD_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$PAAD %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(PAAD_df) &lt;- cols
PAAD_df</code></pre>
<pre class="r"><code># read in PCPG files
PCPG_paths &lt;- filepaths$PCPG %&gt;% pull(filepath)

PCPG_df &lt;- map(PCPG_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$PCPG %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(PCPG_df) &lt;- cols
PCPG_df</code></pre>
<pre class="r"><code># read in READ files
READ_paths &lt;- filepaths$READ %&gt;% pull(filepath)

READ_df &lt;- map(READ_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$READ %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(READ_df) &lt;- cols
READ_df</code></pre>
<pre class="r"><code># read in TGCT files
TGCT_paths &lt;- filepaths$TGCT %&gt;% pull(filepath)

TGCT_df &lt;- map(TGCT_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$TGCT %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(TGCT_df) &lt;- cols
TGCT_df</code></pre>
<pre class="r"><code># read in THYM files
THYM_paths &lt;- filepaths$THYM %&gt;% pull(filepath)

THYM_df &lt;- map(THYM_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$THYM %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(THYM_df) &lt;- cols
THYM_df</code></pre>
<pre class="r"><code># read in KICH files
KICH_paths &lt;- filepaths$KICH %&gt;% pull(filepath)

KICH_df &lt;- map(KICH_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$KICH %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(KICH_df) &lt;- cols
KICH_df</code></pre>
<pre class="r"><code># read in ACC files
ACC_paths &lt;- filepaths$ACC %&gt;% pull(filepath)

ACC_df &lt;- map(ACC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$ACC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(ACC_df) &lt;- cols
ACC_df</code></pre>
<pre class="r"><code># read in MESO files
MESO_paths &lt;- filepaths$MESO %&gt;% pull(filepath)

MESO_df &lt;- map(MESO_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$MESO %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(MESO_df) &lt;- cols
MESO_df</code></pre>
<pre class="r"><code># read in UVM files
UVM_paths &lt;- filepaths$UVM %&gt;% pull(filepath)

UVM_df &lt;- map(UVM_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$UVM %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(UVM_df) &lt;- cols
UVM_df</code></pre>
<pre class="r"><code># read in DLBC files
DLBC_paths &lt;- filepaths$DLBC %&gt;% pull(filepath)

DLBC_df &lt;- map(DLBC_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$DLBC %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(DLBC_df) &lt;- cols
DLBC_df</code></pre>
<pre class="r"><code># read in UCS files
UCS_paths &lt;- filepaths$UCS %&gt;% pull(filepath)

UCS_df &lt;- map(UCS_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$UCS %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(UCS_df) &lt;- cols
UCS_df</code></pre>
<pre class="r"><code># read in CHOL files
CHOL_paths &lt;- filepaths$CHOL %&gt;% pull(filepath)

CHOL_df &lt;- map(CHOL_paths, function(x){read_tsv(x, col_names = FALSE)}) %&gt;% 
  reduce(left_join, by = &quot;X1&quot;)

# get column names
cols &lt;- tcga_manifest$CHOL %&gt;% pull(id)
cols &lt;- c(&quot;gene&quot;, cols)
colnames(CHOL_df) &lt;- cols
CHOL_df</code></pre>
</div>
<div id="gather-up-the-dataframes" class="section level2">
<h2>Gather up the dataframes</h2>
<pre class="r"><code>count_df_list &lt;- list(BRCA_df, GBM_df, OV_df, LUAD_df, UCEC_df, KIRC_df, HNSC_df, LGG_df, THCA_df, LUSC_df, PRAD_df, SKCM_df, COAD_df, STAD_df, BLCA_df, LIHC_df, CESC_df, KIRP_df, SARC_df, LAML_df, ESCA_df, PAAD_df, PCPG_df, READ_df, TGCT_df, THYM_df, KICH_df, ACC_df, MESO_df, UVM_df, DLBC_df, UCS_df, CHOL_df)

# save list of count dataframes for later
saveRDS(count_df_list, &quot;output/tcga_counts_dataframe_list.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
