---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

Welcome to the lab notebook for our research project describing the distribution of macrophage gene signatures across various tumor types in The Cancer Genome Atlas (TCGA) data.

## Introduction

A T cell inflamed signature has been defined and described across tumor types in TCGA that corresponds to response to T cell directed immunotherapy. Is there a similar signature for macrophages that might predict the tumors most likely to respond to macrophage-directed immunotherapy? One challenge has been the diversity and plasticity of activated macrophage states compared to cytotoxic T cells. This project will use experimentally derived activation gene signatures for a range of macrophage subtypes and use high dimensional clustering to computationally define macrophage subsets. The distribution of these subsets will then be examined in multiple tumor types in TCGA data.

## Defining macrophage gene sets

Find ranked differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes. These gene sets will be used for our downstream analyses. The original data is obtained from GEO, ID=[GSE46903](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903).

 * [Download data from GEO](data_download.html).
 * Define clusters: [Principle Component Analysis and Multiple Correlation Analysis](PCA_MCA.html).
 * Get differential gene expression of each cluster compared to control M0 [Differential gene expression for experimental macrophage conditions](invitro_mac_dge.html).
