---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
---

Welcome to the lab notebook for our research project describing the distribution of macrophage gene signatures across various tumor types in The Cancer Genome Atlas (TCGA) data.

## Introduction

A T cell inflamed signature has been defined and described across tumor types in TCGA that corresponds to response to T cell directed immunotherapy. Is there a similar signature for macrophages that might predict the tumors most likely to respond to macrophage-directed immunotherapy? One challenge has been the diversity and plasticity of activated macrophage states compared to cytotoxic T cells. This project will use experimentally derived activation gene signatures for a range of macrophage subtypes and use high dimensional clustering to computationally define macrophage subsets. The distribution of these subsets will then be examined in multiple tumor types in TCGA data.

## Defining macrophage gene sets

Derived differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes for downstream analyses. The original data is obtained from GEO, ID=[GSE46903](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903).

 * [Download data from GEO](data_download.html)
 * [Data cleaning](preprocessing.html)
 * [Clustering and correlation analyses](clustering.html)
 * [Differential gene expression to derive gene lists](invitro_mac_dge.html)
 
#### Figures: 

 Dissimilarity matrix showing correlation scores among samples:
 
![](assets/corrmatrix.png)
 
K-means clustering of treatment groups based on Pearson correlation coefficients (k =7):

![](assets/k7cluster.png)

## Investigating patterns in TGCA data

Apply gene lists from above to TCGA pancancer bulk RNAseq counts data 

 * [Download counts data from Genomic Data Commons](tcga_download.html)
 * [Format counts data](tcga_format.html)
 * [Gene set variation analysis](ss_gsea.html)
 * [Exploration and analysis of GSVA results](gsva_results_exploration.html)

## Applying macrophage clusters to IMTX data

Apply gene lists from above to datasets of tumor gene expression linked to clinical response to immunotherapy in several cancer types

##### Urothelial IMTX cohort

* Urothelial cancer / TGF-b / bladder
  * Paper https://www.ncbi.nlm.nih.gov/pubmed/29443960  
  * Datasets http://research-pub.gene.com/IMvigor210CoreBiologies/ 

[Notebook for import and cleaning](urothelial.html)

##### Melanoma IMTX cohorts

* hugo
* riaz
* gide

[Notebook for import and cleaning](expanded_gsva.Rmd)

#### Results and analysis

* [Logistic model for best response](expanded_gsva_analysis.html)
* [Survival analysis](survival.html)


## Other datasets:

1. Macrophage results that Lei previously generated around CSF1R in the original macrophage project we did. 
The question we had was how your work overlaps with his work. I think it will not correlate well but this is an important point to make in his paper in that it emphasizes that CSF1R is not a good “biomarker” for the pleotropic macrophage states in cancer.
Array Update Comp Files 2018.01.11.zip
13 MB Zip — Click to download
please also keep content from this doc confidential especially the gene names 

2. IMTX single cell RNAseq cohorts (both melanoma)

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE120575 
https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE115978 

3. the CXCL9/CXCL10 gene expression signature (macrophage)

The question we had was whether this signature falls into one of your 6 macrophage clusters?
