---
title: "Differential gene expression: _in vitro_ macrophage model"
author: "mleukam"
date: "2019-04-09"
output: workflowr::wflow_html
---

## Introduction

### Purpose of this analysis

Find ranked differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes. These gene sets will be used for our downstream analyses. 

The source for this method is:

> Gubin, Matthew M., Ekaterina Esaulova, Jeffrey P. Ward, Olga N. Malkova, Daniele Runci, Pamela Wong, Takuro Noguchi, et al. 2018. “High-Dimensional Analysis Delineates Myeloid and Lymphoid Compartment Remodeling during Successful Immune-Checkpoint Cancer Therapy.” Cell 175 (4): 1014–30.e19. 

This is a summary of the method from Gubin et al.:

> Expression data was retrieved from GEO with GSE46903 ID. For analysis, we selected only monocyte and macrophage samples. According to the dataset description, peripheral blood monocytes were stimulated with GM-CSF or M-CSF, and subsequently stimulated with different compounds. We excluded all M-CSF-derived macrophages and non-72 hour time points. M0-GM-CSF 72 hour samples were used as baseline. We used limma package to perform differential expression in order to understand gene changes upon treatments compared to baseline. For each comparison, gene lists were obtained and ranked by t-statistics. To compare human macrophages under different stimuli with mouse monocytes/macrophages from our scRNAseq data, we compared genes, upregulated or downregulated inside each scRNAseq cluster with genes, upregulated or downregulated in each human monocyte/macrophage treatment condition. The goal was to get a list of signed scores that will represent the level of concordance between treatments and each cluster.

> For each cluster, we created two gene sets: 100 upregulated and downregulated genes. GSEA (with fgsea package) was performed using this gene set against ranked gene lists from treatment comparisons. To obtain a score, we subtracted the normalized enrichment score (NES) of downregulated genes from NES of upregulated genes. To further investigate the correlation (or lack thereof) between clusters and treatments, we visualized genes associated with each cluster in human macrophages. We reduced the dimensions of macrophage datasets with Rtsne package and plotted each sample in two dimensions using the ggplot2 package. Samples were then highlighted using Z-scored expression profiles of cluster-associated genes.

### Data Source

The gene chip data used to derive differential expression for various macrophage subtypes was first published here:

> Xue J, Schmidt SV, Sander J, Draffehn A et al. Transcriptome-based network analysis reveals a spectrum model of human macrophage activation. Immunity 2014 Feb 20;40(2):274-88. PMID: 24530056

The original experiment had two conditions. 

1. Cell culture with monocytes exposed to growth factor and a number of different cytokines to induce different activation states. Microarray expression data from cell culture were generated with Illumina HumanHT-12 v3.0 beadchips.

2. Gene sets derived in cell culture were used to categorize expression profiles derived from alveolar macrophages from patients with COPD and asthma. Affymetrix chips were used to measure expression in human samples.

We are interested in the first condition (macrophages in culture). Associated metadata is posted at the GEO:

* [The experiment overview](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903)

* [The record for the control group](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1140443) 

    * M0-GM-CSF treated monocytes at 72hours
    
    * Additional details about how the data were generated

* Expression matrices in which “The data were log2-transformed and normalized using quantile normalization with Partek Genomics Suite” 

  * [Example expression matrix](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?view=data&acc=GSM1140443&id=38019&db=GeoDb_blob97) from the control group

_We will obtain the raw data from the Illumina expression beadchips and normalize ourselves._

### Set up workspace
Clear workspace prior to analysis and load necessary packages
```{r}
rm(list=ls())
```

```{r packages, message=FALSE, warning=FALSE}
library("tidyverse")
library("GEOquery")
library("lumi")
```

## Data import
GEO ID is: [GSE46903](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE46903)

```{r}
# download normalized data with full metadata from GEO as a list of ExpressionSets
gse46903 <- getGEO("GSE46903")
```

Accessing raw data from the GEO: http://genomicsclass.github.io/book/pages/GEOquery.html

Note about raw vs normalized files from GEO: https://www.biostars.org/p/259637/


```{r raw_download}
# download raw data
filePaths = getGEOSuppFiles("GSE46903", 
                            makeDirectory = TRUE)

```

```{bash}
# in terminal
pwd
ls
cd GSE46903
ls
mv GSE* ~/tcga_macs/data
cd ..
rmdir GSE46903
ls
```

```{r}
lumi_input <- lumiR("~/tcga_macs/data/GSE46903_non-normalized.txt.gz") 
lumi_46903 <- lumiExpresso(lumi_input, 
                         normalize.param = list(method = 'rsn'))

write.exprs(lumi_46903, file = "~/tcga_macs/output/processed_lumi_46903.txt")

```

```{r paged.print=TRUE}
# review metadata
show(lumi_46903)
raw_metadata <- pData(phenoData(lumi_46903))
head(raw_metadata)
```

The raw data is labeled with alternate sample names (a 10-digit number followed by underscore and a letter) compared to the GEO records and the sample name in the processed data GSE ExpressionSet object. 

* Check phenotype records in GSE object to see if I can merge with raw data using 10-digit letter name as a key

* Documentation for manipulating ExpressionSet objects in general [here](https://www.bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf)

* Documentation for maniupulating ExpressionSets created by GEOquery [here](https://www.bioconductor.org/packages/devel/bioc/vignettes/GEOquery/inst/doc/GEOquery.html#getting-gse-series-matrix-files-as-an-expressionset)

```{r paged.print=TRUE}
show(pData(phenoData(gse46903[[1]]))[1:20,c(1,6,8)])
```

The 10-digit sample names correspond to experimental conditions and are appended to the phenodata sample names! I will have to get each component ready and then assemble a new expressionset with complete metadata and raw data but this is do-able.

* Get data and annotations out of AnnotatedDataFrame object, documentation [here](https://www.rdocumentation.org/packages/Biobase/versions/2.32.0/topics/AnnotatedDataFrame)

```{r}
# pull phenoData from expression set as an AnnotatedDataFrame
meta46903 <- phenoData(gse46903[[1]])

# get data and annotations as separate objects
meta_data46903 <- pData(meta46903)
anno_data46903 <- varMetadata(meta46903)

# doublecheck structure
str(meta_data46903)
head(meta_data46903)

str(anno_data46903)
head(anno_data46903)

# anno_data is pretty useless - just empty variables with matching rownames from pData
```

