---
title: "Differential gene expression: _in vitro_ macrophage model"
author: "mleukam"
date: "2019-04-09"
output: workflowr::wflow_html
---

## Setup
Clear workspace prior to analysis and load necessary packages

```{r clear workspace}
rm(list = ls())
```

## Load Packages
```{r packages, message=FALSE, warning=FALSE}
library("tidyverse")
library("GEOquery")
library("lumi")
library("lumiHumanIDMapping")
library("lumiHumanAll.db")
library("limma")
library("annotate")
library("factoextra")
library("FactoMineR")
library("ggsci")
library("ggpubr")
```

## Read data
```{r read data}
# Expression set created in PCA_MCA notebook 
# Contains normalized expression matrix made from raw illumina beadchip data
# Also contains metadata from GEO
# Probe ID is unique sequence ID from lumiHumanIDMapping
mac_eset <- readRDS("~/tcga_macs/output/expression_set.rds")

# Expression set created by lumi
# Contains raw reads from illumina beadchip
# variance stablized by log2 transformation
# normalized with RSN (robust spline normalization)
lumi_46903 <- readRDS("~/tcga_macs/output/normalized_probe_intensities.Rds")
```

## Filter

```{r filter}
# retreive normalized data
dataMatrix <- exprs(mac_eset)

# remove the probes that are not detected in raw data
presentCount <- detectionCall(lumi_46903)
selDataMatrix <- dataMatrix[presentCount > 0,]
probeList <- rownames(selDataMatrix)

```

## Define Groups
```{r}
# Get phenotype data from expressionset
pheno_data <- pData(mac_eset)
str(pheno_data)

# OK if rownames are dropped, column is duplicated as "raw_id"
pheno_data <- pheno_data %>%
  as_tibble() %>%
# clean up experimental group data
  mutate(treatment = str_sub(cell_type, 9, 25),
         treatment = str_replace(treatment, "_72h", ""),
         treatment = ifelse(`activation stimuli:ch1` == "con", "control", treatment),
         treatment = trimws(treatment)) %>%
  dplyr::select(raw_id, geo_id, treatment, everything()) %>%
  print()

# assign treatments into clusters from Xue et al PMID: 24530056
pheno_data <- pheno_data %>%
  mutate(treatment = as.factor(treatment),
         clusternum = fct_recode(treatment,
                                 "0" = "control",
                                 "1" = "IFNb",
                                 "1" = "IL10",
                                 "1" = "GC",
                                 "2" = "IL4",
                                 "2" = "L4",
                                 "2" = "IL13",
                                 "2" = "IL4_upLPS",
                                 "2" = "L4_upLPS",
                                 "3" = "upLPS_IC",
                                 "3" = "upLPS",
                                 "3" = "pLPS",
                                 "4" = "P3C_PGE2",
                                 "4" = "P3C",
                                 "4" = "PGE2",
                                 "6" = "IFNg",
                                 "6" = "TNFa",
                                 "6" = "IFNg_TNFa",
                                 "7" = "sLPS",
                                 "7" = "sLPS_IFNg",
                                 "7" = "sLPS_IC",
                                 "8" = "TNFa_PGE2",
                                 "8" = "TNFa_P3C",
                                 "9" = "TPP",
                                 "9" = "TPP_IFNb",
                                 "9" = "TPP_IFNb_IFNg"),
         treatment = fct_recode(treatment, 
                                "IL4_upLPS" = "L4_upLPS",
                                "upLPS" = "pLPS",
                                "IL4" = "L4")) %>%
  dplyr::select(raw_id, geo_id, treatment, clusternum, everything()) %>%
  print()

summary(pheno_data$treatment)
summary(pheno_data$clusternum)
```

## Limma

Design matrix and contrast matrix for individual treatments

```{r}
# set up design matrix
rnames <- pheno_data$raw_id
groups <- as.factor(pheno_data$treatment)
design <- model.matrix(~ 0 + groups)
colnames(design) <- gsub("groups", "", colnames(design))
rownames(design) <- rnames
design[1:5, 1:5]

# define contrasts
contr.matrix <- makeContrasts(GC - control,
                              IFNb - control,
                              IFNg - control,
                              IFNg_TNFa - control,
                              IL10 - control,
                              IL13 - control,
                              IL4 - control,
                              IL4_upLPS - control,
                              P3C - control,
                              P3C_PGE2 - control,
                              PGE2 - control,
                              upLPS - control,
                              sLPS - control,
                              sLPS_IC - control,
                              sLPS_IFNg - control,
                              TNFa - control,
                              TNFa_P3C - control,
                              TNFa_PGE2 - control,
                              TPP - control,
                              TPP_IFNb - control,
                              TPP_IFNb_IFNg - control,
                              upLPS_IC - control,
                              levels = design)
contr.matrix[1:5, 1:5]
```

Fit linear models with limma
```{r}
# fit linear models
fit <- lmFit(selDataMatrix, design)
fit2 <- contrasts.fit(fit, contr.matrix)
fit2 <- eBayes(fit2)

# add gene names
geneSymbol <- getSYMBOL(probeList, 'lumiHumanAll.db')
geneName <- sapply(lookUp(probeList, 'lumiHumanAll.db', 'GENENAME'),
                   function(x) x[1])
fit2$genes <- data.frame(ID = probeList, 
                         geneSymbol = geneSymbol, 
                         geneName = geneName, 
                         stringsAsFactors = FALSE)

# store matrix of significant results
results <- decideTests(fit2)

# collect results
table_GC        <- topTable(fit2, coef = 1, 
                            adjust.method = "BH", 
                            number = 500)
table_IFNb      <- topTable(fit2, 
                            coef = 2, 
                            adjust.method = "BH",
                            number = 500)
table_IFNg      <- topTable(fit2, 
                            coef = 3, 
                            adjust.method = "BH",
                            number = 500)
table_IFNg_TNFa <- topTable(fit2, 
                            coef = 4, 
                            adjust.method = "BH",
                            number = 500)
table_IL10      <- topTable(fit2, 
                            coef = 5, 
                            adjust.method = "BH",
                            number = 500)
table_IL13      <- topTable(fit2, 
                            coef = 6, 
                            adjust.method = "BH",
                            number = 500)
table_IL4       <- topTable(fit2, 
                            coef = 7, 
                            adjust.method = "BH",
                            number = 500)
table_IL4_upLPS <- topTable(fit2, 
                            coef = 8, 
                            adjust.method = "BH",
                            number = 500)
table_P3C       <- topTable(fit2, 
                            coef = 9, 
                            adjust.method = "BH",
                            number = 500)
table_P3C_PGE2  <- topTable(fit2, 
                            coef = 10, 
                            adjust.method = "BH",
                            number = 500)
table_PGE2      <- topTable(fit2, 
                            coef = 11, 
                            adjust.method = "BH",
                            number = 500)
table_upLPS     <- topTable(fit2, 
                            coef = 12, 
                            adjust.method = "BH",
                            number = 500)
table_sLPS      <- topTable(fit2, 
                            coef = 13, 
                            adjust.method = "BH",
                            number = 500)
table_sLPS_IC   <- topTable(fit2, 
                            coef = 14, 
                            adjust.method = "BH",
                            number = 500)
table_sLPS_IFNg <- topTable(fit2, 
                            coef = 15, 
                            adjust.method = "BH",
                            number = 500)
table_TNFa      <- topTable(fit2, 
                            coef = 16, 
                            adjust.method = "BH",
                            number = 500)
table_TNFa_P3C  <- topTable(fit2, 
                            coef = 17, 
                            adjust.method = "BH",
                            number = 500)
table_TNFa_PGE2 <- topTable(fit2, 
                            coef = 18, 
                            adjust.method = "BH",
                            number = 500)
table_TPP       <- topTable(fit2, 
                            coef = 19, 
                            adjust.method = "BH",
                            number = 500)
table_TPP_IFNb  <- topTable(fit2, 
                            coef = 20, 
                            adjust.method = "BH",
                            number = 500)
table_TPP_IFNb_IFNg <- topTable(fit2, 
                                coef = 21, 
                                adjust.method = "BH",
                                number = 500)
table_upLPS_IC  <- topTable(fit2, 
                            coef = 22, 
                            adjust.method = "BH",
                            number = 500)
```

Collecting results
```{r}
# take only significantly differentially expressed genes
# using FDR-corrected P-values
# pull out t value and gene symbol
subtable_GC <- table_GC %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, GC = t)
subtable_IFNb <- table_IFNb %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IFNb = t)
subtable_IFNg <- table_IFNg %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IFNg = t)
subtable_IFNg_TNFa <- table_IFNg_TNFa %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IFNg_TNFa = t)
subtable_IL10 <- table_IL10 %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IL10 = t)
subtable_IL13 <- table_IL13 %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IL13 = t)
subtable_IL4 <- table_IL4 %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IL4 = t)
subtable_IL4_upLPS <- table_IL4_upLPS %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, IL4_upLPS = t)
subtable_P3C <- table_P3C %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, P3C = t)
subtable_P3C_PGE2 <- table_P3C_PGE2 %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, P3C_PGE2 = t)
subtable_PGE2 <- table_PGE2 %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, PGE2 = t)
subtable_upLPS <- table_upLPS %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, upLPS = t)
subtable_sLPS <- table_sLPS %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, sLPS = t)
subtable_sLPS_IC <- table_sLPS_IC %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, sLPS_IC = t)
subtable_sLPS_IFNg <- table_sLPS_IFNg %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, sLPS_IFNg = t)
subtable_TNFa <- table_TNFa %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, TNFa = t)
subtable_TNFa_P3C <- table_TNFa_P3C %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, TNFa_P3C = t)
subtable_TNFa_PGE2 <- table_TNFa_PGE2 %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, TNFa_PGE2 = t)
subtable_TPP <- table_TPP %>% 
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, TPP = t)
subtable_TPP_IFNb <- table_TPP_IFNb %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, TPP_IFNb = t)
subtable_TPP_IFNb_IFNg <- table_TPP_IFNb_IFNg %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, TPP_IFNb_IFNg = t)
subtable_upLPS_IC <- table_upLPS_IC %>%
  filter(adj.P.Val < 0.05) %>%
  dplyr::select(ID, geneSymbol, upLPS = t)

# get  of all the probe symbols in all the lists
GC        <- pull(subtable_GC, ID) 
IFNb      <- pull(subtable_IFNb, ID)
IFNg      <- pull(subtable_IFNg, ID)
IFNg_TNFa <- pull(subtable_IFNg_TNFa, ID)
IL10      <- pull(subtable_IL10, ID)
IL13      <- pull(subtable_IL13, ID)
IL4       <- pull(subtable_IL4, ID)
IL4_upLPS <- pull(subtable_IL4_upLPS, ID)
P3C       <- pull(subtable_P3C, ID)
P3C_PGE2  <- pull(subtable_P3C_PGE2, ID)
PGE2      <- pull(subtable_PGE2, ID)
upLPS     <- pull(subtable_upLPS, ID)
sLPS      <- pull(subtable_sLPS, ID)
sLPS_IC   <- pull(subtable_sLPS_IC, ID)
sLPS_IFNg <- pull(subtable_sLPS_IFNg, ID)
TNFa      <- pull(subtable_TNFa, ID)
TNFa_P3C  <- pull(subtable_TNFa_P3C, ID)
TNFa_PGE2 <- pull(subtable_TNFa_PGE2, ID)
TPP       <- pull(subtable_TPP, ID)
TPP_IFNb  <- pull(subtable_TPP_IFNb, ID)
TPP_IFNbg <- pull(subtable_TPP_IFNb_IFNg, ID)
upLPS_IC  <- pull(subtable_upLPS_IC, ID)

unique_ids <- unique(c(GC, IFNb, IFNg, IFNg_TNFa, IL10, IL13, IL4, IL4_upLPS, P3C, P3C_PGE2, PGE2, upLPS, sLPS, sLPS_IC, sLPS_IFNg, TNFa, TNFa_P3C, TNFa_PGE2, TPP, TPP_IFNb, TPP_IFNbg, upLPS_IC))

# get gene symbols for each of the unique probe IDs
unique_ids_geneSymbol <- getSYMBOL(unique_ids, 'lumiHumanAll.db') %>%
  enframe() %>%
  dplyr::rename(ID = name, geneSymbol = value)

# make table of differential gene expression t values
dge_frame <- enframe(unique_ids) %>%
  dplyr::rename(ID = value) %>%
  left_join(unique_ids_geneSymbol, by = "ID") %>%
  left_join(subtable_GC, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IFNb, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IFNg, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IFNg_TNFa, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IL10, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IL13, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IL4, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_IL4_upLPS, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_P3C, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_P3C_PGE2, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_PGE2, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_upLPS, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_sLPS, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_sLPS_IC, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_sLPS_IFNg, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_TNFa, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_TNFa_P3C, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_TNFa_PGE2, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_TPP, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_TPP_IFNb, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_TPP_IFNb_IFNg, by = c("ID", "geneSymbol")) %>%
  left_join(subtable_upLPS_IC, by = c("ID", "geneSymbol"))

```
 
