---
title: "Cleaning and clustering experimental macrophage gene expression data"
author: "mleukam"
date: "2019-04-14"
output: workflowr::wflow_html
---


## Introduction

The pupose of this analysis is to find ranked differential gene expression sets for each of a number of experimentally derived macrophage states compared to monocytes. These gene sets will be used for our downstream analyses. 

We are roughly following the methods described in:

> Gubin, Matthew M., Ekaterina Esaulova, Jeffrey P. Ward, Olga N. Malkova, Daniele Runci, Pamela Wong, Takuro Noguchi, et al. 2018. “High-Dimensional Analysis Delineates Myeloid and Lymphoid Compartment Remodeling during Successful Immune-Checkpoint Cancer Therapy.” Cell 175 (4): 1014–30.e19. 

For more details regarding the data source and experimental design, please refer to the [data_download](data_download.html) notebook.

### Setup workspace

Clear workspace prior to analysis and load necessary packages
```{r}
rm(list = ls())
```

```{r packages, message=FALSE, warning=FALSE}
library("sva")
library("tidyverse")
library("limma")
library("lumi")
library("lumiHumanIDMapping")
library("lumiHumanAll.db")
library("annotate")
library("factoextra")
library("FactoMineR")
library("ggsci")
library("ggpubr")
library("WGCNA")

```

## Data preparation and cleaning

### Read in raw expression data
```{r}
raw_input <- read_delim("~/tcga_macs/data/GSE46903_non-normalized.txt", delim = "\t")
```

One source for methods (start on page 23): https://bioconductor.org/packages/release/data/experiment/vignettes/BeadArrayUseCases/inst/doc/BeadArrayUseCases.pdf

NB: "We recommend users work with the non-normalized probe-specific data in their analysis where possible. Illumina’s background correction step, which subtracts the intensities of the negative control probes from the intensities of the regular probes, should also be avoided."

```{r}
# read in data using limma function for illumina beadchips
lumi_input <- lumiR("~/tcga_macs/data/GSE46903_non-normalized.txt",
                    sep = "\t",
                    lib.mapping = 'lumiHumanIDMapping',
                    parseColumnName = TRUE,
                    columnNameGrepPattern = list(
                      exprs = 'AVG_Signal', 
                      se.exprs = 'BEAD_STD', 
                      detection = 'Detection', 
                      beadNum = 'Avg_NBEADS')) 
# review results
dim(lumi_input)
lumi_input$E[1:5, 1:5]
```

### Normalize expression values

 * The original authors collected individual experiments into a single output frame, which would have resulted in background correction from the Illumina Genome Studio software.
 * Due to the lack of 'se.exprs' information, 'log2' transformation will be used.
 * Variance Stabilizing Transform method: log2 - the lumi package takes advantage of controls and technical replicates in the probe list to stabilize variance
 * Normalization method: rsn 
 
Citation for superiority of log2 rsn normalization for this specific beadchip: https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-11-349

```{r message=FALSE, warning=FALSE, results="hide"}
lumi_46903 <- lumiExpresso(lumi_input, 
                         normalize.param = list(method = 'rsn'))

saveRDS(lumi_46903, "~/tcga_macs/output/normalized_probe_intensities.Rds")
```

### Format and subset metadata
```{r paged.print=TRUE}
# review metadata
show(lumi_46903)
raw_metadata <- pData(phenoData(lumi_46903))
head(raw_metadata)
```

The raw data is labeled with sample names (a 10-digit number followed by underscore and a letter), not the GEO accession number, whereas the sample names in the processed data GSE ExpressionSet object are GEO accession numbers.

* I will go back to the published pre-processed GSE in the GEO and see if the metadata there can be merged into an expression set with the expression matrix I generated from the raw values.

* [Documentation for manipulating ExpressionSet objects](https://www.bioconductor.org/packages/devel/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf)

* [Documentation for ExpressionSets created by GEOquery](https://www.bioconductor.org/packages/devel/bioc/vignettes/GEOquery/inst/doc/GEOquery.html#getting-gse-series-matrix-files-as-an-expressionset)

```{r paged.print=TRUE}
# read in previously downloaded GSE containing pre-normalized reads and full metadata
gse46903 <- readRDS("~/tcga_macs/data/gse_46903.rds")
show(pData(phenoData(gse46903[[1]]))[1:20,c(1,6,8)])
```

The sample names correspond to experimental conditions where the 10-digit number is the chip and the letter is the lane, forming a unique identifier for each sample. These are stuck in the same column as the description of the experiment.

* [Documentation for working with AnnotatedDataFrame objects](https://www.rdocumentation.org/packages/Biobase/versions/2.32.0/topics/AnnotatedDataFrame)

```{r phenodata, paged.print=TRUE}
# pull phenoData from expression set as an AnnotatedDataFrame
meta46903 <- phenoData(gse46903[[1]])

# get data and annotations separately from AnnotatedDataFrame
meta_data46903 <- pData(meta46903)
anno_data46903 <- varMetadata(meta46903)

# review metadata
class(meta_data46903)
str(meta_data46903)
head(meta_data46903)

# review annotations
class(anno_data46903)
str(anno_data46903)
head(anno_data46903)

# anno_data is pretty useless - just empty variables with matching rownames from pData. Will drop this from further analysis.
```

Format metadata to get raw_id in its own column
```{r format_metadata}
# split "title" column into cell type and raw_id
meta_data46903 <- meta_data46903 %>%
  tidyr::separate(title, 
                  c("cell_type", "raw_id"), 
                  sep = "\\[")

# review results
meta_data46903[1:5, c(1,2)]

# remove trailing closed bracket
# preserve rownames by moving to column
meta_data46903 <- meta_data46903 %>%
  rownames_to_column(var = "geo_id") %>%
  mutate(raw_id = str_remove(raw_id, "]"))

# review results
meta_data46903[1:5, 1:5]
```

Subset metadata 
```{r}
# use macrophages cultured for 72 hours (endpoint)
mac_metadata <- meta_data46903 %>%
  as_tibble() %>%
  filter(`initial differentiation:ch1` %in% c("GM-CSF", "M-CSF"),
         grepl("72h", characteristics_ch1.2)) %>%
  print()

# get the raw IDs to select expression profiles later
mac_samples <- mac_metadata %>% pull(raw_id)

# convert back to data.frame and add rownames back in 
# for compatability with expressionset class
mac_metadata <- as.data.frame(mac_metadata)
rownames(mac_metadata) <- mac_metadata$raw_id
mac_metadata[1:5, 1:5]
```


### Subset expression data
Next step is to subset the expression data, and to ensure column names exactly match rownames of phenotype data

Review and subset expression matrix
```{r}
exp_data <- exprs(lumi_46903)
  
# review expression matrix
class(exp_data)
str(exp_data)
exp_data[1:10, 1:3]

# rownames are unique sequence identifiers from lumiHumanIDmapping package
# colnames are 10-digit and letter sample names
# values are log-2 RDS normalized expression values 
# 384 samples and 47462 probes

# convert to tbl for subsetting
exp_data <- exp_data %>%
  as.data.frame() %>%
  rownames_to_column(var = "probe_id") %>%
  as_tibble()

# remove probe_id for subsetting, will add back in aferwards
probe_id <- exp_data %>% pull(probe_id)
  
# use vector of sample names to select cases  
mac_exprs <- exp_data %>%
  dplyr::select(one_of(mac_samples)) %>%
  add_column(probe_id, .before = "4487653414_E") 

# compare representative columns to ensure the order wasn't lost
mac_exprs %>%
  dplyr::select(probe_id, "4487653414_E", "4487653415_E") %>%
  print()
exp_data %>%
  dplyr::select(probe_id, "4487653414_E", "4487653415_E") %>%
  print()

# they match exactly

# convert back to data.frame and add rownames
mac_exprs <- mac_exprs %>%
  as.data.frame()
rownames(mac_exprs) <- mac_exprs$probe_id

# double check that rownames match probe_id
mac_exprs[1:5, 1:5]

# remove temporary probe_id column
mac_exprs <- subset(mac_exprs, select = -probe_id)
mac_exprs[1:5, 1:5]
```

Ensure that rownames of metadata and column names of expression data match exactly
```{r}
# are they the same length?
length(rownames(mac_metadata))
length(colnames(mac_exprs))

# ask R to check identity
identical(rownames(mac_metadata), colnames(mac_exprs))

# eyeball a few too
rownames(mac_metadata)[1:10]
colnames(mac_exprs)[1:10]

# looks good!
```

### Build ExpressionSet

Following the [documentation](https://www.bioconductor.org/packages/3.7/bioc/vignettes/Biobase/inst/doc/ExpressionSetIntroduction.pdf) for Biconductor ExpressionSet

```{r}
# create a frame for annotation of column names in metadata
# will be empty for now
varnames <- colnames(mac_metadata)
vardata <- data.frame(labelDescription = varnames, 
                      row.names = varnames)

# build an AnnotatedDataFrame
mac_phenoData <- new("AnnotatedDataFrame",
                       data = mac_metadata, 
                       varMetadata = vardata)

# convert expression data to matrix
mac_exprs <- as.matrix(mac_exprs)


mac_eset <- ExpressionSet(assayData = mac_exprs, 
                            phenoData = mac_phenoData,
                            annotation = "Illumina Human HT-12 v3.0 beadchip")

# check properties
featureNames(mac_eset)[1:5]
sampleNames(mac_eset)[1:5]
varLabels(mac_eset)
mat <- exprs(mac_eset)
dim(mat)
mat[1:5, 1:5]

# save expression set
saveRDS(mac_eset, "~/tcga_macs/output/expression_set.rds")
```

## Data review, filtering, batch correction

### Density plots

```{r}
# load data
pheno <- pData(mac_eset)
edata <- exprs(mac_eset)

# tidy data
edata_tbl <- edata %>% 
  as.data.frame() %>%
  rownames_to_column(var = "probeID") %>%
  as_tibble() %>%
  gather(key = "sampleID", value = "intensity", -probeID) %>%
  group_by(sampleID)

# find "valley" in density plot for low value filtering
dplot <- ggplot(edata_tbl, aes(intensity, color = sampleID)) +
  geom_density() + 
  theme(legend.position = "none")

dplot

# zoom in on peaks
dplot + 
  coord_cartesian(xlim = c(5.5, 7), ylim = c(0, 2.5)) + 
  scale_x_continuous(breaks = c(5.9, 6.0, 6.1, 6.2, 6.3, 6.4, 6.5))
```

There are two peaks in the density plot! Will need to find which sample corresponds to which peak and assign a variable for batch correction.

```{r}
# get the density function for each sample
dens <- edata_tbl %>% 
  split(.$sampleID) %>% 
  map(function(df) density(df$intensity))

# find the max y
ymax <- dens %>% 
  map(function(df) max(df$y))

# extract the corresponding x for the max y density for each
xmax <- dens %>% 
  map(function(df) df$x[df$y == max(df$y)])

# quick plot of results
hist(unlist(ymax))
hist(unlist(xmax))
```

### Define Groups
```{r}
# Get phenotype data from expressionset
pheno_data <- pData(mac_eset)
str(pheno_data)

# OK if rownames are dropped, column is duplicated as "raw_id"
pheno_data <- pheno_data %>%
  as_tibble() %>%
# clean up experimental group data
  mutate(treatment = str_sub(cell_type, 9, 25),
         treatment = str_replace(treatment, "_72h", ""),
         treatment = ifelse(`activation stimuli:ch1` == "con", "control", treatment),
         treatment = trimws(treatment)) %>%
  dplyr::select(raw_id, geo_id, treatment, everything()) %>%
  print()

# assign treatments into clusters from Xue et al PMID: 24530056
pheno_data <- pheno_data %>%
  mutate(treatment = as.factor(treatment),
         clusternum = fct_recode(treatment,
                                 "0" = "control",
                                 "1" = "IFNb",
                                 "1" = "IL10",
                                 "1" = "GC",
                                 "2" = "IL4",
                                 "2" = "L4",
                                 "2" = "IL13",
                                 "2" = "IL4_upLPS",
                                 "2" = "L4_upLPS",
                                 "3" = "upLPS_IC",
                                 "3" = "upLPS",
                                 "3" = "pLPS",
                                 "4" = "P3C_PGE2",
                                 "4" = "P3C",
                                 "4" = "PGE2",
                                 "6" = "IFNg",
                                 "6" = "TNFa",
                                 "6" = "IFNg_TNFa",
                                 "7" = "sLPS",
                                 "7" = "sLPS_IFNg",
                                 "7" = "sLPS_IC",
                                 "8" = "TNFa_PGE2",
                                 "8" = "TNFa_P3C",
                                 "9" = "TPP",
                                 "9" = "TPP_IFNb",
                                 "9" = "TPP_IFNb_IFNg"),
         treatment = fct_recode(treatment, 
                                "IL4_upLPS" = "L4_upLPS",
                                "upLPS" = "pLPS",
                                "IL4" = "L4")) %>%
  dplyr::select(raw_id, geo_id, treatment, clusternum, everything()) %>%
  print()

summary(pheno_data$treatment)
summary(pheno_data$clusternum)
```

### Evaluate for batch effect

Source: sva vignette and user's manual: http://bioconductor.org/packages/release/bioc/vignettes/sva/inst/doc/sva.pdf

The sva function returns a list with four components: `sv`, `pprob.gam`, `pprob.b`, `n.sv`. 

* sv is a matrix whose columns correspond to the estimated surrogate variables.
* pprob.gam is the posterior probability that each gene is associated with one or more latent variables. 
* pprob.b is the posterior probability that each gene is associated with the variables of interest. 
* n.sv is the number of surrogate variables estimated by the sva.

```{r}
# create full model matrix
mod <- model.matrix(~ as.factor(treatment), data = pheno_data)

# create null model
mod0 <- model.matrix(~ 1, data = pheno_data)

# apply the sva function to estimate the surrogate variables
sv_obj <- sva(edata, mod, mod0)

# calculate parametric F-test P-values for each row of data matrix
pValues <- f.pvalue(edata, mod, mod0)
qValues <- p.adjust(pValues, method = "BH")

```

From the SVA users' manual:

_The goal of the sva is to remove all unwanted sources of variation while protecting the contrasts due to the primary variables included in mod. This leads to the identification of features that are consistently different between groups, removing all common sources of latent variation.
In some cases, the latent variables may be important sources of biological variability. If the goal of the analysis is to identify heterogeneity in one or more subgroups, the sva function may not be appropriate. For example, suppose that it is expected that cancer samples represent two distinct, but unknown subgroups. If these subgroups have a large impact on expression, then one or more of the estimated surrogate variables may be very highly correlated with subgroup. In contrast, direct adjustment only removes the effect of known batch variables. All sources of latent biological variation will remain in the data using this approach. In other words, if the samples were obtained in different environments, this effect will remain in the data. If important sources of heterogeneity (from different environments, lab effects, etc.) are not accounted for, this may lead to increased false positives._

At this point, it would probably be best not to apply a global batch correction to avoid loss of biologic heterogeneity. There is not clear evidence of a small number of groups that may reflect a batch effect.

### Filter probes
```{r}
# retreive normalized data
dataMatrix <- exprs(mac_eset)

# remove the probes that are not detected in raw data
presentCount <- detectionCall(lumi_46903)
selDataMatrix <- dataMatrix[presentCount > 0,]
probeList <- rownames(selDataMatrix)
```

Detailed instructions for building a contrast matrix and filtering low-expression scores can be found in the [Limma User's Guide](http://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf), particularly chapters 9.2 and 9.3

## PCA

Source: https://rpkgs.datanovia.com/factoextra/index.html

```{r}
# transpose matrix to tidy data format
t_DataMatrix <- t(selDataMatrix)
t_DataMatrix[1:5, 1:5]

# add categories and factors
# select columns of interest from phenotype data
pheno_pca <- pheno_data %>%
  dplyr::select(raw_id, treatment, clusternum)

df_pca <- t_DataMatrix %>%
  as.data.frame() %>%
  rownames_to_column(var = "raw_id") %>%
  left_join(pheno_pca, by = "raw_id") %>%
  as_tibble() %>%
  dplyr::select(raw_id, treatment, clusternum, everything()) %>%
  print()

norm_probe_pca <- PCA(df_pca[,c(4:39471)], graph = FALSE)
head(get_eig(norm_probe_pca))
fviz_eig(norm_probe_pca)
```

Source for overriding shape change with habillage argument: https://github.com/kassambara/factoextra/issues/20

```{r}
# visualize eigenvalues
mypal <- pal_igv("default", alpha = 1)(23)
pca1 <- fviz_pca_ind(norm_probe_pca,
             label = "none", # hide individual labels
             habillage = "none",
             col.ind = df_pca$treatment,
             palette = mypal,
             geom.ind = "point",
             invisible = "quali",
             pointshape = 19,
             title = "PCA of normalized expression"
) +
  labs(subtitle = "By activation agent")
print(pca1)

pca2 <- fviz_pca_ind(norm_probe_pca,
             label = "none", # hide individual labels
             habillage = "none",
             col.ind = df_pca$clusternum,
             palette = mypal,
             geom.ind = "point",
             invisible = "quali",
             pointshape = 19, 
             addEllipses = TRUE,
             title = "PCA of normalized expression"
) +
  labs(subtitle = "By cluster assignment")
print(pca2)
```


## Correlation Coefficient Matrix Analysis

Background and references:
```
Method source for correlation matrix: https://davetang.org/muse/2012/01/31/creating-a-correlation-matrix-with-r/

More on constructing a correlation matrix:
https://www.researchgate.net/post/How_to_calculate_Gene-Gene_Pearson_correlation

WGCNA package: https://horvath.genetics.ucla.edu/html/CoexpressionNetwork/Rpackages/WGCNA/Tutorials/index.html
```

Following Xue et al Immunity (2014), select the 1000 probes with the highest variance to reduce the computational load to make a correlation matrix

Source for selecting highest variance: http://r.789695.n4.nabble.com/Help-on-selecting-genes-showing-highest-variance-td3580645.html

```{r}
# select 1000 highest probes with highest variation
# input is normalized expression matrix with zero expression genes removed
myvars <- apply(selDataMatrix, 1, var, na.rm = TRUE) 
myvars <- sort(myvars, decreasing = TRUE) 
myvars <- myvars[1:1000] 
data_matrix_onek <- selDataMatrix[names(myvars), ] 
dim(data_matrix_onek) 
```

Quick look at clustering
```{r}
# heirarchical cluster
datExpr <- t(data_matrix_onek)
sampleTree = hclust(dist(datExpr), method = "average")
sizeGrWindow(16,9)
par(cex = 0.4)
plot(sampleTree, main = "Sample clustering to detect outliers", 
     sub = "", 
     xlab = "", 
     cex.lab = 1.5,
     cex.axis = 1.5, 
     cex.main = 2.5)

# no significant outliers
```

Set up phenotype data for WGCNA
```{r}
# use pheno_data from above with treatment identification and cluster number
# select columns I want to visualize
traitData <- pheno_data %>%
  dplyr::select(raw_id, treatment, clusternum) %>%
  as.data.frame() %>%
  column_to_rownames("raw_id") %>%
  print()

treat_colors <- pal_igv("default", alpha = 1)(23)
names(treat_colors) <- levels(traitData$treatment)
treat_colors <- enframe(treat_colors) %>%
  dplyr::rename(treatment = name, treat_colors = value)
clust_colors <- pal_igv("default", alpha = 1)(9)
names(clust_colors) <- levels(traitData$clusternum)
clust_colors <- enframe(clust_colors) %>%
  dplyr::rename(clusternum = name, clust_colors = value)

traitColors <- pheno_data %>% 
  left_join(treat_colors, by = "treatment") %>%
  left_join(clust_colors, by = "clusternum") %>%
  dplyr::select(raw_id, treat_colors, clust_colors) %>%
  as.data.frame() %>%
  column_to_rownames("raw_id") %>%
  print()

```

```{r}
sampleTree2 <- hclust(dist(datExpr), method = "average")
plotDendroAndColors(sampleTree2, traitColors,
                    groupLabels = names(traitData), 
                    main = "Sample dendrogram and treatment assignments")

```
