---
title: "ssGSEA"
author: "mleukam"
date: "2019-06-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

## Setup

Clear environment
```{r}
# clear environment
rm(list = ls())
```

Load packages
```{r}
library("tidyverse")
library("edgeR")
library("limma")
library("GSVA")
library("GenomicDataCommons")
```


## Gene list preprocessing

Read in custom gene lists
```{r}
up_files <- list.files("output/", "^upreg_c")
down_files <- list.files("output/", "downreg_c")
file_list <- c(up_files, down_files)
path_list <- paste0("output/", file_list)
dge_list <- map(path_list, read_csv)
names(dge_list) <- str_replace_all(file_list, ".csv", "")
```

Read in lookup table for features, gencode v22 (used by GDC to label features)
```{r}
gencode_gtf <- read_tsv("data/gencode.v22.primary_assembly.annotation.gtf.geneinfo")
```


```{r}
# read in T-cell inflammation signature
tcell <- read_tsv("data/160genes_ensembl.gencode28.txt", col_names = FALSE)

# convert to gene_id
gene_ids <- gencode_gtf %>%
  dplyr::select(gene_id, gene_name)
tcell <- tcell %>% dplyr::rename(gene_name = X1)
tcell_gset <- tcell %>% left_join(gene_ids, by = "gene_name") %>%
  pull(gene_id)

tcell_gset <- tcell_gset[!is.na(tcell_gset)]
```

```{r}

# define function that uses lookup table to convert gene symbols to gene id
# then returns a list of gene ids that define the gene set
ensemblgset <- function(df, gtf){
  gene_ids <- gtf %>%
  dplyr::select(gene_id, gene_name) %>%
  dplyr::rename(gene = gene_name)
gset <- df %>% left_join(gene_ids) %>%
  pull(gene_id)
gset
}

# apply function to list of gene sets
gset_ids <- map(dge_list, function(x){ensemblgset(x, gtf = gencode_gtf)})

# remove nas (small number of gene symbols do not map to gencode gene ids)
gset_ids <- map(gset_ids, function(x){
  x[!is.na(x)]
})

# add T-cell signature to gset list
gset_ids_complete <- c(gset_ids, tcell_gset = list(tcell_gset))
  
```

## Count data preprocessing

Preprocessing following methods outlined here: https://f1000research.com/articles/5-1408/v3

Read in data
```{r}
total_counts <- read_csv("output/selected.total.counts.csv")
```

#### Filter for protein coding genes
```{r}

# filter for protein coding genes
total_counts_prcode <- total_counts %>%
  dplyr::rename(gene_id = gene) %>%
  left_join(gencode_gtf, by = "gene_id") %>%
  dplyr::filter(gene_type == "protein_coding") %>%
  dplyr::select(-gene_name, -gene_type, -gene_status, -level, -havana_gene) %>%
  dplyr::select(gene_id, everything())

nrow(total_counts_prcode)

```

#### Correct for library size: convert to CPM
```{r}
# normalize rows by log cpm using EdgeR
df_data <- total_counts_prcode %>% 
  dplyr::select(-gene_id) %>% as.matrix()
df_names <- total_counts_prcode %>% dplyr::select(gene_id) 
out_data <- cpm(df_data, log = FALSE) %>% as_tibble()
total_counts_prcode_cpm <- bind_cols(df_names, out_data)
```

#### Density plots
```{r}
# Density plots
# tidy data
tidy_cpm <- total_counts_prcode_cpm %>% 
  gather(key = "sampleID", value = "intensity", -gene_id) 

tidy_cpm <- tidy_cpm %>%
  mutate(group = str_sub(sampleID, 1, 4))

# representative plots
brca_cpm <- tidy_cpm %>%
  dplyr::filter(group == "BRCA")
dplot_brca <- ggplot(brca_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = "none")
dplot_brca

gbm_cpm <- tidy_cpm %>%
  dplyr::filter(grepl("GBM*", group))
dplot_gbm <- ggplot(gbm_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = "none") +
  xlim(-6, 15)
dplot_gbm
```

#### Filter by expression levels
```{r}
# move gene names to rownames
totcounts_prcode_cpm_matrix <- total_counts_prcode_cpm %>%
  as.data.frame() %>%
  column_to_rownames(var = "gene_id")
totcounts_prcode_cpm_matrix[1:5, 1:5]

# filter out genes that aren't at least expressed greater than 1 in at least 45 cases (45 being about the size of the smallest group)
total_cpm_stats <- data.frame(
  total = apply(totcounts_prcode_cpm_matrix, 1, function(x) sum(x > 1, na.rm = TRUE)))
keep <- which(total_cpm_stats$total >= 45) 

dim(totcounts_prcode_cpm_matrix)
total_cpm_filtered = totcounts_prcode_cpm_matrix[keep,]
dim(total_cpm_filtered)
```

#### Normalize gene expression distributions

Normalisation by the method of trimmed mean of M-values (TMM) is performed using the calcNormFactors function in edgeR. The normalisation factors calculated here are used as a scaling factor for the library sizes. 
```{r}
# get normalization factors
norm_factors <- calcNormFactors(total_cpm_filtered, method = "TMM")

# apply factor to each column
total_cpm_norm <- map2_dfc(total_cpm_filtered, norm_factors, `*`)
total_cpm_norm <- as.data.frame(total_cpm_norm)
rownames(total_cpm_norm) <- rownames(total_cpm_filtered)
total_cpm_norm[1:5, 1:5]
```

#### Log transformation
```{r}
total_log_cpm_filtered_norm <- log2(total_cpm_norm)
```

#### Review representative density plots
```{r}
# Density plots
# tidy data
final_total_df <- total_log_cpm_filtered_norm %>%
  rownames_to_column(var = "gene_id") %>%
  dplyr::select(gene_id, everything())

tidy_norm_cpm <- final_total_df %>% 
  gather(key = "sampleID", value = "intensity", -gene_id) 

tidy_norm_cpm <- tidy_norm_cpm %>%
  mutate(group = str_sub(sampleID, 1, 4))

# representative plots
brca_norm_cpm <- tidy_norm_cpm %>%
  dplyr::filter(group == "BRCA")
dplot_norm_brca <- ggplot(brca_norm_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = "none")
dplot_norm_brca

gbm_norm_cpm <- tidy_norm_cpm %>%
  dplyr::filter(grepl("GBM*", group))
dplot_norm_gbm <- ggplot(gbm_norm_cpm, aes(intensity)) +
  geom_density() + 
  theme(legend.position = "none") +
  xlim(-6, 15)
dplot_norm_gbm
```


## GSVA
Will be done on cluster due to extreme system requirements
Write out matrix and gene set lists for import into Gardner HPC cluster.
```{r}
# gset list
saveRDS(gset_ids_complete, "output/gset_ids_complete.rds")

expr_matrix <- total_log_cpm_filtered_norm %>% 
  as.data.frame() %>%
  rownames_to_column(var = "gene")

write_csv(expr_matrix, "output/expr_matrix.csv")

```


Get macrophage gene set GSVA scores
```{r eval = FALSE, include = TRUE}

# Run on cluster
# tcga_es <- gsva(expr_matrix, gset_ids_complete,
#                annotation = NULL,
#                method = "gsva",
#                mx.diff = FALSE,
#                verbose = TRUE)

# saveRDS(tcga_es, "/gpfs/data/kline-lab/tcga_macs/tcga_es.rds")
# moved to local hard drive from cluster and stored in output folder

```

